diff -Nrub -x '*~' -x '*.rej' -x '*.orig' -x '*.o' -x '*.pyc' -x CMakeLists.txt.user openjpeg-2.5.2.orig/CMakeLists.txt openjpeg-2.5.2/CMakeLists.txt
--- openjpeg-2.5.2.orig/CMakeLists.txt	2024-02-28 14:32:43.000000000 +0100
+++ openjpeg-2.5.2/CMakeLists.txt	2025-04-15 10:25:33.545800000 +0200
@@ -327,6 +327,12 @@
 set_variable_from_rel_or_absolute_path("libdir" "\\\${prefix}" "${CMAKE_INSTALL_LIBDIR}")
 set_variable_from_rel_or_absolute_path("includedir" "\\\${prefix}" "${CMAKE_INSTALL_INCLUDEDIR}/${OPENJPEG_INSTALL_SUBDIR}")
 
+if(WIN32)
+  set(private_libs)
+else()
+  set(private_libs "-lm")
+endif()
+
 # install in lib and not share (CMAKE_INSTALL_LIBDIR takes care of it for multi-arch)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/lib/openjp2/libopenjp2.pc.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/libopenjp2.pc @ONLY)
diff -Nrub -x '*~' -x '*.rej' -x '*.orig' -x '*.o' -x '*.pyc' -x CMakeLists.txt.user openjpeg-2.5.2.orig/src/lib/openjp2/libopenjp2.pc.cmake.in openjpeg-2.5.2/src/lib/openjp2/libopenjp2.pc.cmake.in
--- openjpeg-2.5.2.orig/src/lib/openjp2/libopenjp2.pc.cmake.in	2024-02-28 14:32:43.000000000 +0100
+++ openjpeg-2.5.2/src/lib/openjp2/libopenjp2.pc.cmake.in	2025-04-15 10:24:46.242610100 +0200
@@ -10,6 +10,6 @@
 URL: http://www.openjpeg.org/
 Version: @OPENJPEG_VERSION@
 Libs: -L${libdir} -lopenjp2
-Libs.private: -lm
+Libs.private: @private_libs@
 Cflags: -I${includedir}
 Cflags.private: -DOPJ_STATIC
