diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' zlib-1.2.11.orig/zconf.h.cmakein zlib-1.2.11/zconf.h.cmakein
--- zlib-1.2.11.orig/zconf.h.cmakein	2017-01-01 08:37:10.000000000 +0100
+++ zlib-1.2.11/zconf.h.cmakein	2019-05-08 08:55:18.373443700 +0200
@@ -433,11 +433,11 @@
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_UNISTD_H) && HAVE_UNISTD_H != 0    /* may be set to #if 1 by ./configure */
 #  define Z_HAVE_UNISTD_H
 #endif
 
-#ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_STDARG_H) && HAVE_STDARG_H != 0    /* may be set to #if 1 by ./configure */
 #  define Z_HAVE_STDARG_H
 #endif
 
