diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' qtbase-opensource-src-5.9.4.orig/.gitignore qtbase-opensource-src-5.9.4/.gitignore
--- qtbase-opensource-src-5.9.4.orig/.gitignore	1970-01-01 01:00:00.000000000 +0100
+++ qtbase-opensource-src-5.9.4/.gitignore	2018-02-21 09:04:44.525023300 +0100
@@ -0,0 +1 @@
+Force Bootstrap
\ No newline at end of file
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' qtbase-opensource-src-5.9.4.orig/src/corelib/io/qstandardpaths.cpp qtbase-opensource-src-5.9.4/src/corelib/io/qstandardpaths.cpp
--- qtbase-opensource-src-5.9.4.orig/src/corelib/io/qstandardpaths.cpp	2018-01-16 07:53:43.000000000 +0100
+++ qtbase-opensource-src-5.9.4/src/corelib/io/qstandardpaths.cpp	2018-02-21 09:04:44.526041800 +0100
@@ -183,7 +183,7 @@
          \li "C:/Users/<USER>"
     \row \li DataLocation
          \li "~/Library/Application Support/<APPNAME>", "/Library/Application Support/<APPNAME>". "<APPDIR>/../Resources"
-         \li "C:/Users/<USER>/AppData/Local/<APPNAME>", "C:/ProgramData/<APPNAME>", "<APPDIR>", "<APPDIR>/data"
+         \li "C:/Users/<USER>/AppData/Local/<APPNAME>", "C:/ProgramData/<APPNAME>", "<APPDIR>", "<APPDIR>/data", "<APPDIR>/data/<APPNAME>"
     \row \li CacheLocation
          \li "~/Library/Caches/<APPNAME>", "/Library/Caches/<APPNAME>"
          \li "C:/Users/<USER>/AppData/Local/<APPNAME>/cache"
@@ -207,10 +207,10 @@
          \li "C:/Users/<USER>/AppData/Local/cache"
     \row \li AppDataLocation
          \li "~/Library/Application Support/<APPNAME>", "/Library/Application Support/<APPNAME>". "<APPDIR>/../Resources"
-         \li "C:/Users/<USER>/AppData/Roaming/<APPNAME>", "C:/ProgramData/<APPNAME>", "<APPDIR>", "<APPDIR>/data"
+         \li "C:/Users/<USER>/AppData/Roaming/<APPNAME>", "C:/ProgramData/<APPNAME>", "<APPDIR>", "<APPDIR>/data", "<APPDIR>/data/<APPNAME>"
     \row \li AppLocalDataLocation
          \li "~/Library/Application Support/<APPNAME>", "/Library/Application Support/<APPNAME>". "<APPDIR>/../Resources"
-         \li "C:/Users/<USER>/AppData/Local/<APPNAME>", "C:/ProgramData/<APPNAME>", "<APPDIR>", "<APPDIR>/data"
+         \li "C:/Users/<USER>/AppData/Local/<APPNAME>", "C:/ProgramData/<APPNAME>", "<APPDIR>", "<APPDIR>/data", "<APPDIR>/data/<APPNAME>"
     \row \li AppConfigLocation
          \li "~/Library/Preferences/<APPNAME>"
          \li "C:/Users/<USER>/AppData/Local/<APPNAME>", "C:/ProgramData/<APPNAME>"
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' qtbase-opensource-src-5.9.4.orig/src/corelib/io/qstandardpaths_win.cpp qtbase-opensource-src-5.9.4/src/corelib/io/qstandardpaths_win.cpp
--- qtbase-opensource-src-5.9.4.orig/src/corelib/io/qstandardpaths_win.cpp	2018-01-16 07:53:43.000000000 +0100
+++ qtbase-opensource-src-5.9.4/src/corelib/io/qstandardpaths_win.cpp	2018-02-21 09:12:43.711248700 +0100
@@ -227,7 +227,15 @@
         QString applicationDirPath = qApp ? QCoreApplication::applicationDirPath()
             : QFileInfo(qAppFileName()).path();
         dirs.append(applicationDirPath);
-        dirs.append(applicationDirPath + QLatin1String("/data"));
+        const QString dataDir = applicationDirPath + QLatin1String("/data");
+        dirs.append(dataDir);
+
+        if (!isGenericConfigLocation(type)) {
+            QString appDataDir = dataDir;
+            appendOrganizationAndApp(appDataDir);
+            if (appDataDir != dataDir)
+                dirs.append(appDataDir);
+        }
 #endif // !QT_BOOTSTRAPPED
     } // isConfigLocation()

