From 38e65d86d62422d66ed0c7235c1728e28940b420 Mon Sep 17 00:00:00 2001
From: Vincent Pinon <vpinon@kde.org>
Date: Sun, 3 Oct 2021 17:08:02 +0200
Subject: [PATCH] Allow building on Linux without libmount & blkid

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b2c0783..7b4c879 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,8 +132,8 @@ set_package_properties(ACL PROPERTIES DESCRIPTION "LibACL"
 
 # Used by KMountPoint
 if (CMAKE_SYSTEM_NAME MATCHES "Linux")
-    find_package(LibMount REQUIRED)
-    find_package(Blkid REQUIRED)
+    find_package(LibMount)
+    find_package(Blkid)
     set(HAVE_LIBS_MOUNT_AND_BLKID ${LibMount_FOUND} AND ${Blkid_FOUND})
 endif()
 
-- 
1.8.3.1

