--- a/cmake/OpenCVGenPkgconfig.cmake	2020-05-01 12:35:26.761330100 -0700
+++ b/cmake/OpenCVGenPkgconfig.cmake	2020-05-01 12:36:24.592119400 -0700
@@ -103,7 +103,7 @@
 add_dependencies(developer_scripts gen-pkgconfig)
 
 
-if(UNIX AND NOT ANDROID)
+if(NOT ANDROID)
   install(FILES ${CMAKE_BINARY_DIR}/unix-install/${OPENCV_PC_FILE_NAME} DESTINATION ${OPENCV_LIB_INSTALL_PATH}/pkgconfig COMPONENT dev)
 endif()
 
