commit 039be0bcb41be4b2311d3d16c54ecfe519ed02ab
Author: Mladen Milinkovic <maxrd2@smoothware.net>
Date:   Mon Mar 9 11:58:11 2020 +0100

    Fix make install with alernate root path

diff --git a/deploy.pri b/deploy.pri
index 3abbde99..0efa9ea4 100644
--- a/deploy.pri
+++ b/deploy.pri
@@ -4,7 +4,7 @@ INSTALL_PREFIX = /usr/local
 share.files = $$PROJECTROOT/qtc_packaging/common/changelog \
 			$$PROJECTROOT/qtc_packaging/common/copyright \
 			$$PROJECTROOT/qtc_packaging/common/README
-share.path = /usr/share/doc/$${TARGET}
+share.path = $$[QT_INSTALL_PREFIX]/share/doc/$${TARGET}
 
 isEqual(TEMPLATE, app) {
 	unix:!symbian {
@@ -12,22 +12,22 @@ isEqual(TEMPLATE, app) {
 			DEFINES += CACHE_APPDIR
 			INSTALL_PREFIX = /opt/$${TARGET}
 			desktopfile.files = $$PROJECTROOT/qtc_packaging/debian_harmattan/$${TARGET}.desktop
-			desktopfile.path = /usr/share/applications
+			desktopfile.path = $$[QT_INSTALL_PREFIX]/share/applications
 			icon.files = $$PROJECTROOT/qtc_packaging/debian_harmattan/$${TARGET}.png
-			icon.path = /usr/share/icons/hicolor/80x80/apps
+			icon.path = $$[QT_INSTALL_PREFIX]/share/icons/hicolor/80x80/apps
 			#debian.files = $$PROJECTROOT/qtc_packaging/harmattan/control
 		} else:maemo5 {
 			INSTALL_PREFIX = /opt/$${TARGET}
 			desktopfile.files = $$PROJECTROOT/qtc_packaging/debian_fremantle/$${TARGET}.desktop
-			desktopfile.path = /usr/share/applications/hildon
+			desktopfile.path = $$[QT_INSTALL_PREFIX]/share/applications/hildon
 			icon.files = $$PROJECTROOT/qtc_packaging/debian_fremantle/$${TARGET}.png
-			icon.path = /usr/share/icons/hicolor/64x64/apps
+			icon.path = $$[QT_INSTALL_PREFIX]/share/icons/hicolor/64x64/apps
 			#debian.files = $$PROJECTROOT/qtc_packaging/fremantle/control
 		} else {
 			desktopfile.files = $$PROJECTROOT/qtc_packaging/debian_generic/$${TARGET}.desktop
-			desktopfile.path = /usr/share/applications
+			desktopfile.path = $$[QT_INSTALL_PREFIX]/share/applications
 			icon.files = $$PROJECTROOT/qtc_packaging/debian_generic/$${TARGET}.png
-			icon.path = /usr/share/icons/hicolor/64x64/apps
+			icon.path = $$[QT_INSTALL_PREFIX]/share/icons/hicolor/64x64/apps
 			#debian.files = $$PROJECTROOT/qtc_packaging/generic/control
 		}
 		INSTALLS += desktopfile icon
diff --git a/examples/QMLPlayer/qtquick2applicationviewer/qtquick2applicationviewer.pri b/examples/QMLPlayer/qtquick2applicationviewer/qtquick2applicationviewer.pri
index e5f7990f..47a18892 100644
--- a/examples/QMLPlayer/qtquick2applicationviewer/qtquick2applicationviewer.pri
+++ b/examples/QMLPlayer/qtquick2applicationviewer/qtquick2applicationviewer.pri
@@ -104,14 +104,14 @@ android-no-sdk {
 } else:unix {
     maemo5 {
         desktopfile.files = $${TARGET}.desktop
-        desktopfile.path = /usr/share/applications/hildon
+        desktopfile.path = $$[QT_INSTALL_PREFIX]/share/applications/hildon
         icon.files = $${TARGET}64.png
-        icon.path = /usr/share/icons/hicolor/64x64/apps
+        icon.path = $$[QT_INSTALL_PREFIX]/share/icons/hicolor/64x64/apps
     } else:!isEmpty(MEEGO_VERSION_MAJOR) {
         desktopfile.files = $${TARGET}_harmattan.desktop
-        desktopfile.path = /usr/share/applications
+        desktopfile.path = $$[QT_INSTALL_PREFIX]/share/applications
         icon.files = $${TARGET}80.png
-        icon.path = /usr/share/icons/hicolor/80x80/apps
+        icon.path = $$[QT_INSTALL_PREFIX]/share/icons/hicolor/80x80/apps
     } else { # Assumed to be a Desktop Unix
         copyCommand =
         for(deploymentfolder, DEPLOYMENTFOLDERS) {
diff --git a/src/libQtAV.pro b/src/libQtAV.pro
index 8e49668c..8361f628 100644
--- a/src/libQtAV.pro
+++ b/src/libQtAV.pro
@@ -621,7 +621,7 @@ mac {
 
 unix:!mac:!cross_compile {
 icon.files = $$PWD/$${TARGET}.svg
-icon.path = /usr/share/icons/hicolor/scalable/apps
+icon.path = $$[QT_INSTALL_PREFIX]/share/icons/hicolor/scalable/apps
 INSTALLS += icon
 #debian
 DEB_INSTALL_LIST = .$$[QT_INSTALL_LIBS]/libQt*AV.so.*
