From 1691274e52b8a76f760bad1333746fde4f5973c7 Mon Sep 17 00:00:00 2001
From: Alexander Stippich <a.stippich@gmx.net>
Date: Mon, 20 Jan 2020 08:11:13 +0100
Subject: [PATCH] disable playlist animations to fix display corruption

BUG: 398093
BUG: 406524
---
 src/qml/PlayListBasicView.qml | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/qml/PlayListBasicView.qml b/src/qml/PlayListBasicView.qml
index 6c50f227..c3491f54 100644
--- a/src/qml/PlayListBasicView.qml
+++ b/src/qml/PlayListBasicView.qml
@@ -61,6 +61,10 @@ ListView {
         anchors.fill: playListView
     }
 
+    /* currently disabled animations due to display corruption
+      because of https://bugreports.qt.io/browse/QTBUG-49868
+      causing https://bugs.kde.org/show_bug.cgi?id=406524
+      and https://bugs.kde.org/show_bug.cgi?id=398093      
     add: Transition {
         NumberAnimation {
             property: "opacity";
@@ -91,6 +95,7 @@ ListView {
             duration: 100;
             easing.type: Easing.InOutQuad}
     }
+    */
 
     model: DelegateModel {
         id: playListModelDelegate
-- 
2.25.0

