From 0199736d72be705903a9979c35d82c48bbea7015 Mon Sep 17 00:00:00 2001
From: Hannah von Reth <vonreth@kde.org>
Date: Wed, 20 Dec 2017 15:48:42 +0100
Subject: [PATCH] Also look for qrencode with debug suffix

---
 cmake/FindQRencode.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/FindQRencode.cmake b/cmake/FindQRencode.cmake
index 0c37b03..b8ee0ac 100644
--- a/cmake/FindQRencode.cmake
+++ b/cmake/FindQRencode.cmake
@@ -54,7 +54,7 @@
 # THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #=============================================================================

-find_library(QRencode_LIBRARIES NAMES qrencode)
+find_library(QRencode_LIBRARIES NAMES qrencode qrencoded)

 find_path(QRencode_INCLUDE_DIRS qrencode.h)

--
2.15.0.windows.1

