commit 2cc4e26cb9c07cf71585600287a58652330050bf
Author: Eric Dejouhanet <eric.dejouhanet@gmail.com>
Date:   Wed Jun 19 11:34:52 2024 +0200

    fix(ci): add cURL libraries to cfitsio libraries.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f98842cd9..33d36a591 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -208,6 +208,12 @@ endif()
 MACRO_BOOL_TO_01(CFITSIO_FOUND HAVE_CFITSIO)
 set_package_properties(CFitsio PROPERTIES DESCRIPTION "FITS IO Library" URL "https://heasarc.gsfc.nasa.gov/fitsio/fitsio.html" TYPE OPTIONAL PURPOSE "Support for the FITS (Flexible Image Transport System) data format in KStars.")
 
+## cfitsio cURL dependency
+if(CFITSIO_FOUND)
+    find_package(CURL)
+    set(CFITSIO_LIBRARIES ${CFITSIO_LIBRARIE} CURL::libcurl)
+endif(CFITSIO_FOUND)
+
 ## INDI Library
 if (ANDROID)
     set(QT_ANDROID $ENV{QT_ANDROID})
