diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' analitza-18.08.1.orig/CMakeLists.txt analitza-18.08.1/CMakeLists.txt
--- analitza-18.08.1.orig/CMakeLists.txt	2018-09-04 02:08:33.000000000 +0200
+++ analitza-18.08.1/CMakeLists.txt	2018-10-22 18:14:09.797914219 +0200
@@ -1,6 +1,5 @@
-project(Analitza)
-
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.0)
+project(Analitza VERSION "18.0.2")
 
 find_package(ECM 1.7.0 REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
