diff -ruNa readstat-1.1.6.orig/configure.ac readstat-1.1.6/configure.ac
--- readstat-1.1.6.orig/configure.ac	2021-05-04 15:15:52.000000000 +0200
+++ readstat-1.1.6/configure.ac	2021-05-07 00:25:31.613083147 +0200
@@ -49,7 +49,7 @@
 AC_SUBST([EXTRA_LIBS])
 
 AS_CASE([$host],
-	[*mingw*], [EXTRA_WARNINGS="-Wno-pedantic-ms-format -Wno-stringop-truncation"],
+	[*mingw*], [EXTRA_WARNINGS=""],
     [*cygwin*], [EXTRA_WARNINGS="-Wno-unused-const-variable"],
     [EXTRA_WARNINGS=""])
 AC_SUBST([EXTRA_WARNINGS])
diff -ruNa readstat-1.1.6.orig/Makefile.am readstat-1.1.6/Makefile.am
--- readstat-1.1.6.orig/Makefile.am	2021-04-14 04:39:19.000000000 +0200
+++ readstat-1.1.6/Makefile.am	2021-05-07 12:33:02.007263756 +0200
@@ -60,7 +60,7 @@
 	$(AM_V_GEN)$(RAGEL) $(RAGELFLAGS) $< -o $@
 endif
 
-libreadstat_la_CFLAGS = -g -Wall @EXTRA_WARNINGS@ -Werror -pedantic-errors -std=c99
+libreadstat_la_CFLAGS = -g -Wall @EXTRA_WARNINGS@
 libreadstat_la_LIBADD = @EXTRA_LIBS@
 libreadstat_la_LDFLAGS = @EXTRA_LDFLAGS@ -version-info @READSTAT_VERSION_INFO@
 
@@ -168,8 +168,8 @@
 	src/bin/util/readstat_dta_days.c \
 	src/bin/util/readstat_sav_date.c
 
-readstat_LDADD = libreadstat.la
-readstat_CFLAGS = -DREADSTAT_VERSION=\"@READSTAT_VERSION@\" -Wall -Werror -pedantic-errors -std=c99
+readstat_LDADD = libreadstat.la -luser32 -lshell32
+readstat_CFLAGS = -DREADSTAT_VERSION=\"@READSTAT_VERSION@\" -Wall
 if HAVE_ZLIB
 readstat_CFLAGS += -DHAVE_ZLIB=1
 endif
@@ -183,8 +183,8 @@
 	src/bin/util/readstat_dta_days.c \
 	src/bin/util/readstat_sav_date.c
 
-extract_metadata_LDADD = libreadstat.la
-extract_metadata_CFLAGS = -Wall -Werror -pedantic-errors -std=c99
+extract_metadata_LDADD = libreadstat.la -luser32 -lshell32
+extract_metadata_CFLAGS = -Wall
 
 if HAVE_XLSXWRITER
 readstat_SOURCES += src/bin/write/mod_xlsx.c
@@ -256,7 +256,7 @@
 	src/test/test_sav.c
 
 generate_corpus_LDADD = libreadstat.la
-generate_corpus_CFLAGS = -g -Wall @EXTRA_WARNINGS@ -Werror -pedantic-errors -std=c99
+generate_corpus_CFLAGS = -g -Wall @EXTRA_WARNINGS@
 
 EXTRA_PROGRAMS += \
 	fuzz_compression_sas_rle \
