diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user qtwebengine-everywhere-src-5.15.2.orig/src/3rdparty/chromium/base/strings/utf_string_conversions.cc qtwebengine-everywhere-src-5.15.2/src/3rdparty/chromium/base/strings/utf_string_conversions.cc
--- qtwebengine-everywhere-src-5.15.2.orig/src/3rdparty/chromium/base/strings/utf_string_conversions.cc	2020-11-07 01:22:36.000000000 +0000
+++ qtwebengine-everywhere-src-5.15.2/src/3rdparty/chromium/base/strings/utf_string_conversions.cc	2022-02-24 15:50:52.000000000 +0000
@@ -202,7 +210,7 @@
   bool res = DoUTFConversion(src_str.data(), src_len32, dest, &dest_len32);
 
   dest_str->resize(dest_len32);
-  dest_str->shrink_to_fit();
+  dest_str->reserve(0);
 
   return res;
 }
