diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user qtwebengine-everywhere-src-5.15.2.orig/src/3rdparty/chromium/build/toolchain/mac/filter_libtool.py qtwebengine-everywhere-src-5.15.2/src/3rdparty/chromium/build/toolchain/mac/filter_libtool.py
--- qtwebengine-everywhere-src-5.15.2.orig/src/3rdparty/chromium/build/toolchain/mac/filter_libtool.py	2020-11-07 01:22:36.000000000 +0000
+++ qtwebengine-everywhere-src-5.15.2/src/3rdparty/chromium/build/toolchain/mac/filter_libtool.py	2022-02-24 16:25:37.000000000 +0000
@@ -39,7 +39,7 @@
   env['ZERO_AR_DATE'] = '1'
   libtoolout = subprocess.Popen(cmd_list, stderr=subprocess.PIPE, env=env)
   _, err = libtoolout.communicate()
-  for line in err.splitlines():
+  for line in err.decode('UTF-8').splitlines():
     if not IsBlacklistedLine(line):
       print(line, file=sys.stderr)
   # Unconditionally touch the output .a file on the command line if present
