commit 90d67c54ba451a0ab42a91cd6653f6e050652480
Author: Volker Krause <vkrause@kde.org>
Date:   Thu Apr 9 14:35:21 2026 +0200

    Fix build with Qt 6.11 on Android

diff --git a/src/dateandtime/android/build.gradle b/src/dateandtime/android/build.gradle
index 048e35ef..6247847e 100644
--- a/src/dateandtime/android/build.gradle
+++ b/src/dateandtime/android/build.gradle
@@ -6,7 +6,7 @@
 buildscript {
     repositories {
         google()
-        jcenter()
+        mavenCentral()
     }
 
     dependencies {
@@ -16,7 +16,7 @@ buildscript {
 
 repositories {
     google()
-    jcenter()
+    mavenCentral()
 }
 
 apply plugin: 'com.android.library'
diff --git a/templates/kirigamiaddons6/android/build.gradle b/templates/kirigamiaddons6/android/build.gradle
index 1d3d49d1..5ab9413b 100644
--- a/templates/kirigamiaddons6/android/build.gradle
+++ b/templates/kirigamiaddons6/android/build.gradle
@@ -6,17 +6,17 @@
 buildscript {
     repositories {
         google()
-        jcenter()
+        mavenCentral()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.2'
+        classpath 'com.android.tools.build:gradle:8.6.0'
     }
 }
 
 repositories {
     google()
-    jcenter()
+    mavenCentral()
 }
 
 
diff --git a/templates/librarymanager6/android/build.gradle b/templates/librarymanager6/android/build.gradle
index 1d3d49d1..5ab9413b 100644
--- a/templates/librarymanager6/android/build.gradle
+++ b/templates/librarymanager6/android/build.gradle
@@ -6,17 +6,17 @@
 buildscript {
     repositories {
         google()
-        jcenter()
+        mavenCentral()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.2'
+        classpath 'com.android.tools.build:gradle:8.6.0'
     }
 }
 
 repositories {
     google()
-    jcenter()
+    mavenCentral()
 }
 
 
