diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' akonadi-contacts-18.12.1.orig/src/contact-editor/editor/generalinfoeditor/autotests/CMakeLists.txt akonadi-contacts-18.12.1/src/contact-editor/editor/generalinfoeditor/autotests/CMakeLists.txt
--- akonadi-contacts-18.12.1.orig/src/contact-editor/editor/generalinfoeditor/autotests/CMakeLists.txt	2019-01-02 08:26:09.000000000 +0100
+++ akonadi-contacts-18.12.1/src/contact-editor/editor/generalinfoeditor/autotests/CMakeLists.txt	2019-01-24 14:54:56.634618000 +0100
@@ -30,7 +30,7 @@
 target_link_libraries(contactselecttypecomboboxtest Qt5::Test  Qt5::Widgets KF5::ContactEditor KF5::I18n )
 
 
-set(nicknamewidgettest_SRCS nicknamewidgettest.cpp)
+set(nicknamewidgettest_SRCS nicknamewidgettest.cpp ../nicknamewidget.cpp)
 add_executable(nicknamewidgettest ${nicknamewidgettest_SRCS})
 add_test(NAME nicknamewidgettest COMMAND nicknamewidgettest)
 ecm_mark_as_test(nicknamewidgettest)
