diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' xmlsec1-1.2.25.orig/docs/Makefile.am xmlsec1-1.2.25/docs/Makefile.am
--- xmlsec1-1.2.25.orig/docs/Makefile.am	2017-09-12 15:21:09.000000000 +0200
+++ xmlsec1-1.2.25/docs/Makefile.am	2018-05-03 20:28:05.000000000 +0200
@@ -24,7 +24,7 @@
 		echo "Copying docs..."; \
 		if [ z"$(srcdir)" != z"$(builddir)" ]; \
 		then \
-		    $(CP) -ru $(srcdir)/*.html $(srcdir)/*.ico $(srcdir)/images $(builddir)/  ; \
+		    $(CP) -rf $(srcdir)/*.html $(srcdir)/*.ico $(srcdir)/images $(builddir)/  ; \
 		    chmod u+w $(builddir)/*.html ; \
 		    chmod u+w $(builddir)/*.ico ; \
 		    chmod -R u+w $(builddir)/images ; \
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' xmlsec1-1.2.25.orig/docs/Makefile.in xmlsec1-1.2.25/docs/Makefile.in
--- xmlsec1-1.2.25.orig/docs/Makefile.in	2017-09-12 15:21:53.000000000 +0200
+++ xmlsec1-1.2.25/docs/Makefile.in	2018-05-03 20:27:53.000000000 +0200
@@ -744,7 +744,7 @@
 		echo "Copying docs..."; \
 		if [ z"$(srcdir)" != z"$(builddir)" ]; \
 		then \
-		    $(CP) -ru $(srcdir)/*.html $(srcdir)/*.ico $(srcdir)/images $(builddir)/  ; \
+		    $(CP) -rf $(srcdir)/*.html $(srcdir)/*.ico $(srcdir)/images $(builddir)/  ; \
 		    chmod u+w $(builddir)/*.html ; \
 		    chmod u+w $(builddir)/*.ico ; \
 		    chmod -R u+w $(builddir)/images ; \
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' xmlsec1-1.2.25.orig/docs/api/Makefile.am xmlsec1-1.2.25/docs/api/Makefile.am
--- xmlsec1-1.2.25.orig/docs/api/Makefile.am	2017-09-12 15:21:09.000000000 +0200
+++ xmlsec1-1.2.25/docs/api/Makefile.am	2018-05-03 20:34:55.000000000 +0200
@@ -77,7 +77,7 @@
 		echo "Copying api-docs..."; \
 		if [ z"$(srcdir)" != z"$(builddir)" ]; \
 		then \
-		    $(CP) -ru $(SOURCE_FILES_TO_COPY) $(builddir)/ ; \
+		    $(CP) -rf $(SOURCE_FILES_TO_COPY) $(builddir)/ ; \
 		fi \
 	)
 endif
@@ -92,10 +92,10 @@
 	$(GTKDOC_MKHTML) xmlsec $(MAIN_SGML_FILE)
 
 $(SGML_CHAPTERS_DIR): $(SGML_CHAPTERS_SRC_DIR) $(SGML_CHAPTERS_DIR)/.sentinel
-	$(CP) -ru $(SGML_CHAPTERS_SRC_DIR)/* $(SGML_CHAPTERS_DIR)
+	$(CP) -rf $(SGML_CHAPTERS_SRC_DIR)/* $(SGML_CHAPTERS_DIR)
 
 $(MAIN_SGML_FILE): $(MAIN_SGML_SRC_FILE)
-	$(CP) -u $(MAIN_SGML_SRC_FILE) $(MAIN_SGML_FILE)
+	$(CP) -f $(MAIN_SGML_SRC_FILE) $(MAIN_SGML_FILE)
 #
 # Prepeare sgml files from sources for each library. We are also
 # doing some "magic" here by automatically adding links to XML DSig and 
@@ -189,7 +189,7 @@
 templates-copy: $(TMPL_SRC_DIR) $(TMPL_DIR)/.sentinel
 	@echo "Copying original template files into '$(TMPL_DIR)' ..."
 	( if [ z"$(TMPL_DIR)" != z"$(TMPL_SRC_DIR)" ] ; then \
-		$(CP) -ru $(TMPL_SRC_DIR)/* $(TMPL_DIR)/ ; \
+		$(CP) -rf $(TMPL_SRC_DIR)/* $(TMPL_DIR)/ ; \
 	  fi ; )
 	chmod -R u+w $(TMPL_DIR)
 
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' xmlsec1-1.2.25.orig/docs/api/Makefile.in xmlsec1-1.2.25/docs/api/Makefile.in
--- xmlsec1-1.2.25.orig/docs/api/Makefile.in	2017-09-12 15:21:53.000000000 +0200
+++ xmlsec1-1.2.25/docs/api/Makefile.in	2018-05-03 20:35:21.000000000 +0200
@@ -614,7 +614,7 @@
 @BUILD_API_DOCS_FALSE@		echo "Copying api-docs..."; \
 @BUILD_API_DOCS_FALSE@		if [ z"$(srcdir)" != z"$(builddir)" ]; \
 @BUILD_API_DOCS_FALSE@		then \
-@BUILD_API_DOCS_FALSE@		    $(CP) -ru $(SOURCE_FILES_TO_COPY) $(builddir)/ ; \
+@BUILD_API_DOCS_FALSE@		    $(CP) -rf $(SOURCE_FILES_TO_COPY) $(builddir)/ ; \
 @BUILD_API_DOCS_FALSE@		fi \
 @BUILD_API_DOCS_FALSE@	)
 
@@ -628,10 +628,10 @@
 	$(GTKDOC_MKHTML) xmlsec $(MAIN_SGML_FILE)
 
 $(SGML_CHAPTERS_DIR): $(SGML_CHAPTERS_SRC_DIR) $(SGML_CHAPTERS_DIR)/.sentinel
-	$(CP) -ru $(SGML_CHAPTERS_SRC_DIR)/* $(SGML_CHAPTERS_DIR)
+	$(CP) -rf $(SGML_CHAPTERS_SRC_DIR)/* $(SGML_CHAPTERS_DIR)
 
 $(MAIN_SGML_FILE): $(MAIN_SGML_SRC_FILE)
-	$(CP) -u $(MAIN_SGML_SRC_FILE) $(MAIN_SGML_FILE)
+	$(CP) -f $(MAIN_SGML_SRC_FILE) $(MAIN_SGML_FILE)
 #
 # Prepeare sgml files from sources for each library. We are also
 # doing some "magic" here by automatically adding links to XML DSig and 
@@ -725,7 +725,7 @@
 templates-copy: $(TMPL_SRC_DIR) $(TMPL_DIR)/.sentinel
 	@echo "Copying original template files into '$(TMPL_DIR)' ..."
 	( if [ z"$(TMPL_DIR)" != z"$(TMPL_SRC_DIR)" ] ; then \
-		$(CP) -ru $(TMPL_SRC_DIR)/* $(TMPL_DIR)/ ; \
+		$(CP) -rf $(TMPL_SRC_DIR)/* $(TMPL_DIR)/ ; \
 	  fi ; )
 	chmod -R u+w $(TMPL_DIR)
 
