diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' boost_1_69_0.orig/libs/locale/build/Jamfile.v2 boost_1_69_0/libs/locale/build/Jamfile.v2
--- boost_1_69_0.orig/libs/locale/build/Jamfile.v2	2018-12-05 20:58:19.000000000 +0100
+++ boost_1_69_0/libs/locale/build/Jamfile.v2	2019-06-21 14:17:47.328806300 +0200
@@ -53,8 +53,8 @@
                             ;
 
 # Xlocale
-obj has_xlocale_obj : ../build/has_xlocale.cpp ;
-exe has_xlocale : has_xlocale_obj ;
+#obj has_xlocale_obj : ../build/has_xlocale.cpp ;
+#exe has_xlocale : has_xlocale_obj ;
 explicit has_xlocale ;
 
 #end xlocale
C:\CraftRoot\dev-utils\bin\diff.exe: boost_1_69_0.orig/libs/wave/test/testwave/testfiles: Invalid argument
C:\CraftRoot\dev-utils\bin\diff.exe: boost_1_69_0/libs/wave/test/testwave/testfiles: Invalid argument
