diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' KDStateMachineEditor-1.2.3.orig/src/core/model/element.h KDStateMachineEditor-1.2.3/src/core/model/element.h
--- KDStateMachineEditor-1.2.3.orig/src/core/model/element.h	2018-02-28 18:13:17.000000000 +0100
+++ KDStateMachineEditor-1.2.3/src/core/model/element.h	2018-06-18 15:36:15.862585200 +0200
@@ -30,6 +30,7 @@
 
 #include <QObject>
 #include <QMetaType>
+#include <QPointF>
 
 QT_BEGIN_NAMESPACE
 class QPainterPath;
