diff --git a/src/quick/CMakeLists.txt b/src/quick/CMakeLists.txt
index b27e996116ec82c8fcce9ff29ec41c0533f51fd2..83bad940378fdb166b5f935cfa52c369efb3e30b 100644
--- a/src/quick/CMakeLists.txt
+++ b/src/quick/CMakeLists.txt
@@ -20,6 +20,10 @@ target_link_libraries(sonnetquickplugin
 set(bindest "${CMAKE_BINARY_DIR}/bin/qml/org/kde/sonnet/")
 set_target_properties(sonnetquickplugin PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${bindest}")
 
+if(MINGW)
+    set_target_properties(sonnetquickplugin PROPERTIES PREFIX "")
+endif()
+
 install(TARGETS sonnetquickplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/sonnet)
 install(FILES qmldir DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/sonnet)
 
