diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' exiv2-0.27.0-Source.orig/src/CMakeLists.txt exiv2-0.27.0-Source/src/CMakeLists.txt
--- exiv2-0.27.0-Source.orig/src/CMakeLists.txt	2018-12-20 14:45:16.000000000 +0100
+++ exiv2-0.27.0-Source/src/CMakeLists.txt	2019-02-01 11:15:55.251520500 +0100
@@ -179,7 +179,7 @@
             target_compile_definitions(exiv2lib PUBLIC ${CONAN_COMPILE_DEFINITIONS_LIBCURL})
             target_link_libraries(exiv2lib PUBLIC ${CONAN_EXE_LINKER_FLAGS_LIBCURL})
         else()
-            target_link_libraries(exiv2lib PRIVATE ${CURL_LIBRARIES})
+            target_link_libraries(exiv2lib PUBLIC ${CURL_LIBRARIES})
         endif()
     endif()
 
@@ -189,18 +189,14 @@
     target_compile_definitions(exiv2lib PRIVATE PSAPI_VERSION=1) # to be compatible with <= WinVista (#905)
 endif()
 
-if (NOT MSVC)
+if (NOT WIN32)
     if ( UNIX AND NOT CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" )
         target_link_libraries( exiv2lib PRIVATE ${CMAKE_DL_LIBS}) # -ldl = dynamic loader used by src/version.cpp
     endif()
 
-    if (CYGWIN OR MINGW OR MSYS)
-        target_link_libraries( exiv2lib PRIVATE psapi ws2_32 )
-    endif()
-
     target_link_libraries( exiv2lib PRIVATE Threads::Threads)
 else()
-    target_link_libraries( exiv2lib PRIVATE psapi ws2_32 )
+    target_link_libraries( exiv2lib PUBLIC psapi ws2_32 Threads::Threads)
 endif()
 
 if( EXIV2_ENABLE_PNG )
@@ -215,7 +211,7 @@
 endif()
 
 if( ICONV_FOUND )
-    target_link_libraries( exiv2lib PRIVATE Iconv::Iconv )
+    target_link_libraries( exiv2lib PUBLIC Iconv::Iconv )
 endif()
 
 
