diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' exiv2-0.27.0-Source.orig/src/futils.cpp exiv2-0.27.0-Source/src/futils.cpp
--- exiv2-0.27.0-Source.orig/src/futils.cpp	2018-12-20 14:45:16.000000000 +0100
+++ exiv2-0.27.0-Source/src/futils.cpp	2019-01-05 14:56:56.303674000 +0100
@@ -34,7 +34,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>

-#ifdef _MSC_VER
+#ifdef _WIN32
     #include <Windows.h>
     # define S_ISREG(m)      (((m) & S_IFMT) == S_IFREG)
     #include <psapi.h>  // For access to GetModuleFileNameEx
