diff -ruNa netcdf-c-4.8.0.orig/include/nclog.h netcdf-c-4.8.0/include/nclog.h
--- netcdf-c-4.8.0.orig/include/nclog.h	2021-03-30 22:23:10.000000000 +0200
+++ netcdf-c-4.8.0/include/nclog.h	2021-05-05 21:12:43.015642625 +0200
@@ -8,6 +8,7 @@
 #define NCLOG_H
 
 #include <stdarg.h>
+#include <stdio.h>
 #include "ncexternl.h"
 
 #define NCENVLOGGING "NCLOGGING"
diff -ruNa netcdf-c-4.8.0.orig/libdispatch/dcrc64.c netcdf-c-4.8.0/libdispatch/dcrc64.c
--- netcdf-c-4.8.0.orig/libdispatch/dcrc64.c	2021-03-30 22:23:10.000000000 +0200
+++ netcdf-c-4.8.0/libdispatch/dcrc64.c	2021-05-06 10:39:26.559599521 +0200
@@ -44,6 +44,11 @@
 
 #include "ncexternl.h"
 
+#ifndef uint64
+#include <stdint.h>
+typedef uint64_t uint64;
+#endif
+
 /* The include of pthread.h below can be commented out in order to not use the
    pthread library for table initialization.  In that case, the initialization
    will not be thread-safe.  That's fine, so long as it can be assured that
diff -ruNa netcdf-c-4.8.0.orig/libsrc/ncx.h netcdf-c-4.8.0/libsrc/ncx.h
--- netcdf-c-4.8.0.orig/libsrc/ncx.h	2021-03-30 22:23:10.000000000 +0200
+++ netcdf-c-4.8.0/libsrc/ncx.h	2021-05-18 20:08:09.988593307 +0200
@@ -78,6 +78,76 @@
 #define X_SIZEOF_INT64		8
 #define X_SIZEOF_UINT64		8
 
+#ifndef SIZEOF_UCHAR
+#ifdef  SIZEOF_UNSIGNED_CHAR
+#define SIZEOF_UCHAR SIZEOF_UNSIGNED_CHAR
+#else
+#define SIZEOF_UCHAR 1
+#endif
+#endif
+
+#ifndef SIZEOF_USHORT
+#ifdef  SIZEOF_UNSIGNED_SHORT_INT
+#define SIZEOF_USHORT SIZEOF_UNSIGNED_SHORT_INT
+#elif defined(SIZEOF_UNSIGNED_SHORT)
+#define SIZEOF_USHORT SIZEOF_UNSIGNED_SHORT
+#else
+#define SIZEOF_USHORT 2
+#endif
+#endif
+
+#ifndef SIZEOF_UINT
+#ifdef  SIZEOF_UNSIGNED_INT
+#define SIZEOF_UINT SIZEOF_UNSIGNED_INT
+#else
+#define SIZEOF_UINT 4
+#endif
+#endif
+
+#ifndef SIZEOF_LONGLONG
+#ifdef  SIZEOF_LONG_LONG
+#define SIZEOF_LONGLONG SIZEOF_LONG_LONG
+#else
+#define SIZEOF_LONGLONG 8
+#endif
+#endif
+
+#ifndef SIZEOF_INT64
+#ifdef  SIZEOF_LONG_LONG
+#define SIZEOF_INT64 SIZEOF_LONG_LONG
+#elif defined(SIZEOF_LONGLONG)
+#define SIZEOF_INT64 SIZEOF_LONGLONG
+#else
+#define SIZEOF_INT64 8
+#endif
+#endif
+
+#ifndef SIZEOF_ULONGLONG
+#ifdef  SIZEOF_UNSIGNED_LONG_LONG
+#define SIZEOF_ULONGLONG SIZEOF_UNSIGNED_LONG_LONG
+#else
+#define SIZEOF_ULONGLONG SIZEOF_INT64
+#endif
+#endif
+
+#ifndef SIZEOF_UINT64
+#ifdef  SIZEOF_UNSIGNED_LONG_LONG
+#define SIZEOF_UINT64 SIZEOF_UNSIGNED_LONG_LONG
+#elif defined(SIZEOF_ULONGLONG)
+#define SIZEOF_UINT64 SIZEOF_ULONGLONG
+#else
+#define SIZEOF_UINT64 SIZEOF_INT64
+#endif
+#endif
+
+#ifndef SIZEOF_FLOAT
+#define SIZEOF_FLOAT 4
+#endif
+
+#ifndef SIZEOF_DOUBLE
+#define SIZEOF_DOUBLE 8
+#endif
+
 /*
  * For now, netcdf is limited to 32 bit sizes,
  * If compiled with support for "large files", then
@@ -171,6 +241,22 @@
 typedef signed char schar;
 #endif
 
+#ifndef uchar
+typedef unsigned char uchar;
+#endif
+#ifndef ushort
+typedef unsigned short ushort;
+#endif
+#ifndef uint
+typedef unsigned int uint;
+#endif
+#ifndef longlong
+typedef long long longlong;
+#endif
+#ifndef ulonglong
+typedef unsigned long long ulonglong;
+#endif
+
 /*
  * Primitive numeric conversion functions.
  * The `put' functions convert from native internal
diff -ruNa netcdf-c-4.8.0.orig/libsrc/ncx.m4 netcdf-c-4.8.0/libsrc/ncx.m4
--- netcdf-c-4.8.0.orig/libsrc/ncx.m4	2021-03-30 22:23:10.000000000 +0200
+++ netcdf-c-4.8.0/libsrc/ncx.m4	2021-05-17 20:11:11.991062366 +0200
@@ -179,7 +179,7 @@
 #ifdef  SIZEOF_UNSIGNED_CHAR
 #define SIZEOF_UCHAR SIZEOF_UNSIGNED_CHAR
 #else
-#error "unknown SIZEOF_UCHAR"
+#error:  "unknown SIZEOF_UCHAR"
 #endif
 #endif
 
diff -ruNa netcdf-c-4.8.0.orig/libsrc/putget.c netcdf-c-4.8.0/libsrc/putget.c
--- netcdf-c-4.8.0.orig/libsrc/putget.c	2021-03-30 22:23:10.000000000 +0200
+++ netcdf-c-4.8.0/libsrc/putget.c	2021-05-13 13:44:25.728611481 +0200
@@ -193,6 +193,9 @@
 }
 #line 112
 
+#ifndef SIZEOF_INT
+#define SIZEOF_INT 4
+#endif
 
 #if (SIZEOF_INT >= X_SIZEOF_INT)
 static int
diff -ruNa netcdf-c-4.8.0.orig/libsrc/putget.m4 netcdf-c-4.8.0/libsrc/putget.m4
--- netcdf-c-4.8.0.orig/libsrc/putget.m4	2021-03-30 22:23:10.000000000 +0200
+++ netcdf-c-4.8.0/libsrc/putget.m4	2021-05-13 13:46:13.008089615 +0200
@@ -111,6 +111,10 @@
 NCFILL(char, char, X_SIZEOF_CHAR, NC_FILL_CHAR)
 NCFILL(short, short, X_SIZEOF_SHORT, NC_FILL_SHORT)
 
+#ifndef SIZEOF_INT
+#define SIZEOF_INT 4
+#endif
+
 #if (SIZEOF_INT >= X_SIZEOF_INT)
 NCFILL(int, int, X_SIZEOF_INT, NC_FILL_INT)
 #elif SIZEOF_LONG == X_SIZEOF_INT
