diff -ruNa netcdf-c-4.7.4.orig/libdispatch/dstring.c netcdf-c-4.7.4/libdispatch/dstring.c
--- netcdf-c-4.7.4.orig/libdispatch/dstring.c	2020-03-27 16:33:36.000000000 +0100
+++ netcdf-c-4.7.4/libdispatch/dstring.c	2020-10-09 10:59:41.682301090 +0200
@@ -14,6 +14,10 @@
 #include "rnd.h"
 #include "ncutf8.h"
 
+#ifndef uchar
+typedef unsigned char uchar;
+#endif
+
 /* There are 3 levels of UTF8 checking: 1=> (exact)validating 2=>relaxed
    and 3=>very relaxed
 */
