diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user openldap-LMDB_0.9.33/libraries/liblmdb.orig/CMakeLists.txt openldap-LMDB_0.9.33/libraries/liblmdb/CMakeLists.txt
--- openldap-LMDB_0.9.33/libraries/liblmdb.orig/CMakeLists.txt	2025-11-26 16:56:41.151600331 +0100
+++ openldap-LMDB_0.9.33/libraries/liblmdb/CMakeLists.txt	2025-11-26 16:56:30.544993112 +0100
@@ -48,7 +48,9 @@
 if(BUILD_SHARED)
     add_library(lmdb SHARED ${mdb_SRCS})
     target_link_libraries(lmdb PUBLIC Threads::Threads)
+    if (MSVC)
     set_target_properties(lmdb PROPERTIES PREFIX "")
+	endif()
     set(mdb_TARGETS ${mdb_TARGETS} lmdb)
 endif()
 
