diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' lmdb-LMDB_0.9.16\libraries\liblmdb.orig/CMakeLists.txt lmdb-LMDB_0.9.16\libraries\liblmdb/CMakeLists.txt
--- lmdb-LMDB_0.9.16\libraries\liblmdb.orig/CMakeLists.txt	2018-10-09 08:59:00.678964500 +0200
+++ lmdb-LMDB_0.9.16\libraries\liblmdb/CMakeLists.txt	2018-10-09 08:58:35.321696600 +0200
@@ -44,8 +44,9 @@
     set(mdb_TARGETS ${mdb_TARGETS} lmdb-static)
 endif()
 if(BUILD_SHARED)
-    add_library(lmdb SHARED ${mdb_SRCS})
-    set(mdb_TARGETS ${mdb_TARGETS} lmdb)
+    add_library(liblmdb SHARED ${mdb_SRCS})
+    set_target_properties(liblmdb PROPERTIES PREFIX "")
+    set(mdb_TARGETS ${mdb_TARGETS} liblmdb)
 endif()
 
 if(BUILD_TOOLS)
