--- CMakeLists.txt.orig	2020-11-15 23:57:43.988548700 -0800
+++ CMakeLists.txt	2020-11-16 00:04:57.625709000 -0800
@@ -80,3 +80,13 @@
 build_plugin(bigsh0t_rect_to_eq RectToEq.cpp)
 build_plugin(bigsh0t_eq_to_rect EqToRect.cpp)
 build_plugin(bigsh0t_stabilize_360 Stabilize360.cpp)
+
+include(GNUInstallDirs)
+install (TARGETS
+    bigsh0t_hemi_to_eq
+    bigsh0t_eq_mask
+    bigsh0t_transform_360
+    bigsh0t_rect_to_eq
+    bigsh0t_eq_to_rect
+    bigsh0t_stabilize_360
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/frei0r-1)
