diff --git a/test/utils.c b/test/utils.c
index cd9ab26d5aabdec9c272137d1081b07f7f21a367..1c64cee2cc3c2b60448448ffd9f0c9c83eebc00e 100644
--- a/test/utils.c
+++ b/test/utils.c
@@ -35,8 +35,8 @@
 /* Random number generator state
  */
 
-prng_t prng_state_data;
-prng_t *prng_state;
+prng_t prng_state_data = {0};
+prng_t *prng_state = NULL;
 
 /*----------------------------------------------------------------------------*\
  *  CRC-32 version 2.0.0 by Craig Bruce, 2006-04-29.
