diff --git a/src/spectral/fft.c b/src/spectral/fft.c
index 4cbf4ab..83be819 100644
--- a/src/spectral/fft.c
+++ b/src/spectral/fft.c
@@ -213,7 +213,7 @@ new_aubio_fft (uint_t winsize) {
   s->pfw = fftw_plan_r2r_1d(winsize, s->in,  s->specdata, FFTW_R2HC, FFTW_ESTIMATE);
   s->pbw = fftw_plan_r2r_1d(winsize, s->specdata, s->out, FFTW_HC2R, FFTW_ESTIMATE);
 #endif
-  pthread_mutex_unlock(&aubio_fftw_mutex);
+  AUBIO_FFTW_UNLOCK(&aubio_fftw_mutex);
   for (i = 0; i < s->winsize; i++) {
     s->in[i] = 0.;
     s->out[i] = 0.;
@@ -314,10 +314,10 @@ void del_aubio_fft(aubio_fft_t * s) {
 void del_aubio_fft(aubio_fft_t * s) {
   /* destroy data */
 #ifdef HAVE_FFTW3             // using FFTW3
-  pthread_mutex_lock(&aubio_fftw_mutex);
+  AUBIO_FFTW_LOCK(&aubio_fftw_mutex);
   fftw_destroy_plan(s->pfw);
   fftw_destroy_plan(s->pbw);
   fftw_free(s->specdata);
-  pthread_mutex_unlock(&aubio_fftw_mutex);
+  AUBIO_FFTW_UNLOCK(&aubio_fftw_mutex);
 
 #elif defined HAVE_ACCELERATE // using ACCELERATE
