commit 745932d29035e926f7f438c6217ed928b1d0346c
Author: Christoph Cullmann <cullmann@kde.org>
Date:   Sat Sep 28 18:22:00 2019 +0200

    improve hi-dpi support
    tested, seems to work properly

diff --git a/src/main.cpp b/src/main.cpp
index 15c2a43..4351c01 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -33,6 +33,12 @@
 
 int main(int argc, char *argv[])
 {
+    /**
+     * enable high dpi support
+     */
+    QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps, true);
+    QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling, true);
+
     QApplication app(argc, argv);
 
     KLocalizedString::setApplicationDomain("filelight");
