diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user R-4.2.0.orig/src/scripts/Makefile.in R-4.2.0/src/scripts/Makefile.in
--- R-4.2.0.orig/src/scripts/Makefile.in	2019-03-29 00:02:38.000000000 +0100
+++ R-4.2.0/src/scripts/Makefile.in	2022-07-11 19:59:34.595581402 +0200
@@ -75,9 +75,7 @@
 	  dat=`sed -e 1d -e 's/^Last Changed Date: //' $(top_builddir)/SVN-REVISION`; \
 	  v="$${v} ($${dat})"; \
 	  y=`$(ECHO) $${dat} | cut -d- -f1`; \
-	  d=`$(ECHO) "$(abs_top_builddir)" | sed 's,/,\\\/,g'`; \
-	  sed "1,/R_HOME_DIR/s/\\(R_HOME_DIR=\\).*/\\1\"$${d}\"/; \
-	       s@REPLACE_BY_VERSION@$${v}@; \
+	  sed "s@REPLACE_BY_VERSION@$${v}@; \
 	       s@REPLACE_BY_YEAR@$${y}@" < R.sh > $@)
 	@chmod +x $@
 ## </NOTE>
@@ -88,10 +86,10 @@
 install: installdirs install-cmds
 	@rm -f $(DESTDIR)$(bindir)/R
 	@(d=`$(ECHO) '$(rhome)' | sed 's,/,\\\/,g';`; \
-	  d2=`$(ECHO) '$(rsharedir)' | sed 's,/,\\\/,g';`; \
-	  d3=`$(ECHO) '$(rincludedir)' | sed 's,/,\\\/,g';`; \
-	  d4=`$(ECHO) '$(rdocdir)' | sed 's,/,\\\/,g';`; \
-	  sed -e "1,/R_HOME_DIR=/s/\\(R_HOME_DIR=\\).*/\\1$${d}/;" -e "s/\\(R_SHARE_DIR=\\).*/\\1$${d2}/;" -e  "s/\\(R_INCLUDE_DIR=\\).*/\\1$${d3}/;" -e "s/\\(R_DOC_DIR=\\).*/\\1$${d4}/;"\
+	  d2="$${R_HOME_DIR}/share"; \
+	  d3="$${R_HOME_DIR}/include"; \
+	  d4="$${R_HOME_DIR}/doc"; \
+	  sed -e "1,/R_HOME_DIR_DISABLED=/s/\\(R_HOME_DIR=\\).*/\\1$${d}/;" -e "s/\\(R_SHARE_DIR=\\).*/\\1$${d2}/;" -e  "s/\\(R_INCLUDE_DIR=\\).*/\\1$${d3}/;" -e "s/\\(R_DOC_DIR=\\).*/\\1$${d4}/;"\
 	  < R.fe > "$(DESTDIR)$(Rexecbindir)/R")
 	@$(INSTALL_SCRIPT)  "$(DESTDIR)$(Rexecbindir)/R" "$(DESTDIR)$(bindir)/R"
 	@chmod 755 "$(DESTDIR)$(bindir)/R" "$(DESTDIR)$(Rexecbindir)/R"
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user R-4.2.0.orig/src/scripts/R.sh.in R-4.2.0/src/scripts/R.sh.in
--- R-4.2.0.orig/src/scripts/R.sh.in	2021-04-21 00:15:00.000000000 +0200
+++ R-4.2.0/src/scripts/R.sh.in	2022-07-11 21:45:21.810077333 +0200
@@ -1,8 +1,18 @@
 #!@R_SHELL@
 # Shell wrapper for R executable.
+INST_ROOT="`dirname $0`/.."
+if [ -x "`dirname $0`/../R" ]; then
+   INST_ROOT="${INST_ROOT}/.."
+fi
+INST_ROOT=`realpath ${INST_ROOT}`
 
-R_HOME_DIR=
-if test "${R_HOME_DIR}" = "@prefix@/@LIBnn@/R"; then
+if [ -f "${INST_ROOT}/etc/ldpaths" ]; then # inside build tree
+   R_HOME_DIR=${INST_ROOT}
+else
+   R_HOME_DIR=${INST_ROOT}/@LIBnn@/R
+fi
+
+if test "${R_HOME_DIR}" = "${INST_ROOT}/@LIBnn@/R"; then
    case "@R_OS@" in
    linux*)
      run_arch=`uname -m`
@@ -16,10 +26,10 @@
           libnn_fallback=lib64
         ;;
      esac
-     if [ -x "@prefix@/${libnn}/R/bin/exec/R" ]; then
-        R_HOME_DIR="@prefix@/${libnn}/R"
-     elif [ -x "@prefix@/${libnn_fallback}/R/bin/exec/R" ]; then
-        R_HOME_DIR="@prefix@/${libnn_fallback}/R"
+     if [ -x "${INST_ROOT}/${libnn}/R/bin/exec/R" ]; then
+        R_HOME_DIR="${INST_ROOT}/${libnn}/R"
+     elif [ -x "${INST_ROOT}/${libnn_fallback}/R/bin/exec/R" ]; then
+        R_HOME_DIR="${INST_ROOT}/${libnn_fallback}/R"
      ## else -- leave alone (might be a sub-arch)
      fi
      ;;
