diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user qtbase-everywhere-src-6.6.0.orig/cmake/FindWrapSystemPCRE2.cmake qtbase-everywhere-src-6.6.0/cmake/FindWrapSystemPCRE2.cmake
--- qtbase-everywhere-src-6.6.0.orig/cmake/FindWrapSystemPCRE2.cmake	2023-09-13 18:02:48.000000000 +0200
+++ qtbase-everywhere-src-6.6.0/cmake/FindWrapSystemPCRE2.cmake	2023-10-28 09:23:21.334688200 +0200
@@ -9,15 +9,6 @@
 
 find_package(PCRE2 ${${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION} COMPONENTS 16BIT QUIET)
 
-set(__pcre2_target_name "PCRE2::16BIT")
-if(PCRE2_FOUND AND TARGET "${__pcre2_target_name}")
-  # Hunter case.
-  set(__pcre2_found TRUE)
-  if(PCRE2_VERSION)
-      set(WrapSystemPCRE2_VERSION "${PCRE2_VERSION}")
-  endif()
-endif()
-
 if(NOT __pcre2_found)
   list(PREPEND WrapSystemPCRE2_REQUIRED_VARS PCRE2_LIBRARIES PCRE2_INCLUDE_DIRS)
 
@@ -51,12 +42,7 @@
                                   VERSION_VAR WrapSystemPCRE2_VERSION)
 if(WrapSystemPCRE2_FOUND)
     add_library(WrapSystemPCRE2::WrapSystemPCRE2 INTERFACE IMPORTED)
-    if(TARGET "${__pcre2_target_name}")
-        target_link_libraries(WrapSystemPCRE2::WrapSystemPCRE2 INTERFACE "${__pcre2_target_name}")
-    else()
         target_link_libraries(WrapSystemPCRE2::WrapSystemPCRE2 INTERFACE ${PCRE2_LIBRARIES})
         target_include_directories(WrapSystemPCRE2::WrapSystemPCRE2 INTERFACE ${PCRE2_INCLUDE_DIRS})
-    endif()
 endif()
-unset(__pcre2_target_name)
 unset(__pcre2_found)
