diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' libpng-1.6.37.orig/CMakeLists.txt libpng-1.6.37/CMakeLists.txt
--- libpng-1.6.37.orig/CMakeLists.txt	2019-04-14 20:10:32.000000000 +0200
+++ libpng-1.6.37/CMakeLists.txt	2019-09-11 10:17:12.153988400 +0200
@@ -31,6 +31,7 @@
 set(PNGLIB_VERSION ${PNGLIB_MAJOR}.${PNGLIB_MINOR}.${PNGLIB_RELEASE})
 
 include(GNUInstallDirs)
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 # needed packages
 
