diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libpng-1.6.40.orig/CMakeLists.txt libpng-1.6.40/CMakeLists.txt
--- libpng-1.6.40.orig/CMakeLists.txt	2023-06-21 21:06:40.000000000 +0200
+++ libpng-1.6.40/CMakeLists.txt	2023-09-05 15:39:45.341199900 +0200
@@ -635,6 +635,10 @@
   set(PNG_STATIC_OUTPUT_NAME "libpng${PNGLIB_ABI_VERSION}_static")
 endif()
 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(PNGLIB_DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+endif()
+
 if(PNG_SHARED)
   add_library(png_shared SHARED ${libpng_sources})
   add_dependencies(png_shared png_genfiles)
@@ -960,12 +964,16 @@
 # We use the same files like ./configure, so we have to set its vars.
 # Only do this on Windows for Cygwin - the files don't make much sense
 # outside of a UNIX look-alike.
-if(NOT WIN32 OR CYGWIN OR MINGW)
+#if(NOT WIN32 OR CYGWIN OR MINGW)
+if(MSVC)
+  set(LIBPNG_IMPORT_PREFIX "lib")
+endif()
+
   set(prefix      ${CMAKE_INSTALL_PREFIX})
   set(exec_prefix ${CMAKE_INSTALL_PREFIX})
   set(libdir      ${CMAKE_INSTALL_FULL_LIBDIR})
   set(includedir  ${CMAKE_INSTALL_FULL_INCLUDEDIR})
-  set(LIBS        "-lz -lm")
+  #set(LIBS        "-lz -lm")
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libpng.pc.in
                  ${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}.pc
                  @ONLY)
@@ -974,7 +982,7 @@
                  ${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}-config
                  @ONLY)
   create_symlink(libpng-config FILE libpng${PNGLIB_ABI_VERSION}-config)
-endif()
+#endif()
 
 # Install.
 if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
@@ -1036,7 +1044,7 @@
   install(FILES png.5
           DESTINATION ${CMAKE_INSTALL_MANDIR}/man5)
   # Install the pkg-config files.
-  if(NOT CMAKE_HOST_WIN32 OR CYGWIN OR MINGW)
+#  if(NOT CMAKE_HOST_WIN32 OR CYGWIN OR MINGW)
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libpng.pc
             DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
     install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/libpng-config
@@ -1046,7 +1054,7 @@
     install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}-config
             DESTINATION ${CMAKE_INSTALL_BINDIR})
   endif()
-endif()
+#endif()
 
 # Create an export file that CMake users can include() to import our targets.
 if(NOT SKIP_INSTALL_EXPORT AND NOT SKIP_INSTALL_ALL)
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libpng-1.6.40.orig/libpng.pc.in libpng-1.6.40/libpng.pc.in
--- libpng-1.6.40.orig/libpng.pc.in	2023-06-21 21:06:40.000000000 +0200
+++ libpng-1.6.40/libpng.pc.in	2023-09-05 15:40:05.494646400 +0200
@@ -7,6 +7,6 @@
 Description: Loads and saves PNG files
 Version: @PNGLIB_VERSION@
 Requires.private: zlib
-Libs: -L${libdir} -lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@
+Libs: -L${libdir} -l@LIBPNG_IMPORT_PREFIX@png@PNGLIB_MAJOR@@PNGLIB_MINOR@@PNGLIB_DEBUG_POSTFIX@
 Libs.private: @LIBS@
 Cflags: -I${includedir}
