diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libpng-1.6.37.orig/CMakeLists.txt libpng-1.6.37/CMakeLists.txt
--- libpng-1.6.37.orig/CMakeLists.txt	2020-11-16 10:42:41.334083400 +0100
+++ libpng-1.6.37/CMakeLists.txt	2020-11-16 10:41:36.699579200 +0100
@@ -527,6 +527,10 @@
 
 unset(PNG_LIB_TARGETS)
 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(PNGLIB_DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+endif()
+
 if(PNG_SHARED)
   add_library(png SHARED ${libpng_sources})
   set(PNG_LIB_TARGETS png)
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libpng-1.6.37.orig/libpng.pc.in libpng-1.6.37/libpng.pc.in
--- libpng-1.6.37.orig/libpng.pc.in	2020-11-16 10:42:41.335083700 +0100
+++ libpng-1.6.37/libpng.pc.in	2020-11-16 10:38:43.900680900 +0100
@@ -7,6 +7,6 @@
 Description: Loads and saves PNG files
 Version: @PNGLIB_VERSION@
 Requires: zlib
-Libs: -L${libdir} -l@LIBPNG_IMPORT_PREFIX@png@PNGLIB_MAJOR@@PNGLIB_MINOR@@CMAKE_DEBUG_POSTFIX@
+Libs: -L${libdir} -l@LIBPNG_IMPORT_PREFIX@png@PNGLIB_MAJOR@@PNGLIB_MINOR@@PNGLIB_DEBUG_POSTFIX@
 Libs.private: @LIBS@
 Cflags: -I${includedir}
