diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libpng-1.6.37.orig/CMakeLists.txt libpng-1.6.37/CMakeLists.txt
--- libpng-1.6.37.orig/CMakeLists.txt	2020-11-23 14:50:20.708632100 +0100
+++ libpng-1.6.37/CMakeLists.txt	2020-11-23 14:47:19.731259700 +0100
@@ -830,7 +830,7 @@
 set(exec_prefix ${CMAKE_INSTALL_PREFIX})
 set(libdir      ${CMAKE_INSTALL_FULL_LIBDIR})
 set(includedir  ${CMAKE_INSTALL_FULL_INCLUDEDIR})
-set(LIBS        "-lz -lm")
+#set(LIBS        "-lz -lm")
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libpng.pc.in
 ${CMAKE_CURRENT_BINARY_DIR}/${PNGLIB_NAME}.pc @ONLY)
 create_symlink(libpng.pc FILE ${PNGLIB_NAME}.pc)
