diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user ladspa_sdk_1.17/src.orig/plugins/filter.c ladspa_sdk_1.17/src/plugins/filter.c
--- ladspa_sdk_1.17/src.orig/plugins/filter.c	2021-09-10 08:44:43.000000000 +0000
+++ ladspa_sdk_1.17/src/plugins/filter.c	2024-10-12 20:53:01.108159827 +0000
@@ -11,6 +11,17 @@
 
 /*****************************************************************************/
 
+#ifdef _MSC_VER
+#define _USE_MATH_DEFINES
+#include <math.h>
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+#ifndef M_PI_2
+#define M_PI_2 1.57079632679489661923
+#endif
+#endif
+
 #include <math.h>
 #include <stdlib.h>
 #include <string.h>
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user ladspa_sdk_1.17/src.orig/plugins/sine.cpp ladspa_sdk_1.17/src/plugins/sine.cpp
--- ladspa_sdk_1.17/src.orig/plugins/sine.cpp	2021-09-10 08:44:43.000000000 +0000
+++ ladspa_sdk_1.17/src/plugins/sine.cpp	2024-10-12 20:53:41.614975914 +0000
@@ -15,6 +15,17 @@
 
 /*****************************************************************************/
 
+#ifdef _MSC_VER
+#define _USE_MATH_DEFINES
+#include <math.h>
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+#ifndef M_PI_2
+#define M_PI_2 1.57079632679489661923
+#endif
+#endif
+
 #include <cmath>
 #include <cstdlib>
 #include <cstring>
