diff --git a/src/cmake/modules/FindSndFile.cmake b/src/cmake/modules/FindSndFile.cmake
new file mode 100644
index 0000000..be68dd3
--- /dev/null
+++ b/src/cmake/modules/FindSndFile.cmake
@@ -0,0 +1,35 @@
+include(FindPackageHandleStandardArgs)
+
+find_path(SndFile_INCLUDE_DIR
+    NAMES
+      sndfile.h
+    PATH_SUFFIXES
+      include
+)
+
+find_library(SndFile_LIBRARY
+    NAMES
+      sndfile sndfile-1
+    PATH_SUFFIXES
+      lib64 lib
+)
+
+find_package_handle_standard_args(SndFile
+  REQUIRED_VARS
+    SndFile_LIBRARY
+    SndFile_INCLUDE_DIR
+)
+
+set(SndFile_LIBRARIES
+    ${SndFile_LIBRARY}
+)
+
+if(SndFile_FOUND AND NOT TARGET SndFile::SndFile)
+    add_library(SndFile::SndFile UNKNOWN IMPORTED)
+    set_target_properties(SndFile::SndFile PROPERTIES
+        IMPORTED_LOCATION "${SndFile_LIBRARY}"
+        INTERFACE_INCLUDE_DIRECTORIES "${SndFile_INCLUDE_DIR}"
+    )
+endif()
+
+mark_as_advanced(SndFile_INCLUDE_DIR SndFile_LIBRARY SndFile_LIBRARIES)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
new file mode 100644
index 0000000..e9545af
--- /dev/null
+++ b/src/CMakeLists.txt
@@ -0,0 +1,88 @@
+cmake_minimum_required(VERSION 3.16)
+
+project(ladspa-sdk VERSION 1.17)
+
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
+include(GNUInstallDirs)
+
+option(BUILD_PROGRAMS "Build the analyseplugin, applyplugin and listplugins executables" ON)
+
+if(BUILD_PROGRAMS)
+    find_package(SndFile REQUIRED)
+endif()
+
+set(INSTALL_PLUGINS_DIR "${CMAKE_INSTALL_LIBDIR}/ladspa")
+
+add_compile_definitions(DEFAULT_LADSPA_PATH="/usr/lib/ladspa/")
+
+add_library(ladspa INTERFACE)
+target_sources(ladspa PUBLIC
+  FILE_SET HEADERS
+    FILES ladspa.h
+)
+
+install(TARGETS ladspa EXPORT LadspaTargets
+  FILE_SET HEADERS DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(EXPORT LadspaTargets NAMESPACE Ladspa::
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/ladspa
+)
+
+function(add_ladspa_plugin ARG_TARGET)
+    set(multiValueArgs SOURCES)
+    cmake_parse_arguments(ARG "" "" "${multiValueArgs}" ${ARGN})
+
+    add_library(${ARG_TARGET} MODULE ${ARG_SOURCES})
+    target_link_libraries(${ARG_TARGET} ladspa)
+    set_target_properties(${ARG_TARGET} PROPERTIES
+      C_VISIBILITY_PRESET hidden
+      CXX_VISIBILITY_PRESET hidden
+      VISIBILITY_INLINES_HIDDEN ON
+    )
+
+    if(MSVC)
+      target_link_options(${ARG_TARGET} PRIVATE
+          "/EXPORT:ladspa_descriptor"
+      )
+    elseif(NOT APPLE)
+      target_link_options(${ARG_TARGET} PRIVATE
+          "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/gcc_exports.map"
+      )
+    endif()
+    if(NOT MSVC)
+      target_link_libraries(${ARG_TARGET} m)
+    endif()
+
+    install(TARGETS ${ARG_TARGET} LIBRARY DESTINATION "${INSTALL_PLUGINS_DIR}")
+endfunction()
+
+function(add_ladspa_tool ARG_TARGET)
+    set(multiValueArgs SOURCES LINK_LIBRARIES)
+    cmake_parse_arguments(ARG "" "" "${multiValueArgs}" ${ARGN})
+
+    add_executable(${ARG_TARGET} ${ARG_SOURCES})
+
+    target_link_libraries(${ARG_TARGET} ladspa ${CMAKE_DL_LIBS} ${ARG_LINK_LIBRARIES})
+    if(NOT MSVC)
+      target_link_libraries(${ARG_TARGET} m)
+    endif()
+
+    set_target_properties(${ARG_TARGET} PROPERTIES
+        C_VISIBILITY_PRESET hidden
+        CXX_VISIBILITY_PRESET hidden
+        VISIBILITY_INLINES_HIDDEN ON
+    )
+
+    install(TARGETS ${ARG_TARGET} DESTINATION "${CMAKE_INSTALL_BINDIR}")
+endfunction()
+
+add_ladspa_plugin(amp SOURCES plugins/amp.c)
+add_ladspa_plugin(delay SOURCES plugins/delay.c)
+add_ladspa_plugin(filter SOURCES plugins/filter.c)
+add_ladspa_plugin(noise SOURCES plugins/noise.c)
+add_ladspa_plugin(sine SOURCES plugins/sine.cpp)
+
+if(BUILD_PROGRAMS)
+    add_ladspa_tool(analyseplugin SOURCES analyseplugin.c load.c default.c)
+    add_ladspa_tool(applyplugin SOURCES applyplugin.c load.c default.c LINK_LIBRARIES SndFile::SndFile)
+    add_ladspa_tool(listplugins SOURCES listplugins.c search.c)
+endif()
