diff --git a/bozosoity-checker.pl b/bozosoity-checker.pl
index b320a86..3d1e8f5 100755
--- a/bozosoity-checker.pl
+++ b/bozosoity-checker.pl
@@ -1,4 +1,5 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
+use warnings;
 
 %used = ();
 
diff --git a/docs/addstyle.pl b/docs/addstyle.pl
index 825d909..f21738d 100755
--- a/docs/addstyle.pl
+++ b/docs/addstyle.pl
@@ -1,4 +1,5 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
+use warnings;
 
 $style = <<EOB;
 <style>
diff --git a/docs/makedocs.pl b/docs/makedocs.pl
index 21c90c2..71c694d 100755
--- a/docs/makedocs.pl
+++ b/docs/makedocs.pl
@@ -1,4 +1,5 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
+use warnings;
 
 use XML::Parser;
 
diff --git a/impulses/reorder.pl b/impulses/reorder.pl
index 7d2e67a..0e17bda 100755
--- a/impulses/reorder.pl
+++ b/impulses/reorder.pl
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 $n = 1;
 while (<>) {
diff --git a/impulses/scale.pl b/impulses/scale.pl
index 381e7af..f5c6d9a 100755
--- a/impulses/scale.pl
+++ b/impulses/scale.pl
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 $scale = shift;
 
diff --git a/makestub.pl b/makestub.pl
index 99e2108..4ba5b45 100755
--- a/makestub.pl
+++ b/makestub.pl
@@ -1,4 +1,5 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
+use warnings;
 
 use List::MoreUtils qw(any);
 use XML::Parser;
diff --git a/metadata/lxml2rdf.pl b/metadata/lxml2rdf.pl
index 779a007..d0f5285 100755
--- a/metadata/lxml2rdf.pl
+++ b/metadata/lxml2rdf.pl
@@ -1,4 +1,5 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
+use warnings;
 
 print <<EOB;
 <?xml version='1.0' encoding='ISO-8859-1'?>
diff --git a/metadata/txt2scale.pl b/metadata/txt2scale.pl
index 9bcc7ae..e3447b6 100755
--- a/metadata/txt2scale.pl
+++ b/metadata/txt2scale.pl
@@ -1,4 +1,5 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
+use warnings;
 
 print <<EOB;
 <?xml version='1.0' encoding='UTF-8'?>
diff --git a/mkspec.pl b/mkspec.pl
index 1014b53..5d4cc03 100755
--- a/mkspec.pl
+++ b/mkspec.pl
@@ -1,4 +1,5 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
+use warnings;
 
 $package = shift(@ARGV);
 $version = shift(@ARGV);
diff --git a/timetest/autotimetest.pl b/timetest/autotimetest.pl
index 09db770..3ed1e67 100755
--- a/timetest/autotimetest.pl
+++ b/timetest/autotimetest.pl
@@ -1,4 +1,5 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
+use warnings;
 
 $html = 0;
 $skip = 0;
diff --git a/timetest/timecmp b/timetest/timecmp
index 9a92e24..5a9093f 100755
--- a/timetest/timecmp
+++ b/timetest/timecmp
@@ -1,4 +1,5 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
+use warnings;
 
 my $pattern = shift @ARGV;
 
diff --git a/timetest/timetest b/timetest/timetest
index cad2dc2..aeb820c 100755
--- a/timetest/timetest
+++ b/timetest/timetest
@@ -1,4 +1,5 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
+use warnings;
 
 my $pattern = shift @ARGV;
 
