diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user dbus-1.13.18.orig/CMakeLists.txt dbus-1.13.18/CMakeLists.txt
--- dbus-1.13.18.orig/CMakeLists.txt	2020-07-01 16:59:09.000000000 +0200
+++ dbus-1.13.18/CMakeLists.txt	2020-09-15 22:25:41.924633200 +0200
@@ -501,8 +501,8 @@
     set(DBUS_SESSION_BUS_LISTEN_ADDRESS "autolaunch:" CACHE STRING "session bus default listening address")
     set(DBUS_SESSION_BUS_CONNECT_ADDRESS "autolaunch:" CACHE STRING "session bus fallback address for clients")
 
-    set(DBUS_SYSTEM_CONFIG_FILE "share/dbus-1/system.conf")
-    set(DBUS_SESSION_CONFIG_FILE "share/dbus-1/session.conf")
+    set(DBUS_SYSTEM_CONFIG_FILE "${CMAKE_INSTALL_DATADIR}/dbus-1/system.conf")
+    set(DBUS_SESSION_CONFIG_FILE "${CMAKE_INSTALL_DATADIR}/dbus-1/session.conf")
   # bus-test expects a non empty string
     set(DBUS_USER "Administrator")
     set(DBUS_TEST_USER "guest")
