diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' dbus-1.12.12.orig/bus/org.freedesktop.dbus-session.plist.in dbus-1.12.12/bus/org.freedesktop.dbus-session.plist.in
--- dbus-1.12.12.orig/bus/org.freedesktop.dbus-session.plist.in	2017-10-30 13:26:18.000000000 +0100
+++ dbus-1.12.12/bus/org.freedesktop.dbus-session.plist.in	2019-03-05 15:30:11.000000000 +0100
@@ -5,15 +5,6 @@
 	<key>Label</key>
 	<string>org.freedesktop.dbus-session</string>
 
-	<key>ServiceIPC</key>
-	<true/>
-
-	<!-- Please uncomment on 10.4; OnDemand doesn't work properly there. -->
-	<!--
-	<key>OnDemand</key>
-	<false />
-	-->
-
 	<key>ProgramArguments</key>
 	<array>
 		<string>@DBUS_DAEMONDIR@/dbus-daemon</string>
