From 0af7a655526041f620e3829afd59c1ad51158aa8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20K=C3=BCnzel?= <julius.kuenzel@kde.org>
Date: Sun, 24 Aug 2025 23:23:55 +0000
Subject: [PATCH] Fix #if that should be #ifdef

Related to !107
---
 src/kmsystemtray.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/kmsystemtray.cpp b/src/kmsystemtray.cpp
index 2bae7493d..200f7c87c 100644
--- a/src/kmsystemtray.cpp
+++ b/src/kmsystemtray.cpp
@@ -49,7 +49,7 @@ KMSystemTray::KMSystemTray(QObject *parent)
     setToolTipIconByName(QStringLiteral("kmail"));
     setIconByName(QStringLiteral("kmail-symbolic"));
 
-#if HAVE_X11
+#ifdef HAVE_X11
     KMMainWidget *mainWidget = kmkernel->getKMMainWidget();
     if (mainWidget) {
         QWidget *mainWin = mainWidget->window();
-- 
GitLab

