diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user vlc-3.0.11.orig/sdk/include/vlc/libvlc_media.h vlc-3.0.11/sdk/include/vlc/libvlc_media.h
--- vlc-3.0.11.orig/sdk/include/vlc/libvlc_media.h	2020-06-04 17:50:05.000000000 +0200
+++ vlc-3.0.11/sdk/include/vlc/libvlc_media.h	2020-11-06 17:21:23.214441900 +0100
@@ -26,6 +26,11 @@
 #ifndef VLC_LIBVLC_MEDIA_H
 #define VLC_LIBVLC_MEDIA_H 1
 
+#ifdef _MSC_VER
+#include <BaseTsd.h> // ssize_t on Windows
+#define ssize_t SSIZE_T
+#endif
+
 # ifdef __cplusplus
 extern "C" {
 # endif
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user vlc-3.0.11.orig/sdk/include/vlc/plugins/vlc_arrays.h vlc-3.0.11/sdk/include/vlc/plugins/vlc_arrays.h
--- vlc-3.0.11.orig/sdk/include/vlc/plugins/vlc_arrays.h	2020-06-04 17:50:05.000000000 +0200
+++ vlc-3.0.11/sdk/include/vlc/plugins/vlc_arrays.h	2020-11-06 17:21:07.742513100 +0100
@@ -25,6 +25,11 @@
 #ifndef VLC_ARRAYS_H_
 #define VLC_ARRAYS_H_
 
+#ifdef _MSC_VER
+#include <BaseTsd.h> // ssize_t on Windows
+#define ssize_t SSIZE_T
+#endif
+
 /**
  * \file
  * This file defines functions, structures and macros for handling arrays in vlc
