diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' curl-7.61.0.orig/lib/CMakeLists.txt curl-7.61.0/lib/CMakeLists.txt
--- curl-7.61.0.orig/lib/CMakeLists.txt	2018-07-09 08:42:12.000000000 +0200
+++ curl-7.61.0/lib/CMakeLists.txt	2018-08-16 12:30:51.077754200 +0200
@@ -94,7 +94,7 @@
 set_target_properties(${LIB_NAME} PROPERTIES PREFIX "")
 set_target_properties(${LIB_NAME} PROPERTIES IMPORT_PREFIX "")
 
-if(WIN32)
+if(MSVC)
   if(NOT CURL_STATICLIB)
     # Add "_imp" as a suffix before the extension to avoid conflicting with the statically linked "libcurl.lib"
     set_target_properties(${LIB_NAME} PROPERTIES IMPORT_SUFFIX "_imp.lib")
