From 441796a4c316b2002f0f97a56727073072b8f3bf Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Tue, 25 May 2021 22:54:01 +0200
Subject: [PATCH 2/2] Use QGuiApplication on Android

---
 src/main.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/main.cpp b/src/main.cpp
index f39926a..4ebc18d 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -4,7 +4,14 @@
  * SPDX-License-Identifier: GPL-3.0-or-later
  */
 
+#include <QtGlobal>
+
+#ifdef Q_OS_ANDROID
+#include <QGuiApplication>
+#else
 #include <QApplication>
+#endif
+
 #include <QQmlApplicationEngine>
 #include <QtQml>
 #include <QUrl>
@@ -27,7 +34,11 @@
 Q_DECL_EXPORT int main(int argc, char *argv[])
 {
     QGuiApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+#ifdef Q_OS_ANDROID
+    QGuiApplication app {argc, argv};
+#else
     QApplication app {argc, argv};
+#endif
     KLocalizedString::setApplicationDomain("kongress");
 
     KAboutData about {QStringLiteral("kongress"), i18n("Kongress"), QStringLiteral(KONGRESS_VERSION_STRING), i18n("KDE Conference Companion"), KAboutLicense::GPL_V3, i18n("© 2021 KDE Community")};
-- 
2.31.1

