From 08cb06fa59255ef6feac9662ff9edabb245024e8 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Tue, 25 May 2021 23:07:34 +0200
Subject: [PATCH 3/3] Use QGuiApplication instead of QApplication where
 possible

---
 src/main.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/main.cpp b/src/main.cpp
index 4ebc18d..6075b77 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -54,11 +54,11 @@ Q_DECL_EXPORT int main(int argc, char *argv[])
     parser.process(app);
     about.processCommandLine(&parser);
 
-    QApplication::setApplicationName(about.componentName());
-    QApplication::setApplicationDisplayName(about.displayName());
-    QApplication::setOrganizationDomain(about.organizationDomain());
-    QApplication::setApplicationVersion(about.version());
-    QApplication::setWindowIcon(QIcon::fromTheme(QStringLiteral("org.kde.kongress")));
+    QGuiApplication::setApplicationName(about.componentName());
+    QGuiApplication::setApplicationDisplayName(about.displayName());
+    QGuiApplication::setOrganizationDomain(about.organizationDomain());
+    QGuiApplication::setApplicationVersion(about.version());
+    QGuiApplication::setWindowIcon(QIcon::fromTheme(QStringLiteral("org.kde.kongress")));
 
     qmlRegisterType<LocalCalendar>("org.kde.kongress", 0, 1, "LocalCalendar");
     qmlRegisterType<EventModel>("org.kde.kongress", 0, 1, "EventModel");
-- 
2.31.1

