diff -Nrub -x '*~' -x '*.rej' -x '*.orig' -x '*.o' -x '*.pyc' -x CMakeLists.txt.user -x .idea -x .clang-format qtopenapi-everywhere-src-6.11.1.orig/src/tools/qtopenapi-generator/CMakeLists.txt qtopenapi-everywhere-src-6.11.1/src/tools/qtopenapi-generator/CMakeLists.txt
--- qtopenapi-everywhere-src-6.11.1.orig/src/tools/qtopenapi-generator/CMakeLists.txt	2026-05-08 03:19:01.000000000 +0200
+++ qtopenapi-everywhere-src-6.11.1/src/tools/qtopenapi-generator/CMakeLists.txt	2026-09-14 12:26:06.449796400 +0200
@@ -86,9 +86,6 @@
 if(need_call_maven)
     message(STATUS "Building the Qt6 Client Code Generator with Maven")
     set(cmd_suffix "")
-    if(CMAKE_HOST_WIN32)
-        set(cmd_suffix ".cmd")
-    endif()
     set(extra_args "")
     if(NOT QT_FEATURE_developer_build)
         list(APPEND extra_args
