diff --git a/src/qtwidgets/views/TabBar.cpp b/src/qtwidgets/views/TabBar.cpp
index 10f88cdc0..53c90ca69 100644
--- a/src/qtwidgets/views/TabBar.cpp
+++ b/src/qtwidgets/views/TabBar.cpp
@@ -25,6 +25,9 @@
 #include <QMouseEvent>
 #include <QApplication>
 #include <QProxyStyle>
+#include <QStyleOptionTab>
+#include <QPainter>
+
 
 using namespace KDDockWidgets;
 using namespace KDDockWidgets::QtWidgets;
@@ -54,6 +57,27 @@ public:
         }
         return baseStyle()->styleHint(hint, option, widget, returnData);
     }
+#if defined(Q_OS_WIN) || defined(Q_OS_MAC)
+    void drawControl(ControlElement element, const QStyleOption* option, QPainter* painter, const QWidget* widget) const override {
+        if (element == CE_TabBarTabLabel) {
+            if (auto tab = qstyleoption_cast<const QStyleOptionTab*>(option)) {
+                if (tab->state & QStyle::State_Selected) {
+                    painter->fillRect(tab->rect, qApp->palette().alternateBase());
+                } else {
+                    painter->fillRect(tab->rect, qApp->palette().button());
+                }
+                return QProxyStyle::drawControl(element, tab, painter, widget);
+            }
+        }
+        if (element == CE_TabBarTab) {
+            if (auto tab = qstyleoption_cast<const QStyleOptionTab*>(option)) {
+                painter->fillRect(tab->rect, qApp->palette().base());
+                return QProxyStyle::drawControl(element, tab, painter, widget);
+            }
+        }
+        QProxyStyle::drawControl(element, option, painter, widget);
+    }
+#endif
 };
 }
 
