diff --git a/src/client/shm_pool.cpp b/src/client/shm_pool.cpp
index ebfaa3f874b08d2837f0aac65d3b25a0681779df..a532ac2b65e52f3ed5ee9e4fceed2881351b0529 100644
--- a/src/client/shm_pool.cpp
+++ b/src/client/shm_pool.cpp
@@ -116,7 +116,7 @@ EventQueue *ShmPool::eventQueue()
 
 bool ShmPool::Private::createPool()
 {
-#if HAVE_MEMFD
+#ifdef HAVE_MEMFD
     fd = memfd_create("kwayland-shared", MFD_CLOEXEC | MFD_ALLOW_SEALING);
     if (fd >= 0) {
         fcntl(fd, F_ADD_SEALS, F_SEAL_SHRINK | F_SEAL_SEAL);
