diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' qt-creator-opensource-src-4.10.1.orig/src/tools/clangpchmanagerbackend/CMakeLists.txt qt-creator-opensource-src-4.10.1/src/tools/clangpchmanagerbackend/CMakeLists.txt
--- qt-creator-opensource-src-4.10.1.orig/src/tools/clangpchmanagerbackend/CMakeLists.txt	2019-10-02 21:32:00.000000000 +0200
+++ qt-creator-opensource-src-4.10.1/src/tools/clangpchmanagerbackend/CMakeLists.txt	2019-10-17 13:52:04.374911600 +0200
@@ -3,7 +3,7 @@
 add_qtc_executable(clangpchmanagerbackend
   DEPENDS
     clangrefactoringbackend_lib clangpchmanagerbackend_lib
-    clangTooling libclang Sqlite ClangSupport clangToolingRefactor clangQuery clangIndex
+    clangTooling libclang Sqlite ClangSupport clangToolingRefactoring clangQuery clangIndex
   SOURCES
     clangpchmanagerbackendmain.cpp
 )
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' qt-creator-opensource-src-4.10.1.orig/tests/unit/unittest/CMakeLists.txt qt-creator-opensource-src-4.10.1/tests/unit/unittest/CMakeLists.txt
--- qt-creator-opensource-src-4.10.1.orig/tests/unit/unittest/CMakeLists.txt	2019-10-02 21:32:00.000000000 +0200
+++ qt-creator-opensource-src-4.10.1/tests/unit/unittest/CMakeLists.txt	2019-10-17 13:52:07.965603000 +0200
@@ -269,7 +269,7 @@
     mockrefactoringserver.h
   )
   target_link_libraries(unittest
-    PRIVATE clangTooling clangIndex clangQuery clangToolingRefactor)
+    PRIVATE clangTooling clangIndex clangQuery clangToolingRefactoring)
 endif()
 
 if (TARGET clangFormat)
