diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user lensfun-0.3.2.orig/libs/lensfun/CMakeLists.txt lensfun-0.3.2/libs/lensfun/CMakeLists.txt
--- lensfun-0.3.2.orig/libs/lensfun/CMakeLists.txt	2015-11-15 18:07:26.000000000 +0100
+++ lensfun-0.3.2/libs/lensfun/CMakeLists.txt	2020-02-26 11:02:02.000000000 +0100
@@ -21,6 +21,7 @@
   TARGET_LINK_LIBRARIES(lensfun tre_regex)
 ENDIF()
 TARGET_LINK_LIBRARIES(lensfun ${GLIB2_LIBRARIES})
+target_link_directories(lensfun PRIVATE ${GLIB2_LIBRARY_DIRS})
 
 INSTALL(TARGETS lensfun 
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
