diff --git i/CMakeLists.txt w/CMakeLists.txt
index ceac8fc..5de53dc 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -162,13 +162,13 @@ IF(BUILD_DOC)
   ADD_SUBDIRECTORY(docs)
 ENDIF()
 
-IF(WIN32 AND NOT BUILD_STATIC)
-  FIND_FILE(GLIB2_DLL 
-            NAMES glib-2.dll glib-2-vs9.dll
-            PATHS "${GLIB2_BASE_DIR}/bin"
-            NO_SYSTEM_ENVIRONMENT_PATH)
-  INSTALL(FILES ${GLIB2_DLL} DESTINATION ${CMAKE_INSTALL_BINDIR})
-ENDIF()
+#IF(WIN32 AND NOT BUILD_STATIC)
+#  FIND_FILE(GLIB2_DLL 
+#            NAMES glib-2.dll glib-2-vs9.dll
+#            PATHS "${GLIB2_BASE_DIR}/bin"
+#            NO_SYSTEM_ENVIRONMENT_PATH)
+#  INSTALL(FILES ${GLIB2_DLL} DESTINATION ${CMAKE_INSTALL_BINDIR})
+#ENDIF()
 
 # print some information
 MESSAGE(STATUS "-------------------------------------------------------------------------------------")
diff --git i/apps/CMakeLists.txt w/apps/CMakeLists.txt
index 70c77fd..870e7a1 100644
--- i/apps/CMakeLists.txt
+++ w/apps/CMakeLists.txt
@@ -21,7 +21,7 @@ ENDIF(INSTALL_HELPER_SCRIPTS)
 
 # Inspired by http://bloerg.net/2012/11/10/cmake-and-distutils.html
 
-FIND_PROGRAM(PYTHON "python3")
+#FIND_PROGRAM(PYTHON "python3")
 IF(PYTHON)
     SET(SETUP_PY_IN "${CMAKE_CURRENT_SOURCE_DIR}/setup.py.in")
     SET(SETUP_PY    "${CMAKE_CURRENT_BINARY_DIR}/setup.py")