diff --git a/shell/main.cpp b/shell/main.cpp
index 3c4bb47231b3928565a796c292e1a48ca7c2f6cc..ea0d3bcec3f73826950a8141a708ce9e3ff51d60 100644
--- a/shell/main.cpp
+++ b/shell/main.cpp
@@ -31,7 +31,12 @@
 
 int main(int argc, char** argv)
 {
-    QGuiApplication::setAttribute(Qt::AA_UseHighDpiPixmaps);
+    /**
+     * enable high dpi support
+     */
+    QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps, true);
+    QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling, true);
+    QApplication::setStyle(QStringLiteral("breeze"));
+
 #if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
     QCoreApplication::setAttribute(Qt::AA_CompressTabletEvents);
 #endif
@@ -86,7 +91,7 @@ int main(int argc, char** argv)
                 break;
         }
     }
-    
+
     return app.exec();
 }
 
