diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libffi-3.3.orig/CMakeLists.txt libffi-3.3/CMakeLists.txt
--- libffi-3.3.orig/CMakeLists.txt	2020-12-16 12:17:09.671971800 +0100
+++ libffi-3.3/CMakeLists.txt	2020-12-16 12:17:03.495163500 +0100
@@ -41,6 +41,14 @@
 
 set_target_properties(ffi_shared PROPERTIES OUTPUT_NAME ffi)
 
+set(PACKAGE_NAME libffi)
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix "\${prefix}/bin")
+set(libdir "\${prefix}/lib")
+set(toolexeclibdir "\${prefix}/bin")
+set(includedir "\${prefix}/include")
+configure_file(libffi.pc.in ${CMAKE_BINARY_DIR}/libffi.pc @ONLY)
+
 install(TARGETS ffi_static ffi_shared
     EXPORT ${PROJECT_NAME}Targets
     RUNTIME DESTINATION bin
@@ -54,3 +62,7 @@
     ${CMAKE_BINARY_DIR}/include/ffi.h
     ${CMAKE_BINARY_DIR}/include/ffitarget.h
     DESTINATION include)
+    
+install(FILES 
+    ${CMAKE_BINARY_DIR}/libffi.pc
+    DESTINATION lib/pkgconfig)
