diff --git a/src/opentime/CMakeLists.txt b/src/opentime/CMakeLists.txt
index a274ef1..09a23f2 100644
--- a/src/opentime/CMakeLists.txt
+++ b/src/opentime/CMakeLists.txt
@@ -43,12 +43,6 @@ else()
         OPENTIME_STATIC)
 endif()
 
-if(APPLE)
-    set_target_properties(opentime PROPERTIES 
-        INSTALL_NAME_DIR "@loader_path"
-        MACOSX_RPATH ON)
-endif()
-
 target_compile_options(opentime PRIVATE
      $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
      -Wall>
diff --git a/src/opentimelineio/CMakeLists.txt b/src/opentimelineio/CMakeLists.txt
index 7c62556..5760969 100644
--- a/src/opentimelineio/CMakeLists.txt
+++ b/src/opentimelineio/CMakeLists.txt
@@ -118,12 +118,6 @@ else()
         OTIO_STATIC)
 endif()
 
-if(APPLE)
-    set_target_properties(opentimelineio PROPERTIES 
-        INSTALL_NAME_DIR "@loader_path"
-        MACOSX_RPATH ON)
-endif()
-
 # override any global CXX_FLAGS settings that came from dependencies
 target_compile_options(opentimelineio PRIVATE
      $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
