diff --git a/src/opentime/CMakeLists.txt b/src/opentime/CMakeLists.txt
index b7cc47c..838483f 100644
--- a/src/opentime/CMakeLists.txt
+++ b/src/opentime/CMakeLists.txt
@@ -24,12 +24,6 @@ set_target_properties(opentime PROPERTIES
     POSITION_INDEPENDENT_CODE TRUE
     WINDOWS_EXPORT_ALL_SYMBOLS true)
 
-if(APPLE)
-    set_target_properties(opentime PROPERTIES 
-        INSTALL_NAME_DIR "@loader_path"
-        MACOSX_RPATH ON)
-endif()
-
 target_compile_options(opentime PRIVATE
      $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
      -Wall>
diff --git a/src/opentimelineio/CMakeLists.txt b/src/opentimelineio/CMakeLists.txt
index 728b485..9e72806 100644
--- a/src/opentimelineio/CMakeLists.txt
+++ b/src/opentimelineio/CMakeLists.txt
@@ -98,12 +98,6 @@ set_target_properties(opentimelineio PROPERTIES
     POSITION_INDEPENDENT_CODE TRUE
     WINDOWS_EXPORT_ALL_SYMBOLS true)
 
-if(APPLE)
-    set_target_properties(opentimelineio PROPERTIES 
-        INSTALL_NAME_DIR "@loader_path"
-        MACOSX_RPATH ON)
-endif()
-
 # override any global CXX_FLAGS settings that came from dependencies
 target_compile_options(opentimelineio PRIVATE
      $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
