--- /dev/null	2024-01-12 16:00:00.000000000 +0100
+++ CMakeLists.txt	2024-01-12 16:00:00.000000000 +0100
@@ -0,0 +1,74 @@
+cmake_minimum_required(VERSION 3.16)
+
+project(ladspa-tap VERSION 1.0.1)
+
+include(GNUInstallDirs)
+
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+add_compile_options(-Wall -fomit-frame-pointer -funroll-loops -ffast-math)
+
+# m
+link_libraries(m)
+
+if(NOT APPLE AND NOT WIN32)
+    link_libraries(rt)
+endif()
+
+#include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+
+add_library(tap_reverb MODULE tap_reverb.c tap_reverb.h tap_reverb_presets.h tap_utils.h)
+add_library(tap_dynamics_m MODULE tap_dynamics_m.c tap_dynamics_presets.h tap_utils.h)
+add_library(tap_dynamics_st MODULE tap_dynamics_st.c tap_dynamics_presets.h tap_utils.h)
+
+add_library(tap_autopan MODULE tap_autopan.c tap_utils.h)
+add_library(tap_deesser MODULE tap_deesser.c tap_utils.h)
+add_library(tap_echo MODULE tap_echo.c tap_utils.h)
+add_library(tap_eq MODULE tap_eq.c tap_utils.h)
+add_library(tap_pinknoise MODULE tap_pinknoise.c tap_utils.h)
+add_library(tap_reflector MODULE tap_reflector.c tap_utils.h)
+add_library(tap_rotspeak MODULE tap_rotspeak.c tap_utils.h)
+add_library(tap_tremolo MODULE tap_tremolo.c tap_utils.h)
+add_library(tap_vibrato MODULE tap_vibrato.c tap_utils.h)
+add_library(tap_chorusflanger MODULE tap_chorusflanger.c tap_utils.h)
+add_library(tap_doubler MODULE tap_doubler.c tap_utils.h)
+add_library(tap_eqbw MODULE tap_eqbw.c tap_utils.h)
+add_library(tap_limiter MODULE tap_limiter.c tap_utils.h)
+add_library(tap_pitch MODULE tap_pitch.c tap_utils.h)
+add_library(tap_sigmoid MODULE tap_sigmoid.c tap_utils.h)
+add_library(tap_tubewarmth MODULE tap_tubewarmth.c tap_utils.h)
+
+
+set(INSTALL_PLUGINS_DIR "${CMAKE_INSTALL_LIBDIR}/ladspa")
+set(INSTALL_LRDF_DIR "${CMAKE_INSTALL_DATAROOTDIR}/ladspa/rdf")
+
+install(
+    TARGETS
+        tap_autopan
+        tap_deesser
+        tap_dynamics_m
+        tap_echo
+        tap_eq
+        tap_pinknoise
+        tap_reflector
+        tap_rotspeak
+        tap_tremolo
+        tap_vibrato
+        tap_chorusflanger
+        tap_doubler
+        tap_dynamics_st
+        tap_eqbw
+        tap_limiter
+        tap_pitch
+        tap_reverb
+        tap_sigmoid
+        tap_tubewarmth
+    LIBRARY
+    DESTINATION "${INSTALL_PLUGINS_DIR}"
+)
+
+install(
+    FILES
+        tap-plugins.rdf
+        tap_reverb.rdf
+    DESTINATION "${INSTALL_LRDF_DIR}"
+)

