From 517672e703b92835e554094c915dbc9ebee1b486 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20K=C3=BCnzel?= <jk.kdedev@smartlab.uber.space>
Date: Wed, 3 Jan 2024 12:49:02 +0000
Subject: [PATCH] FIx non-void return

---
 tests/testread2.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/testread2.cpp b/tests/testread2.cpp
index 2b1cde17..6935e7e5 100644
--- a/tests/testread2.cpp
+++ b/tests/testread2.cpp
@@ -10,6 +10,8 @@
 #include <converter/kcontacts/vcardconverter.h>
 #include <kcontacts/addressee.h>
 
+#include <stdlib.h>
+
 using namespace KContacts;
 
 int main()
@@ -47,4 +49,5 @@ int main()
             // qDebug()<<"\t\t"<< (*itr1).fullEmail() << "VS." << (*itr2).fullEmail();
         }
     }
+    return EXIT_SUCCESS;
 }
-- 
GitLab

