--- CMakeLists.txt.orig	2020-08-28 15:39:42.000000000 -0700
+++ CMakeLists.txt	2020-11-15 12:54:21.057275900 -0800
@@ -766,3 +766,6 @@
     endif ()
   endif ()
 endif ()
+
+configure_file(harfbuzz.pc.in harfbuzz.pc @ONLY)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/harfbuzz.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
--- /dev/null	2020-11-15 12:41:05.660371700 -0800
+++ harfbuzz.pc.in	2020-11-15 12:40:39.925558200 -0800
@@ -0,0 +1,6 @@
+Name: harfbuzz
+Description: HarfBuzz text shaping library
+Version: @HB_VERSION@
+
+Libs: -L@CMAKE_INSTALL_FULL_LIBDIR@ -lharfbuzz
+Cflags: -I@CMAKE_INSTALL_FULL_INCLUDEDIR@/harfbuzz
