diff --git a/kstyle/CMakeLists.txt b/kstyle/CMakeLists.txt
index 4aef5296..7b481570 100644
--- a/kstyle/CMakeLists.txt
+++ b/kstyle/CMakeLists.txt
@@ -9,7 +9,7 @@ find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
     ConfigWidgets
     WindowSystem)
 
-find_package(Qt5 ${QT_MIN_VERSION} COMPONENTS Quick X11Extras)
+find_package(Qt5 ${QT_MIN_VERSION} OPTIONAL_COMPONENTS Quick X11Extras)
 set(BREEZE_HAVE_QTQUICK ${Qt5Quick_FOUND})
 set(BREEZE_HAVE_QTX11EXTRAS ${Qt5X11Extras_FOUND})
 
-- 
2.35.1

