diff -ruNa libixion-0.19.0.orig/configure.ac libixion-0.19.0/configure.ac
--- libixion-0.19.0.orig/configure.ac	2023-09-27 01:47:25.000000000 +0200
+++ libixion-0.19.0/configure.ac	2025-01-14 17:45:30.793457863 +0100
@@ -20,7 +20,7 @@
 AM_INIT_AUTOMAKE([1.11 foreign dist-bzip2 dist-xz])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AX_CXX_COMPILE_STDCXX_17([noext], [mandatory])
-CXXFLAGS="$CXXFLAGS -fvisibility=hidden -Wall -Wshadow -Wno-write-strings -pthread"
+CXXFLAGS="$CXXFLAGS -Wall"
 
 AC_CANONICAL_HOST
 
@@ -31,7 +31,7 @@
     ;;
     *)
         _os=
-        LDFLAGS="$LDFLAGS -Wl,--no-as-needed -pthread -ldl"
+        LDFLAGS="$LDFLAGS"
     ;;
 esac
 
