diff -ruNa a/m4/ax_cxx_compile_stdcxx.m4 b/m4/ax_cxx_compile_stdcxx.m4
--- a/m4/ax_cxx_compile_stdcxx.m4	2022-04-22 02:12:40.000000000 +0200
+++ b/m4/ax_cxx_compile_stdcxx.m4	2024-12-07 10:59:55.217575836 +0100
@@ -92,7 +92,7 @@
     dnl http://h21007.www2.hp.com/portal/download/files/unprot/aCxx/PDF_Release_Notes/769149-001.pdf
     dnl Cray's crayCC needs "-h std=c++11"
     for alternative in ${ax_cxx_compile_alternatives}; do
-      for switch in -std=c++${alternative} +std=c++${alternative} "-h std=c++${alternative}"; do
+      for switch in -std:c++${alternative} -std=c++${alternative} +std=c++${alternative} "-h std=c++${alternative}"; do
         cachevar=AS_TR_SH([ax_cv_cxx_compile_cxx$1_$switch])
         AC_CACHE_CHECK(whether $CXX supports C++$1 features with $switch,
                        $cachevar,
