diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user dpkg-1.21.22.orig/src/deb/build.c dpkg-1.21.22/src/deb/build.c
--- dpkg-1.21.22.orig/src/deb/build.c	2023-07-26 09:15:09.932354315 +0000
+++ dpkg-1.21.22/src/deb/build.c	2023-07-26 09:13:44.094435456 +0000
@@ -510,7 +510,7 @@
     if (options->mode)
       command_add_args(&cmd, "--mode", options->mode, NULL);
     if (options->root_owner_group)
-      command_add_args(&cmd, "--owner", "root:0", "--group", "root:0", NULL);
+      command_add_args(&cmd, "--owner", "0", "--group", "0", NULL);
     command_add_args(&cmd, "--null", "--no-unquote", "--no-recursion",
                            "-T", "-", NULL);
     command_exec(&cmd);
