diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user gpgme-1.14.0.orig/CMakeLists.txt gpgme-1.14.0/CMakeLists.txt
--- gpgme-1.14.0.orig/CMakeLists.txt	2020-11-06 20:20:47.562703700 +0100
+++ gpgme-1.14.0/CMakeLists.txt	2020-11-06 20:16:07.232210400 +0100
@@ -15,27 +15,8 @@
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-if (WIN32)
-    find_file(GPGME_HEADER gpgme.h REQUIRED)
-    get_filename_component(GPGME_INCLUDE_DIR ${GPGME_HEADER} DIRECTORY)
-    find_library(GPGME_LIB gpgme.lib REQUIRED)
-
-    find_file(ASSUAN_HEADER assuan.h REQUIRED)
-    get_filename_component(ASSUAN_INCLUDE_DIR ${ASSUAN_HEADER} DIRECTORY)
-    find_library(ASSUAN_LIB assuan.lib REQUIRED)
-
-    find_file(GPG_ERROR_HEADER gpg-error.h REQUIRED)
-    get_filename_component(GPG_ERROR_INCLUDE_DIR ${GPG_ERROR_HEADER} DIRECTORY)
-    find_library(GPG_ERROR_LIB gpg-error.lib REQUIRED)
-
-
-    set(GPGME_INCLUDE_DIRS ${GPGME_INCLUDE_DIR} ${ASSUAN_INCLUDE_DIR} ${GPG_ERROR_INCLUDE_DIR})
-    set(GPGME_LIBRARIES ${GPGME_LIB} ${ASSUAN_LIB} ${GPG_ERROR_LIB})
-    set(GPGME_FOUND ON)
-else()
-    find_package(PkgConfig REQUIRED)
-    pkg_check_modules(GPGME REQUIRED gpgme)
-endif()
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(GPGME REQUIRED gpgme)
 
 include(CMakeDependentOption)
 include(CheckIncludeFile)
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user gpgme-1.14.0.orig/lang/cpp/src/CMakeLists.txt gpgme-1.14.0/lang/cpp/src/CMakeLists.txt
--- gpgme-1.14.0.orig/lang/cpp/src/CMakeLists.txt	2020-11-06 20:20:47.565704200 +0100
+++ gpgme-1.14.0/lang/cpp/src/CMakeLists.txt	2020-11-06 20:20:15.161481300 +0100
@@ -88,6 +88,7 @@
     $<INSTALL_INTERFACE:${GPGME_INCLUDE_DIRS}>
 )
 target_link_libraries(gpgmepp ${GPGME_LIBRARIES})
+target_link_directories(gpgmepp PUBLIC ${GPGME_LIBRARY_DIRS})
 
 parse_lt(LIBGPGMEPP)
 math(EXPR MAJOR "${LIBGPGMEPP_LT_CURRENT}-${LIBGPGMEPP_LT_AGE}")
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user gpgme-1.14.0.orig/lang/cpp/src/importresult.cpp gpgme-1.14.0/lang/cpp/src/importresult.cpp
--- gpgme-1.14.0.orig/lang/cpp/src/importresult.cpp	2019-02-28 14:01:47.000000000 +0100
+++ gpgme-1.14.0/lang/cpp/src/importresult.cpp	2020-11-06 20:16:07.233210600 +0100
@@ -35,7 +35,9 @@
 #include <cstring>
 
 #include <string.h>
+#ifdef HAVE_STRINGS_H
 #include <strings.h>
+#endif
 #include <istream>
 #include <iterator>
 
