diff -Nrub -x *~ -x *\.rej -x *\.orig -x *\.o -x *\.pyc -x CMakeLists.txt.user gpgme-1.21.0.orig/lang/cpp/src/gpgmepp_export.h gpgme-1.21.0/lang/cpp/src/gpgmepp_export.h
--- gpgme-1.21.0.orig/lang/cpp/src/gpgmepp_export.h	2023-11-09 17:33:55
+++ gpgme-1.21.0/lang/cpp/src/gpgmepp_export.h	2023-11-09 17:33:45
@@ -59,7 +59,7 @@
 #endif
 
 #ifndef GPGMEPP_DEPRECATED_EXPORT
-#  define GPGMEPP_DEPRECATED_EXPORT GPGMEPP_EXPORT GPGMEPP_DEPRECATED
+#  define GPGMEPP_DEPRECATED_EXPORT GPGMEPP_DEPRECATED GPGMEPP_EXPORT
 #endif
 
 #ifndef GPGMEPP_DEPRECATED_NO_EXPORT
