diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user gpgme-1.14.0.orig/lang/cpp/src/CMakeLists.txt gpgme-1.14.0/lang/cpp/src/CMakeLists.txt
--- gpgme-1.14.0.orig/lang/cpp/src/CMakeLists.txt	2020-11-16 09:02:09.863366800 +0100
+++ gpgme-1.14.0/lang/cpp/src/CMakeLists.txt	2020-11-16 09:02:04.184475900 +0100
@@ -90,7 +90,7 @@
     $<INSTALL_INTERFACE:${GPGME_INCLUDE_DIRS}>
     $<INSTALL_INTERFACE:${GPGME_INCLUDE_DIRS}/gpgme++>
 )
-target_link_libraries(gpgmepp ${GPGME_LIBRARIES})
+target_link_libraries(gpgmepp PUBLIC ${GPGME_LIBRARIES})
 target_link_directories(gpgmepp PUBLIC ${GPGME_LIBRARY_DIRS})
 
 parse_lt(LIBGPGMEPP)
