diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' qtbase-everywhere-src-5.10.1.orig/src/corelib/io/qstandardpaths_mac.mm qtbase-everywhere-src-5.10.1/src/corelib/io/qstandardpaths_mac.mm
--- qtbase-everywhere-src-5.10.1.orig/src/corelib/io/qstandardpaths_mac.mm	2018-05-02 20:25:14.000000000 -0700
+++ qtbase-everywhere-src-5.10.1/src/corelib/io/qstandardpaths_mac.mm	2018-05-02 20:23:31.000000000 -0700
@@ -200,7 +200,7 @@
         }
     }
 
-    if (type == AppDataLocation || type == AppLocalDataLocation) {
+    if (type == AppDataLocation || type == AppLocalDataLocation || type == GenericDataLocation) {
         CFBundleRef mainBundle = CFBundleGetMainBundle();
         if (mainBundle) {
             if (QCFType<CFURLRef> resourcesURL = CFBundleCopyResourcesDirectoryURL(mainBundle)) {
