diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user qtbase-everywhere-src-5.15.2.orig/src/corelib/mimetypes/mime/generate.pl qtbase-everywhere-src-5.15.2/src/corelib/mimetypes/mime/generate.pl
--- qtbase-everywhere-src-5.15.2.orig/src/corelib/mimetypes/mime/generate.pl	2020-10-27 09:02:11.000000000 +0100
+++ qtbase-everywhere-src-5.15.2/src/corelib/mimetypes/mime/generate.pl	2021-11-04 14:52:12.868204800 +0100
@@ -77,6 +77,7 @@
 if ($cmd) {
     # Run the command and read everything
     open CMD, "$cmd |";
+    binmode CMD;
     $data = <CMD>;
     close CMD;
     die("Failed to run $cmd") if ($? >> 8);
