diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' freetype-2.9.1.orig/CMakeLists.txt freetype-2.9.1/CMakeLists.txt
--- freetype-2.9.1.orig/CMakeLists.txt	2018-05-01 12:45:45.000000000 +0200
+++ freetype-2.9.1/CMakeLists.txt	2018-09-25 10:50:38.248394100 +0200
@@ -415,11 +415,19 @@
       PATTERN "internal" EXCLUDE
       PATTERN "ftconfig.h" EXCLUDE
       PATTERN "ftoption.h" EXCLUDE)
+  if(UNIX)
   install(
     FILES ${PROJECT_BINARY_DIR}/include/freetype/config/ftconfig.h
           ${PROJECT_BINARY_DIR}/include/freetype/config/ftoption.h
       DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/freetype2/freetype/config
       COMPONENT headers)
+    else()
+      install(
+        FILES ${PROJECT_SOURCE_DIR}/include/freetype/config/ftconfig.h
+              ${PROJECT_SOURCE_DIR}/include/freetype/config/ftoption.h
+          DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/freetype2/freetype/config
+          COMPONENT headers)
+    endif()
 endif ()
 
 if (NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
