diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' freetype-2.9.1.orig/CMakeLists.txt freetype-2.9.1/CMakeLists.txt
--- freetype-2.9.1.orig/CMakeLists.txt	2018-09-26 00:34:46.786315300 +0200
+++ freetype-2.9.1/CMakeLists.txt	2018-09-26 00:34:31.885969800 +0200
@@ -328,7 +328,7 @@
 endif()
 
 
-add_library(freetype
+add_library(freetype SHARED
   ${PUBLIC_HEADERS}
   ${PUBLIC_CONFIG_HEADERS}
   ${PRIVATE_HEADERS}
@@ -344,7 +344,7 @@
 
 if (WIN32)
   target_compile_definitions(
-    freetype PRIVATE _CRT_SECURE_NO_WARNINGS _CRT_NONSTDC_NO_WARNINGS)
+    freetype PRIVATE _CRT_SECURE_NO_WARNINGS _CRT_NONSTDC_NO_WARNINGS DLL_EXPORT)
 endif ()
 
 if (BUILD_SHARED_LIBS)
@@ -463,10 +463,10 @@
   install(
     TARGETS freetype
       EXPORT freetype-targets
+      RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
       LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-      FRAMEWORK DESTINATION Library/Frameworks
-      COMPONENT libraries)
+      FRAMEWORK DESTINATION Library/Frameworks)
   install(
     EXPORT freetype-targets
       DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/freetype
