From 0d92f75534b2e00a49c9b1851941ce084c6ffbad Mon Sep 17 00:00:00 2001
From: Kurt Hindenburg <kurt.hindenburg@gmail.com>
Date: Fri, 31 Mar 2023 17:29:03 -0400
Subject: [PATCH] Fix macOS build due to SOCK_CLOEXEC

---
 src/lib/util/ksignalhandler.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/util/ksignalhandler.cpp b/src/lib/util/ksignalhandler.cpp
index 1a644805b..88ed29752 100644
--- a/src/lib/util/ksignalhandler.cpp
+++ b/src/lib/util/ksignalhandler.cpp
@@ -33,7 +33,7 @@ KSignalHandler::KSignalHandler()
     : d(new KSignalHandlerPrivate)
 {
     d->q = this;
-#ifndef Q_OS_WIN
+#if !(defined(Q_OS_WIN) || defined(Q_OS_MACOS))
     if (::socketpair(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0, KSignalHandlerPrivate::signalFd)) {
         qCWarning(KCOREADDONS_DEBUG) << "Couldn't create a socketpair";
         return;
-- 
GitLab

