diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user eigen-3.4.0.orig/CMakeLists.txt eigen-3.4.0/CMakeLists.txt
--- a/CMakeLists.txt	2021-08-18 22:41:58.000000000 +0200
+++ b/CMakeLists.txt	2023-06-23 19:58:32.616257316 +0200
@@ -435,11 +435,11 @@
       )
 endif()
 set(CMAKEPACKAGE_INSTALL_DIR
-    "${CMAKE_INSTALL_DATADIR}/eigen3/cmake"
+    "${CMAKE_INSTALL_LIBDIR}/cmake/eigen3"
     CACHE PATH "The directory relative to CMAKE_INSTALL_PREFIX where Eigen3Config.cmake is installed"
     )
 set(PKGCONFIG_INSTALL_DIR
-    "${CMAKE_INSTALL_DATADIR}/pkgconfig"
+    "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
     CACHE PATH "The directory relative to CMAKE_INSTALL_PREFIX where eigen3.pc is installed"
     )
 
