diff --git a/src/kleo/keyresolvercore.cpp b/src/kleo/keyresolvercore.cpp
index b25f69b4ec674023cf11a250d26b70915879c74d..13786b9fbeab5e3264b520e57e89f2d953b8786f 100644
--- a/src/kleo/keyresolvercore.cpp
+++ b/src/kleo/keyresolvercore.cpp
@@ -26,6 +26,7 @@
 #include <libkleo/gnupg.h>
 #include <libkleo/keycache.h>
 
+#include "kleo/debug.h"
 #include <libkleo_debug.h>
 
 #include <gpgme++/key.h>
@@ -36,16 +37,6 @@ using namespace GpgME;
 namespace
 {
 
-QDebug operator<<(QDebug debug, const GpgME::Key &key)
-{
-    if (key.isNull()) {
-        debug << "Null";
-    } else {
-        debug << Formatting::summaryLine(key);
-    }
-    return debug.maybeSpace();
-}
-
 static inline bool ValidEncryptionKey(const Key &key)
 {
     if (key.isNull() || key.isRevoked() || key.isExpired() || key.isDisabled() || !key.canEncrypt()) {
