diff --git a/src/kile.cpp b/src/kile.cpp
index a624737f..951bf333 100644
--- a/src/kile.cpp
+++ b/src/kile.cpp
@@ -404,6 +404,12 @@ Kile::Kile(bool allowRestore, QWidget *parent)
     // version 3.0 regarding the newly introduced live preview feature
     const QString& lastVersionRunFor = KileConfig::systemCheckLastVersionRunForAtStartUp();
     if(lastVersionRunFor.isEmpty() || compareVersionStrings(lastVersionRunFor, "2.9.91") < 0) {
+
+        // work around the problem that Sonnet's language auto-detection feature doesn't work
+        // together with KatePart (as of 08 November 2019)
+        QSettings settings(QStringLiteral("KDE"), QStringLiteral("Sonnet"));
+        settings.setValue(QStringLiteral("autodetectLanguage"), false);
+
         slotPerformCheck();
         KileConfig::setSystemCheckLastVersionRunForAtStartUp(kileFullVersion);
     }
