diff --git a/chromium/tools/typescript/validate_tsconfig.py b/chromium/tools/typescript/validate_tsconfig.py
index a31d2d8e143..814c463d7cb 100644
--- a/src/3rdparty/chromium/tools/typescript/validate_tsconfig.py
+++ b/src/3rdparty/chromium/tools/typescript/validate_tsconfig.py
@@ -270,10 +270,15 @@ def validateDefinitionDeps(definitions_files, target_path, gen_dir,
   for f in definitions_files:
     # File path relative to the current working directory.
     f_from_cwd = os.path.relpath(f, _CWD).replace('\\', '/')
+    # We need to resolve the paths because KDE Craft
+    # uses symlinks to not hit the Windows path length issue
+    _CWD_RESOLVED = pathlib.Path(_CWD).resolve()
+    f_from_cwd_resolved = os.path.relpath(f, _CWD_RESOLVED).replace('\\', '/')
     is_gen_file = f_from_cwd.startswith(gen_dir_from_build)
     f_from_gen = os.path.relpath(f, gen_dir).replace('\\', '/')
     if not is_gen_file and f_from_gen not in definitions_normalized and \
-        not any(f_from_cwd.startswith(exception) for exception in exceptions):
+        not any(f_from_cwd.startswith(exception) for exception in exceptions) \
+          not any(f_from_cwd_resolved.startswith(exception) for exception in exceptions):
       missing_inputs.append(
           os.path.relpath(f_from_cwd, _SRC_DIR).replace('\\', '/'))
 
