diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user fftw-3.3.8.orig/CMakeLists.txt fftw-3.3.8/CMakeLists.txt
--- fftw-3.3.8.orig/CMakeLists.txt	2018-05-24 14:03:22.000000000 +0200
+++ fftw-3.3.8/CMakeLists.txt	2020-04-29 10:07:36.000000000 +0200
@@ -412,6 +412,12 @@
          DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
          COMPONENT Development)
 
+install (FILES
+          ${CMAKE_CURRENT_BINARY_DIR}/fftw${PREC_SUFFIX}.pc
+         DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
+         RENAME fftw3${PREC_SUFFIX}.pc
+         COMPONENT Development)
+
 # cmake file
 set (FFTW3_LIBRARIES "FFTW3::${fftw3_lib}")
 configure_file (FFTW3Config.cmake.in FFTW3${PREC_SUFFIX}Config.cmake @ONLY)
