diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' perl-5.28.0\win32.orig/Makefile perl-5.28.0\win32/Makefile
--- perl-5.28.0\win32.orig/Makefile	2018-10-02 16:12:05.312581000 +0200
+++ perl-5.28.0\Makefile	2018-10-02 16:11:49.836494000 +0200
@@ -112,7 +112,7 @@
 # uncomment exactly one of the following
 #
 # Visual C++ 6.0 (aka Visual C++ 98)
-CCTYPE		= MSVC60
+# CCTYPE		= MSVC60
 # Visual C++ .NET 2002/2003 (aka Visual C++ 7.0/7.1) (full version)
 #CCTYPE		= MSVC70
 # Visual C++ Toolkit 2003 (aka Visual C++ 7.1) (free command-line tools)
