diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' perl-5.28.0\win32.orig/GNUmakefile perl-5.28.0\GNUmakefile
--- perl-5.28.0\win32.orig/GNUmakefile	2018-05-25 07:04:45.000000000 +0200
+++ perl-5.28.0\GNUmakefile	2018-11-29 13:41:42.553705600 +0100
@@ -47,7 +47,7 @@
 # newly built perl.
 #
 INST_DRV := c:
-INST_TOP := $(INST_DRV)\perl
+INST_TOP	= $(CRAFT_DESTDIR)
 
 #
 # Uncomment if you want to build a 32-bit Perl using a 32-bit compiler
@@ -148,14 +148,14 @@
 # Comment this out if you want the legacy default behavior of including '.' at
 # the end of @INC.
 #
-DEFAULT_INC_EXCLUDES_DOT := define
+#DEFAULT_INC_EXCLUDES_DOT := define
 
 #
 # Uncomment this if you want to disable looking up values from
 # HKEY_CURRENT_USER\Software\Perl and HKEY_LOCAL_MACHINE\Software\Perl in
 # the Registry.
 #
-#USE_NO_REGISTRY := define
+USE_NO_REGISTRY := define
 
 #
 # uncomment exactly one of the following
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' perl-5.28.0\win32.orig/Makefile perl-5.28.0\Makefile
--- perl-5.28.0\win32.orig/Makefile	2018-05-21 15:22:09.000000000 +0200
+++ perl-5.28.0\Makefile	2018-11-29 13:42:33.730958500 +0100
@@ -21,13 +21,13 @@
 # newly built perl.
 #
 INST_DRV	= c:
-INST_TOP	= $(INST_DRV)\perl
+INST_TOP	= $(CRAFT_DESTDIR)
 
 #
 # Uncomment if you want to build a 32-bit Perl using a 32-bit compiler
 # on a 64-bit version of Windows.
 #
-#WIN64		= undef
+WIN64		= $(CRAFT_WIN64)
 
 #
 # Comment this out if you DON'T want your perl installation to be versioned.
@@ -99,20 +99,20 @@
 # Comment this out if you want the legacy default behavior of including '.' at
 # the end of @INC.
 #
-DEFAULT_INC_EXCLUDES_DOT = define
+# DEFAULT_INC_EXCLUDES_DOT = define
 
 #
 # Uncomment this if you want to disable looking up values from
 # HKEY_CURRENT_USER\Software\Perl and HKEY_LOCAL_MACHINE\Software\Perl in
 # the Registry.
 #
-#USE_NO_REGISTRY = define
+USE_NO_REGISTRY = define
 
 #
 # uncomment exactly one of the following
 #
 # Visual C++ 6.0 (aka Visual C++ 98)
-CCTYPE		= MSVC60
+# CCTYPE		= MSVC60
 # Visual C++ .NET 2002/2003 (aka Visual C++ 7.0/7.1) (full version)
 #CCTYPE		= MSVC70
 # Visual C++ Toolkit 2003 (aka Visual C++ 7.1) (free command-line tools)
@@ -959,7 +959,7 @@
 
 $(GLOBEXE) : perlglob$(o)
 	$(LINK32) $(BLINK_FLAGS) $(LIBFILES) -out:$@ \
-	    perlglob$(o) setargv$(o)
+	    perlglob$(o) $(SETARGV_OBJ)
 	$(EMBED_EXE_MANI)
 
 perlglob$(o)  : perlglob.c
