diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' perl-5.29.3\win32.orig/Makefile perl-5.29.3\win32/Makefile
--- perl-5.29.3\win32.orig/Makefile	2018-08-26 18:55:18.000000000 +0200
+++ perl-5.29.3\Makefile	2018-10-01 16:36:20.126345700 +0200
@@ -21,7 +21,7 @@
 # newly built perl.
 #
 INST_DRV	= c:
-INST_TOP	= $(INST_DRV)\perl
+INST_TOP	= $(CRAFT_DESTDIR)
 
 #
 # Uncomment if you want to build a 32-bit Perl using a 32-bit compiler
