diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' perl-5.28.0\win32.orig/GNUmakefile perl-5.28.0\win32/GNUmakefile
--- perl-5.28.0\win32.orig/GNUmakefile	2018-05-25 07:04:45.000000000 +0200
+++ perl-5.28.0\GNUmakefile	2018-11-28 14:21:44.130066100 +0100
@@ -47,7 +47,7 @@
 # newly built perl.
 #
 INST_DRV := c:
-INST_TOP := $(INST_DRV)\perl
+INST_TOP	= $(CRAFT_DESTDIR)
 
 #
 # Uncomment if you want to build a 32-bit Perl using a 32-bit compiler
@@ -148,14 +148,14 @@
 # Comment this out if you want the legacy default behavior of including '.' at
 # the end of @INC.
 #
-DEFAULT_INC_EXCLUDES_DOT := define
+#DEFAULT_INC_EXCLUDES_DOT := define
 
 #
 # Uncomment this if you want to disable looking up values from
 # HKEY_CURRENT_USER\Software\Perl and HKEY_LOCAL_MACHINE\Software\Perl in
 # the Registry.
 #
-#USE_NO_REGISTRY := define
+USE_NO_REGISTRY := define
 
 #
 # uncomment exactly one of the following
