diff -Nrub -x '*~' -x '*.rej' -x '*.orig' -x '*.o' -x '*.pyc' -x CMakeLists.txt.user libheif-1.18.2.orig/CMakeLists.txt libheif-1.18.2/CMakeLists.txt
--- libheif-1.18.2.orig/CMakeLists.txt	2024-08-07 21:21:18.000000000 +0200
+++ libheif-1.18.2/CMakeLists.txt	2025-04-15 10:31:09.106142600 +0200
@@ -415,14 +415,6 @@
 
 list(JOIN REQUIRES_PRIVATE " " REQUIRES_PRIVATE)
 
-include(CheckCXXSymbolExists)
-check_cxx_symbol_exists(_LIBCPP_VERSION cstdlib HAVE_LIBCPP)
-if(HAVE_LIBCPP)
-  set(LIBS_PRIVATE "-lc++")
-else()
-  set(LIBS_PRIVATE "-lstdc++")
-endif()
-
 configure_file(libheif.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libheif.pc @ONLY)
 
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libheif.pc
