From 9685e8b32b432ec8575ac14e79b47a56f214236e Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Thu, 24 Jun 2021 21:55:09 +0200
Subject: [PATCH] Fix build on Android

---
 src/kweathercore_p.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/kweathercore_p.h b/src/kweathercore_p.h
index 4d784a9..633391b 100644
--- a/src/kweathercore_p.h
+++ b/src/kweathercore_p.h
@@ -36,14 +36,14 @@ static constexpr auto toFixedString = [](double num) {
 };
 inline QDir getCacheDirectory(double latitude, double longitude)
 {
-#ifdef __ANDROID__
+#ifdef Q_OS_ANDROID
     QDir dir(QStandardPaths::writableLocation(QStandardPaths::CacheLocation) +
              QStringLiteral("/cache/") + toFixedString(latitude) +
              QStringLiteral("/") + toFixedString(longitude));
     if (!dir.exists())
         dir.mkpath(QStringLiteral("."));
     return dir;
-#endif
+#else
     QDir dir(QStandardPaths::writableLocation(QStandardPaths::GenericCacheLocation)
              + QStringLiteral("/kweather/cache/")
              + toFixedString(latitude)
@@ -51,6 +51,7 @@ inline QDir getCacheDirectory(double latitude, double longitude)
     if (!dir.exists())
         dir.mkpath(QStringLiteral("."));
     return dir;
+#endif
 }
 static constexpr auto urgencyStrToEnum = [](const QString &str) {
     if (str == QStringLiteral("Immediate"))
-- 
2.32.0

