commit b98ab6cbd6ccbc7b8d979d795129d6b806cb85f1
Author: Volker Krause <vkrause@kde.org>
Date:   Sat Jul 8 07:24:50 2023 +0200

    Fix binary stripping in APKs with Qt 5.15.10

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8729f54..adfb548 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,7 @@ project(kongress VERSION ${RELEASE_SERVICE_VERSION})
 set(KF_MIN_VERSION "5.95.0")
 set(QT_MIN_VERSION "5.15.2")
 if (ANDROID)
-    set(QT_MIN_VERSION "5.15.8")
+    set(QT_MIN_VERSION "5.15.10")
 endif()
 set (KDE_COMPILERSETTINGS_LEVEL "5.82.0")
 
diff --git a/android/build.gradle b/android/build.gradle
index 5383f4f..309f582 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -47,6 +47,7 @@ android {
     compileSdkVersion androidCompileSdkVersion.toInteger()
 
     buildToolsVersion androidBuildToolsVersion
+    ndkVersion androidNdkVersion
 
     sourceSets {
         main {
