From 0b0dcad3c5664f8499eaf9c9385a8120f178b451 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20K=C3=BCnzel?= <jk.kdedev@smartlab.uber.space>
Date: Wed, 3 Jan 2024 10:11:24 +0000
Subject: [PATCH] [tests] Fix undefined reference to qMain(int, char**)

---
 autotests/fbrecurring.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/autotests/fbrecurring.cpp b/autotests/fbrecurring.cpp
index fa68f1bf3..026db57b8 100644
--- a/autotests/fbrecurring.cpp
+++ b/autotests/fbrecurring.cpp
@@ -19,7 +19,7 @@
 using namespace KCalendarCore;
 using namespace std;
 
-int main()
+int main(int argc, char **argv)
 {
     ICalFormat f;
 
-- 
GitLab

