diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' gwenhywfar-4.19.0.orig/configure gwenhywfar-4.19.0/configure
--- gwenhywfar-4.19.0.orig/configure	2017-07-30 15:20:32.000000000 +0100
+++ gwenhywfar-4.19.0/configure	2018-02-18 11:55:48.614397600 +0100
@@ -22480,6 +22480,7 @@
 
     # Get Qt version from qmake
     QT_DIR=`$QT_QMAKE --version | grep -o -E /.+`
+    QT_LIBS="-L`$QT_QMAKE --version | grep -o -E .:.+` -lQt5Widgets -lQt5Gui -lQt5Core"
 
     # All variables are defined, report the result
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $have_qt: