diff --git a/data/meson.build b/data/meson.build
index 09ed7a9..9fa8f91 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,14 +1,9 @@
 
 install_man('update-mime-database.1')
 
-freedesktop_org_xml = i18n.merge_file(
-    input: 'freedesktop.org.xml.in',
-    output: 'freedesktop.org.xml',
-    data_dirs: '.',
-    po_dir: '../po',
-    type: 'xml',
-    install: true,
-    install_dir: get_option('datadir') / 'mime' / 'packages',
+install_data('freedesktop.org.xml.in',
+    rename: 'freedesktop.org.xml',
+    install_dir: get_option('datadir') / 'mime' / 'packages'
 )
 
 install_data(
