From 1187fe9a95a04da5524d1c804e89fd260ab2515a Mon Sep 17 00:00:00 2001
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 59bf65f..ef45463 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,6 +55,7 @@ set (SRC_SOURCES
          lib/ofx_container_security.cpp
          lib/ofx_container_statement.cpp
          lib/ofx_container_account.cpp
+         lib/ofx_container_position.cpp
          lib/ofx_container_transaction.cpp
          lib/ofx_containers_misc.cpp
          lib/ofx_request.cpp
-- 
2.36.1

