diff --git a/lib/ofx_preproc.cpp b/lib/ofx_preproc.cpp
index 93f2f0b..692b040 100644
--- a/lib/ofx_preproc.cpp
+++ b/lib/ofx_preproc.cpp
@@ -310,8 +310,12 @@ int ofx_proc_file(LibofxContextPtr ctx, const char * p_filename)
             const char* inchar = s_buffer.c_str();
             char * outchar = iconv_buffer;
             int iconv_retval = iconv (conversion_descriptor,
-                                      const_cast<char**>(&inchar), &inbytesleft,
-                                      &outchar, &outbytesleft);
+#ifdef _WIN32 //Win32's iconv implementation takes a const char** inchar
+                                      &inchar,
+#else
+                                      const_cast<char**>(&inchar),
+#endif
+                                      &inbytesleft, &outchar, &outbytesleft);
             if (iconv_retval == -1)
             {
               message_out(ERROR, "ofx_proc_file(): Iconv conversion error");
-- 
2.36.1

