From bfbfee78bdbeb8775b8a405dc3d4081a2245a036 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dawid=20Wro=CC=81bel?= <me@dawidwrobel.com>
Date: Wed, 29 Jun 2022 15:24:15 +0200
Subject: [PATCH] Don't deconst iconv input buffer under Windows

Fixes #51
---
 lib/ofx_preproc.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/ofx_preproc.cpp b/lib/ofx_preproc.cpp
index 75ce5f9..53ead78 100644
--- a/lib/ofx_preproc.cpp
+++ b/lib/ofx_preproc.cpp
@@ -308,7 +308,11 @@ int ofx_proc_file(LibofxContextPtr ctx, const char * p_filename)
             const char* inchar = s_buffer.c_str();
             char * outchar = iconv_buffer;
             int iconv_retval = iconv (conversion_descriptor,
+#ifdef _WIN32 //Win32's iconv implementation takes a const char** inchar
+                                      &inchar, &inbytesleft,
+#else
                                       const_cast<char**>(&inchar), &inbytesleft,
+#endif
                                       &outchar, &outbytesleft);
             if (iconv_retval == -1)
             {