diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' libofx-0.9.13.orig/lib/ofx_preproc.cpp libofx-0.9.13/lib/ofx_preproc.cpp
--- libofx-0.9.13.orig/lib/ofx_preproc.cpp	2018-05-02 20:39:38.000000000 +0100
+++ libofx-0.9.13/lib/ofx_preproc.cpp	2018-05-05 09:04:35.646746800 +0100
@@ -309,7 +309,7 @@
 #if defined(__sun) || defined(__NetBSD__)
             const char * inchar = (const char *)s_buffer.c_str();
 #else
-            char * inchar = (char *)s_buffer.c_str();
+            const char * inchar = (char *)s_buffer.c_str();
 #endif
             char * outchar = iconv_buffer;
             int iconv_retval = iconv (conversion_descriptor,
