diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' libofx-0.9.12.orig/lib/getopt.h libofx-0.9.12/lib/getopt.h
--- libofx-0.9.12.orig/lib/getopt.h	2018-04-12 15:18:47.000000000 +0200
+++ libofx-0.9.12/lib/getopt.h	2018-04-12 15:18:19.000000000 +0200
@@ -1,7 +1,7 @@
 #ifndef _getopt_h
 #define _getopt_h

-#ifndef _MSC_VER
+#if !defined(_MSC_VER) && !(defined(__APPLE__) || defined(__MACOSX__))
 #include <getopt.h>
 #else
 struct option {
