diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d6fa24e..68cd97e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -136,7 +136,7 @@ target_link_libraries(KIconEnginePlugin
 
 # install in an extra directory we will add to the plugin path later just for this one engine
 # we can overwrite the system one with that trick
-install(TARGETS KIconEnginePlugin DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/kiconthemes6/iconengines)
+install(TARGETS KIconEnginePlugin DESTINATION ${KDE_INSTALL_QTPLUGINDIR}/iconengines)
 
 ecm_qt_install_logging_categories(
     EXPORT KICONTHEMES
