diff --git a/src/kicontheme.cpp b/src/kicontheme.cpp
index 779c588..d1b1b66 100644
--- a/src/kicontheme.cpp
+++ b/src/kicontheme.cpp
@@ -53,6 +53,16 @@ static bool initThemeUsed = false;
 // startup function to set theme once the app got constructed
 static void initThemeHelper()
 {
+    // make sure we add application install path to search path, for e.g. bundles on Windows
+    if (initThemeUsed) {
+        // do that similar to QCoreApplicationPrivate::appendApplicationPathToLibraryPaths() with minimal extra API use
+        QString path = QCoreApplication::applicationFilePath();
+        path.truncate(path.lastIndexOf(QLatin1Char('/')));
+        if (const QString ourPath = path + QStringLiteral("/kiconthemes6"); QFile::exists(ourPath)) {
+            QCoreApplication::addLibraryPath(ourPath);
+        }
+    }
+
     // Makes sure the icon theme fallback is set to breeze or one of its
     // variants. Most of our apps use "lots" of icons that most of the times
     // are only available with breeze, we still honour the user icon theme
