diff -Nrub -x *~ -x *\.rej -x *\.orig -x *\.o -x *\.pyc -x CMakeLists.txt.user brotli-1.0.9.orig/CMakeLists.txt brotli-1.0.9/CMakeLists.txt
--- brotli-1.0.9.orig/CMakeLists.txt	2020-08-27 16:12:55
+++ brotli-1.0.9/CMakeLists.txt	2023-07-07 20:07:06
@@ -2,7 +2,7 @@
 # several CI services, such as Travis and Drone, use it.  Solaris 11
 # has 2.8.6, and it's not difficult to support if you already have to
 # support 2.8.7.
-cmake_minimum_required(VERSION 2.8.6)
+cmake_minimum_required(VERSION 3.15)
 
 project(brotli C)
 
