From 743f2fc34165201273cd4e69afd29d096e1e923f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20K=C3=BCnzel?= <jk.kdedev@smartlab.uber.space>
Date: Sun, 12 Nov 2023 17:48:20 +0000
Subject: [PATCH] Remove unused include to fix build without X11

---
 src/kstatusnotifieritemdbus_p.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/kstatusnotifieritemdbus_p.cpp b/src/kstatusnotifieritemdbus_p.cpp
index f01af5d..389ee99 100644
--- a/src/kstatusnotifieritemdbus_p.cpp
+++ b/src/kstatusnotifieritemdbus_p.cpp
@@ -12,7 +12,6 @@
 
 #include <QMenu>
 
-#include <KWindowInfo>
 #include <kwindowsystem.h>
 
 #include "statusnotifierwatcher_interface.h"
-- 
GitLab

