From 755ffb7bc9ddc4d0d26528169ff9660738e31f6f Mon Sep 17 00:00:00 2001
From: Hannah von Reth <hannah.vonreth@owncloud.com>
Date: Fri, 8 Jul 2022 14:29:46 +0200
Subject: [PATCH] Allow to build clazy on mac

---
 CMakeLists.txt | 12 +++++++++++-
 clazy.cmake    |  4 ++--
 2 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 434a57c9..904e3b8c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,7 +81,7 @@ endif()
 
 # Look for std::regex support
 message("Looking for std::regex support...")
-try_run(REGEX_RUN_RESULT COMPILE_RESULT ${CMAKE_BINARY_DIR} ${CMAKE_CURRENT_LIST_DIR}/.cmake_has_regex_test.cpp)
+try_run(REGEX_RUN_RESULT COMPILE_RESULT ${CMAKE_BINARY_DIR} ${CMAKE_CURRENT_LIST_DIR}/.cmake_has_regex_test.cpp CXX_STANDARD 17 CXX_STANDARD_REQUIRED true)
 
 if(NOT REGEX_RUN_RESULT EQUAL 0)
   message("Using boost::regex instead of std::regex")
@@ -211,11 +211,21 @@ if (NOT CLAZY_BUILD_WITH_CLANG)
   )
 
   if(NOT WIN32)
+    # readlink is used to resolve canonical path in the clazy shell script
+    set(READLINK_ARGS "-f")
     if(APPLE)
       find_program(READLINK_CMD greadlink)
     else()
       find_program(READLINK_CMD readlink)
     endif()
+    set(READLINK_CMD NO)
+    if(NOT READLINK_CMD)
+      find_package(Python3 COMPONENTS Interpreter)
+      if(TARGET Python3::Interpreter)
+        set(READLINK_CMD "${Python3_EXECUTABLE}")
+        set(READLINK_ARGS "-c \"from pathlib import Path;import sys; print(Path(sys.argv[1]).resolve())\"")
+      endif()
+    endif()
     if(NOT READLINK_CMD)
       message(FATAL_ERROR "Could not find a proper readlink.  On Mac OSX you should install coreutils using homebrew in order to use the GNU readlink")
     endif()
diff --git a/clazy.cmake b/clazy.cmake
index 96e14640..b62dd30c 100644
--- a/clazy.cmake
+++ b/clazy.cmake
@@ -1,7 +1,7 @@
 #!/usr/bin/env sh
 
-libdir=$("@READLINK_CMD@" -f "$(dirname ${0})/@BIN_RELATIVE_LIBDIR@")
-sharedir=$("@READLINK_CMD@" -f "$(dirname ${0})/@BIN_RELATIVE_SHAREDIR@")
+libdir=$("@READLINK_CMD@" @READLINK_ARGS@ "$(dirname ${0})/@BIN_RELATIVE_LIBDIR@")
+sharedir=$("@READLINK_CMD@" @READLINK_ARGS@ "$(dirname ${0})/@BIN_RELATIVE_SHAREDIR@")
 
 HELP() {
   echo "Usage: `basename $0` [options] [clang++-options]"
-- 
2.32.1 (Apple Git-133)

