diff -Nrub -x '*~' -x '*.rej' -x '*.orig' -x '*.o' -x '*.pyc' -x CMakeLists.txt.user -x .idea -x .clang-format kdegraphics-mobipocket-25.12.0.orig/tools/mobidump.cpp kdegraphics-mobipocket-25.12.0/tools/mobidump.cpp
--- kdegraphics-mobipocket-25.12.0.orig/tools/mobidump.cpp	2025-11-16 23:57:58.000000000 +0100
+++ kdegraphics-mobipocket-25.12.0/tools/mobidump.cpp	2025-12-11 16:14:13.537237800 +0100
@@ -53,4 +53,5 @@
         out << "\"" << doc.text() << "\"" << Qt::endl;
     }
     out << "===" << Qt::endl << Qt::endl;
+    return 0;
 }
