diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user gwenhywfar-5.10.2.orig/src/sio/syncio_tls.c gwenhywfar-5.10.2/src/sio/syncio_tls.c
--- gwenhywfar-5.10.2.orig/src/sio/syncio_tls.c	2023-07-27 21:48:18.000000000 +0000
+++ gwenhywfar-5.10.2/src/sio/syncio_tls.c	2023-10-24 18:47:54.106305270 +0000
@@ -530,7 +530,7 @@
     /* try to find OpenSSL certificates */
 # ifdef OS_WIN32
     if (trustFileSet==0) {
-      char defaultPath[2*MAX_PATH+1];
+      char defaultPath[2*_MAX_PATH+1];
       const char *defaultFile = "ca-bundle.crt";
       GWEN_STRINGLIST *paths;
       GWEN_BUFFER *nbuf;
