--- tools/xmlmerge//Makefile.in.ori	2021-12-30 18:37:51.440451404 +0100
+++ tools/xmlmerge//Makefile.in	2021-12-30 18:38:42.920431560 +0100
@@ -90,7 +90,8 @@
 build_triplet = @build@
 host_triplet = @host@
 bin_PROGRAMS = xmlmerge$(EXEEXT)
-@IS_WINDOWS_TRUE@am__append_1 = $(PROG_LIBRARIES)
+@IS_WINDOWS_TRUE@am__append_1 = -mconsole
+@IS_WINDOWS_TRUE@am__append_2 = $(PROG_LIBRARIES)
 subdir = tools/xmlmerge
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/acx_compile_warn.m4 \
@@ -475,8 +476,8 @@
 noinst_HEADERS = args.h
 xmlmerge_SOURCES = args.c main.c
 xmlmerge_LDADD = $(top_builddir)/src/$(gwenhywfar_internal_libname) \
-	$(am__append_1)
-@IS_WINDOWS_TRUE@xmlmerge_LDFLAGS = -mconsole
+	$(am__append_2)
+xmlmerge_LDFLAGS = @i18n_libs@ $(am__append_1)
 all: all-am
 
 .SUFFIXES:
--- tools/typemaker//Makefile.in.ori	2021-12-30 18:37:51.444451403 +0100
+++ tools/typemaker//Makefile.in	2021-12-30 18:38:42.812431613 +0100
@@ -90,7 +90,8 @@
 build_triplet = @build@
 host_triplet = @host@
 bin_PROGRAMS = typemaker$(EXEEXT)
-@IS_WINDOWS_TRUE@am__append_1 = $(PROG_LIBRARIES)
+@IS_WINDOWS_TRUE@am__append_1 = -mconsole
+@IS_WINDOWS_TRUE@am__append_2 = $(PROG_LIBRARIES)
 subdir = tools/typemaker
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/acx_compile_warn.m4 \
@@ -477,8 +478,8 @@
 noinst_HEADERS = args.h typemaker_p.h
 typemaker_SOURCES = args.c main.c phc.c lhc.c ahc.c code_c.c
 typemaker_LDADD = $(top_builddir)/src/$(gwenhywfar_internal_libname) \
-	$(am__append_1)
-@IS_WINDOWS_TRUE@typemaker_LDFLAGS = -mconsole
+	$(am__append_2)
+typemaker_LDFLAGS = @i18n_libs@ $(am__append_1)
 all: all-am
 
 .SUFFIXES:
