diff -Nrub -x '*~' -x '*.rej' -x '*.orig' -x '*.o' -x '*.pyc' -x CMakeLists.txt.user libssh2-1.11.1.orig/libssh2.pc.in libssh2-1.11.1/libssh2.pc.in
--- libssh2-1.11.1.orig/libssh2.pc.in	2024-10-16 10:03:21.000000000 +0200
+++ libssh2-1.11.1/libssh2.pc.in	2025-10-27 16:54:22.276059200 +0100
@@ -16,6 +16,6 @@
 Version: @LIBSSH2_VERSION@
 Requires: @LIBSSH2_PC_REQUIRES@
 Requires.private: @LIBSSH2_PC_REQUIRES_PRIVATE@
-Libs: -L${libdir} -lssh2 @LIBSSH2_PC_LIBS@
+Libs: -L${libdir} -llibssh2 @LIBSSH2_PC_LIBS@
 Libs.private: @LIBSSH2_PC_LIBS_PRIVATE@
 Cflags: -I${includedir}
