diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libssh2-1.10.0.orig/libssh2.pc.in libssh2-1.10.0/libssh2.pc.in
--- libssh2-1.10.0.orig/libssh2.pc.in	2016-02-24 23:44:12.000000000 +0100
+++ libssh2-1.10.0/libssh2.pc.in	2022-10-26 10:52:01.664608700 +0200
@@ -12,6 +12,6 @@
 Description: Library for SSH-based communication
 Version: @LIBSSH2VER@
 Requires.private: @LIBSREQUIRED@
-Libs: -L${libdir} -lssh2 @LIBS@
+Libs: -L${libdir} -lssh2
 Libs.private: @LIBS@
 Cflags: -I${includedir}
