diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libssh2-1.11.0.orig/libssh2.pc.in libssh2-1.11.0/libssh2.pc.in
--- libssh2-1.11.0.orig/libssh2.pc.in	2016-02-24 23:44:12.000000000 +0100
+++ libssh2-1.11.0/libssh2.pc.in	2023-07-02 12:04:15.367501100 +0200
@@ -12,6 +12,6 @@
 Description: Library for SSH-based communication
 Version: @LIBSSH2VER@
 Requires.private: @LIBSREQUIRED@
-Libs: -L${libdir} -lssh2 @LIBS@
+Libs: -L${libdir} -llibssh2 @LIBS@
 Libs.private: @LIBS@
 Cflags: -I${includedir}
