diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' sqlite-amalgamation-3240000.orig/sqlite.pc.in sqlite-amalgamation-3240000/sqlite.pc.in
--- sqlite-amalgamation-3240000.orig/sqlite.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ sqlite-amalgamation-3240000/sqlite.pc.in	2017-12-21 20:31:28.000000000 +0100
@@ -0,0 +1,13 @@
+# Package Information for pkg-config
+
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: SQLite
+Description: SQL database engine
+Version: @RELEASE@
+Libs: -L${libdir} -lsqlite
+Libs.private: @LIBS@
+Cflags: -I${includedir}
