diff --git a/cmake/FindLibIntl.cmake b/cmake/FindLibIntl.cmake
index 6ec18a9..a86b3a6 100644
--- a/cmake/FindLibIntl.cmake
+++ b/cmake/FindLibIntl.cmake
@@ -50,7 +50,6 @@ check_cxx_symbol_exists(dngettext libintl.h LibIntl_SYMBOL_FOUND)
 include(FindPackageHandleStandardArgs)
 if(LibIntl_SYMBOL_FOUND)
     message(STATUS "libintl is part of libc, no extra library is required.")
-    set(LibIntl_LIBRARIES "")
     if(LibIntl_INCLUDE_DIRS)
         find_package_handle_standard_args(LibIntl REQUIRED_VARS LibIntl_INCLUDE_DIRS)
     else()
