diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' openal-soft-openal-soft-1.19.1.orig/cmake/FindDSound.cmake openal-soft-openal-soft-1.19.1/cmake/FindDSound.cmake
--- openal-soft-openal-soft-1.19.1.orig/cmake/FindDSound.cmake	2019-10-19 23:14:49.980324900 +0200
+++ openal-soft-openal-soft-1.19.1/cmake/FindDSound.cmake	2019-10-19 23:14:23.701668500 +0200
@@ -14,19 +14,20 @@
     get_windowssdk_library_dirs(${WINDOWSSDK_PREFERRED_DIR} WINSDK_LIB_DIRS)
     get_windowssdk_include_dirs(${WINDOWSSDK_PREFERRED_DIR} WINSDK_INCLUDE_DIRS)
   endif()
+    set(SEARCH_HINTS "${DXSDK_DIR}" ${WINSDK_INCLUDE_DIRS})
 endif()
 
 # DSOUND_INCLUDE_DIR
 find_path(DSOUND_INCLUDE_DIR
           NAMES "dsound.h" 
-          PATHS "${DXSDK_DIR}" ${WINSDK_INCLUDE_DIRS} 
+          PATHS ${SEARCH_HINTS}
           PATH_SUFFIXES include
           DOC "The DirectSound include directory")
 
 # DSOUND_LIBRARY
 find_library(DSOUND_LIBRARY
              NAMES dsound
-             PATHS "${DXSDK_DIR}" ${WINSDK_LIB_DIRS}
+             PATHS ${SEARCH_HINTS}
              PATH_SUFFIXES lib lib/x86 lib/x64
              DOC "The DirectSound library")
 
