diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' openal-soft-openal-soft-1.19.1.orig/cmake/FindDSound.cmake openal-soft-openal-soft-1.19.1/cmake/FindDSound.cmake
--- openal-soft-openal-soft-1.19.1.orig/cmake/FindDSound.cmake	2018-10-12 00:05:31.000000000 +0200
+++ openal-soft-openal-soft-1.19.1/cmake/FindDSound.cmake	2019-10-17 16:04:36.707549100 +0200
@@ -8,7 +8,7 @@
 #   DSOUND_LIBRARY     - the dsound library
 #
 
-if (WIN32)
+if (MSVC)
   include(FindWindowsSDK)
   if (WINDOWSSDK_FOUND)
     get_windowssdk_library_dirs(${WINDOWSSDK_PREFERRED_DIR} WINSDK_LIB_DIRS)
