From e06579df315c53363a0e97cafec1b37f360c0e45 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Sun, 20 Mar 2022 20:16:04 +0100
Subject: [PATCH] Port to target-based ecm_add_app_icon

---
 src/CMakeLists.txt | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b97a5d7c0..104e3a48b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -342,11 +342,7 @@ target_sources(dolphin PRIVATE
 
 # Sets the icon on Windows and OSX
 file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/*system-file-manager.png")
-ecm_add_app_icon(dolphin_APPICON_SRCS ICONS ${ICONS_SRCS})
-# ecm_add_app_icon 5.83 will take target as arg, use that once it is min req.
-target_sources(dolphin PRIVATE
-    ${dolphin_APPICON_SRCS}
-)
+ecm_add_app_icon(dolphin ICONS ${ICONS_SRCS})
 
 target_link_libraries(dolphin
     PRIVATE
-- 
2.35.1

