diff --git a/configure.json b/configure.json
new file mode 100644
index 0000000..10fe83d
--- /dev/null
+++ b/configure.json
@@ -0,0 +1,5 @@
+{
+    "subconfigs": [
+        "src/svg"
+    ]
+}
diff --git a/src/svg/configure.json b/src/svg/configure.json
new file mode 100644
index 0000000..c525fad
--- /dev/null
+++ b/src/svg/configure.json
@@ -0,0 +1,31 @@
+{
+    "module": "svg",
+    "depends": [
+        "gui"
+    ],
+
+    "features": {
+        "qsvgwidget": {
+            "label": "QSvgWidget",
+            "purpose": "Widget for SVG display",
+            "section": "SVG",
+            "output": [ "privateFeature" ]
+        },
+        "qgraphicssvgitem": {
+            "label": "QGraphicsSvgItem",
+            "purpose": "QGraphicsView item for SVG display",
+            "section": "SVG",
+            "output": [ "privateFeature" ]
+        }
+    },
+
+    "summary": [
+        {
+            "section": "Qt SVG",
+            "entries": [
+                "qsvgwidget",
+                "qgraphicssvgitem"
+            ]
+        }
+    ]
+}
diff --git a/src/svg/svg.pro b/src/svg/svg.pro
index b6f17a7..8562604 100644
--- a/src/svg/svg.pro
+++ b/src/svg/svg.pro
@@ -1,6 +1,6 @@
 TARGET     = QtSvg
 QT         = core-private gui-private
-qtHaveModule(widgets): QT += widgets-private
+qtConfig(qsvgwidget)|qtConfig(qgraphicssvgitem): QT += widgets-private
 
 DEFINES   += QT_NO_USING_NAMESPACE
 msvc:equals(QT_ARCH, i386): QMAKE_LFLAGS += /BASE:0x66000000
@@ -17,12 +17,9 @@ HEADERS += \
     qsvgfont_p.h            \
     qsvgtinydocument_p.h    \
     qsvgrenderer.h          \
-    qsvgwidget.h            \
-    qgraphicssvgitem.h      \
     qsvggenerator.h \
     qtsvgglobal.h
 
-
 SOURCES += \
     qsvggraphics.cpp        \
     qsvghandler.cpp         \
@@ -32,10 +29,18 @@ SOURCES += \
     qsvgfont.cpp            \
     qsvgtinydocument.cpp    \
     qsvgrenderer.cpp        \
-    qsvgwidget.cpp          \
-    qgraphicssvgitem.cpp    \
     qsvggenerator.cpp
 
+qtConfig(qsvgwidget) {
+    HEADERS += qsvgwidget.h
+    SOURCES += qsvgwidget.cpp
+}
+
+qtConfig(qgraphicssvgitem) {
+    HEADERS += qgraphicssvgitem.h
+    SOURCES += qgraphicssvgitem.cpp
+}
+
 qtConfig(system-zlib): \
     QMAKE_USE_PRIVATE += zlib
 else: \
