diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user llvm-9.0.0.src.orig/cmake/config-ix.cmake llvm-9.0.0.src/cmake/config-ix.cmake
--- llvm-9.0.0.src.orig/cmake/config-ix.cmake	2019-08-01 10:47:45.000000000 +0200
+++ llvm-9.0.0.src/cmake/config-ix.cmake	2020-01-20 13:41:19.000000000 +0100
@@ -166,7 +166,7 @@
         else()
           include_directories(${LIBXML2_INCLUDE_DIR})
         endif()
-        set(LIBXML2_LIBS "xml2")
+        set(LIBXML2_LIBS ${LIBXML2_LIBRARIES})
       endif()
     endif()
   endif()
