diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' llvm-8.0.0.src.orig/cmake/config-ix.cmake llvm-8.0.0.src/cmake/config-ix.cmake
--- llvm-8.0.0.src.orig/cmake/config-ix.cmake	2018-11-26 01:03:39.000000000 +0100
+++ llvm-8.0.0.src/cmake/config-ix.cmake	2019-04-11 11:23:55.266187300 +0200
@@ -154,7 +154,7 @@
         else()
           include_directories(${LIBXML2_INCLUDE_DIR})
         endif()
-        set(LIBXML2_LIBS "xml2")
+        set(LIBXML2_LIBS ${LIBXML2_LIBRARIES})
       endif()
     endif()
   endif()
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' llvm-8.0.0.src.orig/lib/Testing/Support/CMakeLists.txt llvm-8.0.0.src/lib/Testing/Support/CMakeLists.txt
--- llvm-8.0.0.src.orig/lib/Testing/Support/CMakeLists.txt	2018-09-06 01:30:17.000000000 +0200
+++ llvm-8.0.0.src/lib/Testing/Support/CMakeLists.txt	2019-04-11 11:23:55.297465900 +0200
@@ -1,7 +1,8 @@
-add_definitions(-DGTEST_LANG_CXX11=1)
-add_definitions(-DGTEST_HAS_TR1_TUPLE=0)
+if (LLVM_INCLUDE_TESTS)
+    add_definitions(-DGTEST_LANG_CXX11=1)
+    add_definitions(-DGTEST_HAS_TR1_TUPLE=0)
 
-add_llvm_library(LLVMTestingSupport
+    add_llvm_library(LLVMTestingSupport
   Error.cpp
   SupportHelpers.cpp
 
@@ -14,6 +15,7 @@
   Support
   )
 
-include_directories(${LLVM_MAIN_SRC_DIR}/utils/unittest/googletest/include)
-include_directories(${LLVM_MAIN_SRC_DIR}/utils/unittest/googlemock/include)
-target_link_libraries(LLVMTestingSupport PRIVATE gtest)
+    include_directories(${LLVM_MAIN_SRC_DIR}/utils/unittest/googletest/include)
+    include_directories(${LLVM_MAIN_SRC_DIR}/utils/unittest/googlemock/include)
+    target_link_libraries(LLVMTestingSupport PRIVATE gtest)
+endif()
