diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' llvm-7.0.1.src.orig/lib/Testing/Support/CMakeLists.txt llvm-7.0.1.src/lib/Testing/Support/CMakeLists.txt
--- llvm-7.0.1.src.orig/lib/Testing/Support/CMakeLists.txt	2018-04-08 08:49:17.000000000 +0200
+++ llvm-7.0.1.src/lib/Testing/Support/CMakeLists.txt	2019-01-02 14:51:22.131390600 +0100
@@ -1,7 +1,8 @@
-add_definitions(-DGTEST_LANG_CXX11=1)
-add_definitions(-DGTEST_HAS_TR1_TUPLE=0)
+if (LLVM_INCLUDE_TESTS)
+    add_definitions(-DGTEST_LANG_CXX11=1)
+    add_definitions(-DGTEST_HAS_TR1_TUPLE=0)
 
-add_llvm_library(LLVMTestingSupport
+    add_llvm_library(LLVMTestingSupport
   Error.cpp
 
   BUILDTREE_ONLY
@@ -13,6 +14,7 @@
   Support
   )
 
-include_directories(${LLVM_MAIN_SRC_DIR}/utils/unittest/googletest/include)
-include_directories(${LLVM_MAIN_SRC_DIR}/utils/unittest/googlemock/include)
-target_link_libraries(LLVMTestingSupport PRIVATE gtest)
+    include_directories(${LLVM_MAIN_SRC_DIR}/utils/unittest/googletest/include)
+    include_directories(${LLVM_MAIN_SRC_DIR}/utils/unittest/googlemock/include)
+    target_link_libraries(LLVMTestingSupport PRIVATE gtest)
+endif()
