--- Makefile	2021-10-08 08:50:03.639166600 +0200
+++ Makefile	2021-10-08 09:14:22.209531900 +0200
@@ -33,19 +33,23 @@
 
 
 # GENERAL
-
-OS := $(shell uname -s)
-
-CC      = gcc
 CFLAGS  = -I. -O3 -Wall -fomit-frame-pointer -funroll-loops -ffast-math -c -fPIC -DPIC
-ifeq ($(OS),Darwin)
-LDFLAGS = -nostartfiles -shared -Wl,-install_name,symbolic -lc -lm
+LDFLAGS = -shared -lm
+ifeq ($(OS),Windows_NT)
+  SUFFIX = dll
 else
-LDFLAGS = -nostartfiles -shared -Wl,-Bsymbolic -lc -lm -lrt
+  SUFFIX = so
+  LDFLAGS += -nostartfiles -lm
+  OS := $(shell uname -s)
+  ifeq ($(OS),Darwin)
+    LDFLAGS += -Wl,-install_name,symbolic
+  else
+    LDFLAGS += -Wl,-Bsymbolic -lrt
+  endif
 endif
 MODULES = $(wildcard *.c)
 
-all: $(MODULES:%.c=%.so)
+all: $(MODULES:%.c=%.$(SUFFIX))
 
 # RULES TO BUILD PLUGINS FROM C CODE
 
@@ -56,19 +60,19 @@
 %.o: %.c tap_utils.h ladspa.h
 	$(CC) $(CFLAGS) $< -o $@
 
-%.so: %.o
+%.$(SUFFIX): %.o
 	$(CC) -o $@ $< $(LDFLAGS)
 
 # OTHER TARGETS
 
 install: all
 	-mkdir -p          $(INSTALL_PLUGINS_DIR)
-	cp *.so            $(INSTALL_PLUGINS_DIR)
+	cp *.$(SUFFIX)     $(INSTALL_PLUGINS_DIR)
 	-mkdir -p          $(INSTALL_LRDF_DIR)
 	cp tap-plugins.rdf $(INSTALL_LRDF_DIR)
 	cp tap_reverb.rdf  $(INSTALL_LRDF_DIR)
 
 clean:
-	-rm -f *.so *.o *~
+	-rm -f *.$(SUFFIX) *.o *~
 
 .PHONY: all install clean
