diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' lldb-5.0.1.src.orig/cmake/LLDBDependencies.cmake lldb-5.0.1.src/cmake/LLDBDependencies.cmake
--- lldb-5.0.1.src.orig/cmake/LLDBDependencies.cmake	2017-02-08 22:00:46.000000000 +0100
+++ lldb-5.0.1.src/cmake/LLDBDependencies.cmake	2018-02-16 08:36:28.383938300 +0100
@@ -6,6 +6,12 @@
     ws2_32
     rpcrt4
     )
+  if (MINGW)
+    list(APPEND LLDB_SYSTEM_LIBS
+      dbghelp
+      psapi
+      )
+  endif ()
 endif ()
 
 if (NOT LLDB_DISABLE_LIBEDIT)
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' lldb-5.0.1.src.orig/include/lldb/Host/windows/PosixApi.h lldb-5.0.1.src/include/lldb/Host/windows/PosixApi.h
--- lldb-5.0.1.src.orig/include/lldb/Host/windows/PosixApi.h	2017-01-25 12:10:52.000000000 +0100
+++ lldb-5.0.1.src/include/lldb/Host/windows/PosixApi.h	2018-02-16 08:39:34.083487900 +0100
@@ -73,13 +73,9 @@
 #define S_ISDIR(mode) (((mode)&S_IFMT) == S_IFDIR)
 #endif
 
-#endif // _MSC_VER
-
 // Various useful posix functions that are not present in Windows.  We provide
 // custom implementations.
 int vasprintf(char **ret, const char *fmt, va_list ap);
-char *strcasestr(const char *s, const char *find);
-char *realpath(const char *name, char *resolved);
 
 int usleep(uint32_t useconds);
 char *basename(char *path);
@@ -88,6 +84,20 @@
 int strcasecmp(const char *s1, const char *s2);
 int strncasecmp(const char *s1, const char *s2, size_t n);
 
+#elif __MINGW64_VERSION_MAJOR
+
+// pid_t definition
+#include <sys/types.h>
+
+// Various useful posix functions that are not present in Windows.  We provide
+// custom implementations.
+int vasprintf(char **ret, const char *fmt, va_list ap);
+
+#endif
+
+char *strcasestr(const char *s, const char *find);
+char *realpath(const char *name, char *resolved);
+
 // empty functions
 inline int posix_openpt(int flag) { LLVM_BUILTIN_UNREACHABLE; }
 
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' lldb-5.0.1.src.orig/include/lldb/Host/windows/windows.h lldb-5.0.1.src/include/lldb/Host/windows/windows.h
--- lldb-5.0.1.src.orig/include/lldb/Host/windows/windows.h	2016-12-15 16:00:41.000000000 +0100
+++ lldb-5.0.1.src/include/lldb/Host/windows/windows.h	2018-02-16 08:36:28.407305200 +0100
@@ -28,4 +28,8 @@
 #define FAR
 #define NEAR
 
+#ifndef _MSC_VER
+#include <sys/stat.h>
+#endif
+
 #endif // LLDB_lldb_windows_h_
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' lldb-5.0.1.src.orig/source/Host/common/MainLoop.cpp lldb-5.0.1.src/source/Host/common/MainLoop.cpp
--- lldb-5.0.1.src.orig/source/Host/common/MainLoop.cpp	2017-07-18 16:03:47.000000000 +0200
+++ lldb-5.0.1.src/source/Host/common/MainLoop.cpp	2018-02-16 08:36:28.420263800 +0100
@@ -27,6 +27,7 @@
 #if HAVE_SYS_EVENT_H
 #include <sys/event.h>
 #elif defined(LLVM_ON_WIN32)
+#include "lldb/Host/windows/windows.h"
 #include <winsock2.h>
 #else
 #include <poll.h>
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' lldb-5.0.1.src.orig/source/Plugins/Process/gdb-remote/GDBRemoteCommunicationServerPlatform.cpp lldb-5.0.1.src/source/Plugins/Process/gdb-remote/GDBRemoteCommunicationServerPlatform.cpp
--- lldb-5.0.1.src.orig/source/Plugins/Process/gdb-remote/GDBRemoteCommunicationServerPlatform.cpp	2017-06-27 12:45:31.000000000 +0200
+++ lldb-5.0.1.src/source/Plugins/Process/gdb-remote/GDBRemoteCommunicationServerPlatform.cpp	2018-02-16 08:36:28.431262500 +0100
@@ -12,6 +12,7 @@
 #include <errno.h>
 
 // C Includes
+#include <unistd.h>
 // C++ Includes
 #include <chrono>
 #include <cstring>
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' lldb-5.0.1.src.orig/source/Plugins/Process/gdb-remote/ProcessGDBRemote.h lldb-5.0.1.src/source/Plugins/Process/gdb-remote/ProcessGDBRemote.h
--- lldb-5.0.1.src.orig/source/Plugins/Process/gdb-remote/ProcessGDBRemote.h	2017-06-27 12:45:31.000000000 +0200
+++ lldb-5.0.1.src/source/Plugins/Process/gdb-remote/ProcessGDBRemote.h	2018-02-16 08:36:28.441294600 +0100
@@ -11,6 +11,7 @@
 #define liblldb_ProcessGDBRemote_h_
 
 // C Includes
+#include <unistd.h>
 // C++ Includes
 #include <atomic>
 #include <map>
R:\dev-utils\bin\diff.exe: lldb-5.0.1.src.orig/test/testcases: Not a directory
R:\dev-utils\bin\diff.exe: lldb-5.0.1.src/test/testcases: Not a directory
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' lldb-5.0.1.src.orig/tools/lldb-mi/MIDataTypes.h lldb-5.0.1.src/tools/lldb-mi/MIDataTypes.h
--- lldb-5.0.1.src.orig/tools/lldb-mi/MIDataTypes.h	2016-09-06 22:57:50.000000000 +0200
+++ lldb-5.0.1.src/tools/lldb-mi/MIDataTypes.h	2018-02-16 08:36:28.450261800 +0100
@@ -19,6 +19,8 @@
 // Windows headers:
 #ifdef _WIN32
 
+#include <windows.h>
+
 // Debugging:
 #ifdef _DEBUG
 #include <crtdbg.h> // C-runtime debugging library (defines _ASSERT).
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' lldb-5.0.1.src.orig/tools/lldb-mi/Platform.h lldb-5.0.1.src/tools/lldb-mi/Platform.h
--- lldb-5.0.1.src.orig/tools/lldb-mi/Platform.h	2017-03-03 21:21:59.000000000 +0100
+++ lldb-5.0.1.src/tools/lldb-mi/Platform.h	2018-02-16 08:36:28.459291200 +0100
@@ -78,8 +78,12 @@
 
 #include <getopt.h>
 #include <libgen.h>
+
+#ifndef __MINGW64_VERSION_MAJOR
 #include <sys/ioctl.h>
 #include <termios.h>
+#endif
+
 #include <unistd.h>
 
 #include <pthread.h>
