diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user tellico-3.3.2.orig/src/mainwindow.cpp tellico-3.3.2/src/mainwindow.cpp
--- tellico-3.3.2.orig/src/mainwindow.cpp	2020-08-08 02:51:39.000000000 +0200
+++ tellico-3.3.2/src/mainwindow.cpp	2020-08-22 11:21:27.838486000 +0200
@@ -227,7 +227,8 @@
   connect(ImageFactory::self(), &ImageFactory::imageLocationMismatch,
           this, &MainWindow::slotImageLocationMismatch);
   // Init DBUS
-  NewStuff::Manager::self();
+  // XXX Crash on quit https://bugs.kde.org/show_bug.cgi?id=422468
+  //NewStuff::Manager::self();
 }
 
 void MainWindow::initStatusBar() {
