From e4d0b0bc6b110b3c942f66d209d9eff6999251c2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20K=C3=BCnzel?= <jk.kdedev@smartlab.uber.space>
Date: Wed, 3 Jan 2024 12:10:16 +0000
Subject: [PATCH] Fix non-void return

---
 tests/proxymodeltestapp/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/proxymodeltestapp/main.cpp b/tests/proxymodeltestapp/main.cpp
index fa72cbe..23994f9 100644
--- a/tests/proxymodeltestapp/main.cpp
+++ b/tests/proxymodeltestapp/main.cpp
@@ -13,5 +13,5 @@ int main(int argc, char **argv)
     QApplication app(argc, argv);
     MainWindow *mw = new MainWindow();
     mw->show();
-    app.exec();
+    return app.exec();
 }
-- 
GitLab

