From 001c87209d6b73a3aae857ecc7883b7bd2f23b89 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20K=C3=BCnzel?= <julius.kuenzel@kde.org>
Date: Wed, 16 Oct 2024 15:59:34 +0000
Subject: [PATCH] Add namespace for Android as required by newer gradle
 versions

---
 src/editor/android/build.gradle | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/editor/android/build.gradle b/src/editor/android/build.gradle
index 72ac4b49..2072656f 100644
--- a/src/editor/android/build.gradle
+++ b/src/editor/android/build.gradle
@@ -44,5 +44,6 @@ android {
     defaultConfig {
         minSdkVersion @ANDROID_API_LEVEL@
         targetSdkVersion @ANDROID_SDK_COMPILE_API@
+        namespace "org.kde.osm.editorcontroller"
     }
 }
-- 
GitLab

