diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libxml2-2.9.7.orig/encoding.c libxml2-2.9.7/encoding.c
--- libxml2-2.9.7.orig/encoding.c	2017-07-23 07:10:06.000000000 +0200
+++ libxml2-2.9.7/encoding.c	2022-07-09 17:29:23.501611400 +0200
@@ -1884,12 +1884,12 @@
         /* encoding => UTF-16 => UTF-8 */
         ucnv_convertEx(cd->utf8, cd->uconv, &ucv_out, ucv_out + *outlen,
                        &ucv_in, ucv_in + *inlen, NULL, NULL, NULL, NULL,
-                       0, TRUE, &err);
+                       0, true, &err);
     } else {
         /* UTF-8 => UTF-16 => encoding */
         ucnv_convertEx(cd->uconv, cd->utf8, &ucv_out, ucv_out + *outlen,
                        &ucv_in, ucv_in + *inlen, NULL, NULL, NULL, NULL,
-                       0, TRUE, &err);
+                       0, true, &err);
     }
     *inlen = ucv_in - (const char*) in;
     *outlen = ucv_out - (char *) out;
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libxml2-2.9.7.orig/win32/Makefile.msvc libxml2-2.9.7/win32/Makefile.msvc
--- libxml2-2.9.7.orig/win32/Makefile.msvc	2017-10-26 09:54:40.000000000 +0200
+++ libxml2-2.9.7/win32/Makefile.msvc	2022-07-09 17:30:40.220459700 +0200
@@ -75,7 +75,7 @@
 LIBS = $(LIBS) iconv.lib
 !endif 
 !if "$(WITH_ICU)" == "1"
-LIBS = $(LIBS) icu.lib
+LIBS = $(LIBS) icu.lib icuuc.lib  icuin.lib icudt.lib
 !endif
 !if "$(WITH_ZLIB)" == "1"
 # could be named differently zdll or zlib
