From 0d07b1fd6f27ef8680a348499292277491dce9d2 Mon Sep 17 00:00:00 2001
From: Hannah von Reth <hannah.vonreth@owncloud.com>
Date: Sun, 10 Jul 2022 20:50:29 +0200
Subject: [PATCH] craft: 15694db887c72f0d04f929d7362482327e0f0ad0

---
 src/buildtools/config/common.pri | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/buildtools/config/common.pri b/src/buildtools/config/common.pri
index cf990c79b..910a88cac 100644
--- a/src/buildtools/config/common.pri
+++ b/src/buildtools/config/common.pri
@@ -26,6 +26,7 @@ gn_args += \
     skia_use_dawn=false \
     toolkit_views=false \
     treat_warnings_as_errors=false \
+    fatal_linker_warnings=false \
     use_allocator_shim=false \
     use_allocator=\"none\" \
     use_custom_libcxx=false \
-- 
2.36.1.windows.1

