diff --git a/data/kdenlive.notifyrc b/data/kdenlive.notifyrc
index c9de78ee1..5598a8c44 100644
--- a/data/kdenlive.notifyrc
+++ b/data/kdenlive.notifyrc
@@ -19,6 +19,7 @@ Comment[fr]=Kdenlive
 Comment[ga]=Kdenlive
 Comment[gl]=Kdenlive
 Comment[hu]=Kdenlive
+Comment[ia]=Kdenlive
 Comment[id]=Kdenlive
 Comment[it]=Kdenlive
 Comment[ja]=Kdenlive
@@ -86,7 +87,7 @@ Name[sv]=Återgivning klar
 Name[tr]=Hazırlama tamamlandı
 Name[uk]=Обробку завершено
 Name[x-test]=xxRendering finishedxx
-Name[zh_CN]=渲染完成
+Name[zh_CN]=导出任务已完成
 Name[zh_TW]=已完成輸出
 Comment=Rendering is over
 Comment[ar]=قد انتهى التّصيير
@@ -127,7 +128,7 @@ Comment[sv]=Återgivningen är gjord
 Comment[tr]=Hazırlama bitti
 Comment[uk]=Виконання обробки завершено
 Comment[x-test]=xxRendering is overxx
-Comment[zh_CN]=渲染结束
+Comment[zh_CN]=导出任务已完成
 Comment[zh_TW]=輸出已結束
 Action=Popup
 
@@ -171,7 +172,7 @@ Name[sv]=Återgivning startad
 Name[tr]=Hazırlama başladı
 Name[uk]=Обробку розпочато
 Name[x-test]=xxRendering startedxx
-Name[zh_CN]=渲染开始
+Name[zh_CN]=导出任务已开始
 Name[zh_TW]=輸出已開始
 Comment=Rendering was started
 Comment[ar]=قد بدأ التّصيير
@@ -212,7 +213,7 @@ Comment[sv]=Återgivningen har startats
 Comment[tr]=Hazırlama başlatıldı
 Comment[uk]=Було розпочато обробку
 Comment[x-test]=xxRendering was startedxx
-Comment[zh_CN]=渲染开始
+Comment[zh_CN]=导出任务已开始
 Comment[zh_TW]=輸出已開始
 Action=Popup
 
@@ -256,7 +257,7 @@ Name[sv]=Ram lagrad
 Name[tr]=Kare yakalandı
 Name[uk]=Захоплено кадр
 Name[x-test]=xxFrame capturedxx
-Name[zh_CN]=已抓取帧
+Name[zh_CN]=已捕获帧
 Name[zh_TW]=影格已擷取
 Comment=A frame was captured to disk
 Comment[ar]=قد التُقط إطار إلى القرص
@@ -297,7 +298,7 @@ Comment[sv]=En ram har lagrats på disk
 Comment[tr]=Bir kare diske yakalandı
 Comment[uk]=Було захоплено кадр, програма зберегла його на диску
 Comment[x-test]=xxA frame was captured to diskxx
-Comment[zh_CN]=一帧图像已被抓取到磁盘
+Comment[zh_CN]=一帧图像已被捕获到存储设备
 Comment[zh_TW]=已擷取一個影格到磁碟中
 Sound=Oxygen-Sys-App-Message.ogg
 Action=Sound
@@ -342,7 +343,7 @@ Name[sv]=Klar att lagra
 Name[tr]=Yakalamak için hazır
 Name[uk]=Приготовано до захоплення
 Name[x-test]=xxReady to capturexx
-Name[zh_CN]=准备抓取
+Name[zh_CN]=准备捕获
 Name[zh_TW]=準備擷取
 Sound=Oxygen-Sys-App-Positive.ogg
 Action=Sound
@@ -366,6 +367,7 @@ Name[fr]=Erreur
 Name[ga]=Earráid
 Name[gl]=Erro
 Name[hu]=Hiba
+Name[ia]=Error
 Name[id]=Error
 Name[it]=Errore
 Name[ja]=エラー
@@ -432,7 +434,7 @@ Comment[sv]=Ett fel uppstod i Kdenlive
 Comment[tr]=Kdenlive içerisinde bir hata oluştu
 Comment[uk]=У коді Kdenlive сталася помилка
 Comment[x-test]=xxAn error occurred in Kdenlivexx
-Comment[zh_CN]=Kdenlive 发生了一个错误
+Comment[zh_CN]=Kdenlive 发生错误
 Comment[zh_TW]=Kdenlive 中發生錯誤
 Sound=Oxygen-Sys-Warning.ogg
 Action=Sound
diff --git a/data/kdenlive_keyboardschemes.knsrc b/data/kdenlive_keyboardschemes.knsrc
index e7781827e..0ef90b989 100644
--- a/data/kdenlive_keyboardschemes.knsrc
+++ b/data/kdenlive_keyboardschemes.knsrc
@@ -13,6 +13,7 @@ Name[eu]=Kdenliveren teklatu antolaerak
 Name[fi]=Kdenliven näppäimistöteemat
 Name[fr]=Raccourcis claviers Kdenlive
 Name[gl]=Esquemas de teclado de Kdenlive
+Name[ia]=Schemas de claviero de Kdenlive
 Name[id]=Skema Keyboard Kdenlive
 Name[it]=Schemi da tastiera di Kdenlive
 Name[ko]=Kdenlive 키보드 단축키 배열
diff --git a/data/kdenlive_renderprofiles.knsrc b/data/kdenlive_renderprofiles.knsrc
index d3e7c7ed8..eb3b14312 100644
--- a/data/kdenlive_renderprofiles.knsrc
+++ b/data/kdenlive_renderprofiles.knsrc
@@ -29,7 +29,7 @@ Name[sv]=Kdenlive-återgivningsprofiler
 Name[tr]=Kdenlive İşleme Profilleri
 Name[uk]=Профілі обробки Kdenlive
 Name[x-test]=xxKdenlive Render Profilesxx
-Name[zh_CN]=Kdenlive 渲染配置文件
+Name[zh_CN]=Kdenlive 导出配置
 Name[zh_TW]=Kdenlive 繪製設定檔
 ProvidersUrl=https://autoconfig.kde.org/ocs/providers.xml
 Categories=Kdenlive Export Profiles
diff --git a/data/kdenlive_wipes.knsrc b/data/kdenlive_wipes.knsrc
index c5d344df4..b550412ac 100644
--- a/data/kdenlive_wipes.knsrc
+++ b/data/kdenlive_wipes.knsrc
@@ -29,7 +29,7 @@ Name[sv]=Kdenlive-övergångar
 Name[tr]=Kdenlive Geçiş Ekranları
 Name[uk]=Перехідні витирання Kdenlive
 Name[x-test]=xxKdenlive Transition Wipesxx
-Name[zh_CN]=Kdenlive 过渡擦除
+Name[zh_CN]=Kdenlive 转场消除方案
 Name[zh_TW]=Kdenlive 過渡 Wipes
 ProvidersUrl=https://autoconfig.kde.org/ocs/providers.xml
 Categories=Kdenlive FX
diff --git a/data/org.kde.kdenlive.appdata.xml b/data/org.kde.kdenlive.appdata.xml
index 86accaf34..ee3ace7ce 100644
--- a/data/org.kde.kdenlive.appdata.xml
+++ b/data/org.kde.kdenlive.appdata.xml
@@ -19,6 +19,7 @@
   <name xml:lang="fi">Kdenlive</name>
   <name xml:lang="fr">Kdenlive</name>
   <name xml:lang="gl">Kdenlive</name>
+  <name xml:lang="ia">Kdenlive</name>
   <name xml:lang="id">Kdenlive</name>
   <name xml:lang="it">Kdenlive</name>
   <name xml:lang="ko">Kdenlive</name>
@@ -52,6 +53,7 @@
   <summary xml:lang="fi">Videoeditori</summary>
   <summary xml:lang="fr">Éditeur vidéo</summary>
   <summary xml:lang="gl">Editor de vídeo</summary>
+  <summary xml:lang="ia">Editor (redactor) de Video</summary>
   <summary xml:lang="id">Editor Video</summary>
   <summary xml:lang="it">Editor video</summary>
   <summary xml:lang="ko">동영상 편집기</summary>
@@ -67,7 +69,7 @@
   <summary xml:lang="tr">Video Düzenleyicisi</summary>
   <summary xml:lang="uk">Відеоредактор</summary>
   <summary xml:lang="x-test">xxVideo Editorxx</summary>
-  <summary xml:lang="zh-CN">视频编辑器</summary>
+  <summary xml:lang="zh-CN">视频编辑工具</summary>
   <summary xml:lang="zh-TW">影像編輯器</summary>
   <description>
     <p>Kdenlive is a non linear video editor. It is based on the MLT framework and accepts many audio and video formats, allows you to add effects, transitions and render into the format of your choice.</p>
@@ -100,7 +102,7 @@
     <p xml:lang="tr">Kdenlive doğrusal olmayan bir video düzenleyicisidir. MLT çalışma çerçevesi tabanlıdır ve birçok ses ile video biçimini destekleyerek; efektler ve geçişler eklemenizi ve istediğiniz biçime dönüştürmenizi sağlar.</p>
     <p xml:lang="uk">Kdenlive — нелінійний відеоредактор. Програму засновано на наборі засобів MLT, вона може працювати з даними у багатьох форматах зберігання відео та звукових даних, дозволяє додавати ефекти, переходи та надає змогу зберегти результати редагування у вибраному вами форматі.</p>
     <p xml:lang="x-test">xxKdenlive is a non linear video editor. It is based on the MLT framework and accepts many audio and video formats, allows you to add effects, transitions and render into the format of your choice.xx</p>
-    <p xml:lang="zh-CN">Kdenlive 是一个非线性视频编辑器。它基于 MLT 框架并且支持许多音频和视频格式，允许您添加特效，转场并渲染为您想要的格式。</p>
+    <p xml:lang="zh-CN">Kdenlive 是一款非线性视频编辑工具。它基于 MLT 多媒体程序框架开发，支持输入多种音频和视频格式，可以添加各种特效和转场，并导出为您想要的格式。</p>
     <p xml:lang="zh-TW">Kdenlive 是一套非線性的影像編輯器。基於 MLT 框架，可接受許多影音格式，讓您可以加入特效，過場並轉編碼成您選擇的格式。</p>
     <p>Features:</p>
     <p xml:lang="ar">الميزات:</p>
@@ -118,6 +120,7 @@
     <p xml:lang="fi">Ominaisuuksia:</p>
     <p xml:lang="fr">Fonctionnalités :</p>
     <p xml:lang="gl">Funcionalidades:</p>
+    <p xml:lang="ia">Characteristicas:</p>
     <p xml:lang="id">Fitur:</p>
     <p xml:lang="it">Funzionalità:</p>
     <p xml:lang="ko">기능:</p>
@@ -198,7 +201,7 @@
       <li xml:lang="tr">Birçok efekt ve geçiş.</li>
       <li xml:lang="uk">Багато ефектів та переходів.</li>
       <li xml:lang="x-test">xxMany effects and transitions.xx</li>
-      <li xml:lang="zh-CN">提供许多特效和转场。</li>
+      <li xml:lang="zh-CN">提供多种特效和转场。</li>
       <li xml:lang="zh-TW">許多特效與轉場效果。</li>
       <li>Color scopes</li>
       <li xml:lang="ar">نطاقات الألوان</li>
diff --git a/data/org.kde.kdenlive.desktop b/data/org.kde.kdenlive.desktop
index 48898aba4..1fc75ddfb 100644
--- a/data/org.kde.kdenlive.desktop
+++ b/data/org.kde.kdenlive.desktop
@@ -18,6 +18,7 @@ Name[fr]=Kdenlive
 Name[ga]=Kdenlive
 Name[gl]=Kdenlive
 Name[hu]=Kdenlive
+Name[ia]=Kdenlive
 Name[id]=Kdenlive
 Name[it]=Kdenlive
 Name[ja]=Kdenlive
@@ -62,6 +63,7 @@ GenericName[fr]=Éditeur vidéo
 GenericName[ga]=Eagarthóir Físe
 GenericName[gl]=Editor de vídeo
 GenericName[hu]=Videoszerkesztő
+GenericName[ia]=Editor (redactor) de video
 GenericName[id]=Pengedit Video
 GenericName[it]=Editor video
 GenericName[ja]=ビデオエディタ
@@ -84,7 +86,7 @@ GenericName[sv]=Videoeditor
 GenericName[tr]=Video Düzenleyici
 GenericName[uk]=Відеоредактор
 GenericName[x-test]=xxVideo Editorxx
-GenericName[zh_CN]=视频编辑器
+GenericName[zh_CN]=视频编辑工具
 GenericName[zh_TW]=影像編輯器
 Comment=Nonlinear video editor by KDE
 Comment[ar]=محرّر فيديوهات غير خطّيّ من كدي
@@ -116,7 +118,7 @@ Comment[sv]=Icke-linjär videoeditor av KDE
 Comment[tr]=KDE için doğrusal olmayan video düzenleyici
 Comment[uk]=Нелінійний редактор відео від KDE
 Comment[x-test]=xxNonlinear video editor by KDExx
-Comment[zh_CN]=KDE 推出的非线性视频编辑器
+Comment[zh_CN]=KDE 社区开发的非线性视频编辑工具
 Comment[zh_TW]=來自 KDE 的非線性影像編輯器
 Type=Application
 Exec=kdenlive %F
diff --git a/data/org.kde.kdenlive.xml b/data/org.kde.kdenlive.xml
index 3f92da607..9b35908a7 100644
--- a/data/org.kde.kdenlive.xml
+++ b/data/org.kde.kdenlive.xml
@@ -8,6 +8,7 @@
     <comment xml:lang="es">Documento de proyecto de video de Kdenlive</comment>
     <comment xml:lang="eu">Kdenlive-ko bideo-proiektuaren dokumentua</comment>
     <comment xml:lang="fr">Document de projet vidéo pour Kdenlive</comment>
+    <comment xml:lang="ia">Documento de projecto video de Kdenlive</comment>
     <comment xml:lang="it">Progetto video di Kdenlive</comment>
     <comment xml:lang="nl">Video projectdocument van Kdenlive</comment>
     <comment xml:lang="nn">Kdenlive-videoprosjektdokument</comment>
@@ -16,7 +17,7 @@
     <comment xml:lang="ru">Видеопроект Kdenlive</comment>
     <comment xml:lang="sv">Kdenlive-videoprojektdokument</comment>
     <comment xml:lang="uk">документ відеопроєкту Kdenlive</comment>
-    <comment xml:lang="zh_CN">Kdenlive 视频项目文档</comment>
+    <comment xml:lang="zh_CN">Kdenlive 视频工程文件</comment>
     <sub-class-of type="video/mlt-playlist"/>
     <glob pattern="*.kdenlive"/>
   </mime-type>
@@ -28,6 +29,7 @@
     <comment xml:lang="es">Título de video de Kdenlive</comment>
     <comment xml:lang="eu">Kdenlive-ko bideo-izenburua</comment>
     <comment xml:lang="fr">Titre de vidéo pour Kdenlive</comment>
+    <comment xml:lang="ia">Titulo video de Kdenlive</comment>
     <comment xml:lang="it">Titolo video di Kdenlive</comment>
     <comment xml:lang="nl">Titel van Kdenlive-video</comment>
     <comment xml:lang="nn">Kdenlive-videotittel</comment>
diff --git a/src/timeline2/view/qml/timeline.qml b/src/timeline2/view/qml/timeline.qml
index 08a1b4e83..9e1567354 100644
--- a/src/timeline2/view/qml/timeline.qml
+++ b/src/timeline2/view/qml/timeline.qml
@@ -286,8 +286,8 @@ Rectangle {
     function regainFocus(mousePos) {
         var currentMouseTrack = Logic.getTrackIdFromPos(mousePos.y - ruler.height - subtitleTrack.height + scrollView.contentY)
         // Try to find correct item
-        //console.log('checking item on TK: ', currentMouseTrack, ' AT: ', mousePos, ' SCROLL POS: ', (mousePos.y - ruler.height - subtitleTrack.height + scrollView.contentY))
-        var tentativeClip = getItemAtPos(currentMouseTrack, mousePos.x - trackHeaders.width, dragProxy.isComposition)
+        //console.log('checking item on TK: ', currentMouseTrack, ' AT: XPOS', (mousePos.x - trackHeaders.width), ', SCOLL:', scrollView.contentX, ', RES: ', ((mousePos.x - trackHeaders.width + scrollView.contentX) / timeline.scaleFactor), ' SCROLL POS: ', (mousePos.y - ruler.height - subtitleTrack.height + scrollView.contentY))
+        var tentativeClip = getItemAtPos(currentMouseTrack, mousePos.x - trackHeaders.width + scrollView.contentX, dragProxy.isComposition)
         if (tentativeClip && tentativeClip.clipId) {
             dragProxy.draggedItem = tentativeClip.clipId
             var tk = controller.getItemTrackId(tentativeClip.clipId)
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 99660c200..7922c5d26 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -1,4 +1,4 @@
-include_directories(${MLT_INCLUDE_DIR} ..)
+include_directories(${MLT_INCLUDE_DIR} ${MLTPP_INCLUDE_DIR} ..)
 kde_enable_exceptions()
 add_executable(runTests
     TestMain.cpp
