diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1a69afd2..fdb54bd0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -157,6 +157,12 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_BUILD_TYPE STREQUAL "Debug")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-c++20-compat -Wno-dangling-reference")
 endif ()
 
+if(MSVC)
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /experimental:c11atomics")
+  # Enable math defines for M_PI and M_PI_2
+  add_compile_definitions(_USE_MATH_DEFINES)
+endif()
+
 if(MSVC)
   list(APPEND MLT_COMPILE_OPTIONS "$<$<CONFIG:Release>:/fp:fast>")
 elseif(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")

