diff --git a/src/framework/mlt_pool.c b/src/framework/mlt_pool.c
index e94f668c..878aa009 100644
--- a/src/framework/mlt_pool.c
+++ b/src/framework/mlt_pool.c
@@ -95,7 +95,12 @@ typedef struct mlt_pool_s
  * optimized libraries (sse/altivec).
  */
 
+// In the future with C23 we might be able to use alignas
+#ifdef _MSC_VER
+typedef struct __declspec(align(16)) mlt_release_s
+#else
 typedef struct __attribute__((aligned(16))) mlt_release_s
+#endif
 {
     mlt_pool pool;
     int references;
