diff --git a/src/framework/mlt_types.h b/src/framework/mlt_types.h
index 58486d96..bcec11aa 100644
--- a/src/framework/mlt_types.h
+++ b/src/framework/mlt_types.h
@@ -292,20 +292,24 @@ typedef void *(*mlt_thread_function_t)(void *);      /**< generic thread functio
 #ifdef _WIN32
 #include <pthread.h>
 /* Win32 compatibility function declarations */
-#if !defined(__MINGW32__)
+#if !defined(__MINGW32__) && !defined(_MSC_VER)
 MLT_EXPORT int usleep(unsigned int useconds);
 #endif
 #ifndef WIN_PTHREADS_TIME_H
 MLT_EXPORT int nanosleep(const struct timespec *rqtp, struct timespec *rmtp);
 #endif
+#ifndef _MSC_VER
 MLT_EXPORT int setenv(const char *name, const char *value, int overwrite);
+#endif
 MLT_EXPORT char *getlocale();
 MLT_EXPORT FILE *win32_fopen(const char *filename_utf8, const char *mode_utf8);
 #include <sys/stat.h>
 #include <sys/types.h>
 MLT_EXPORT int win32_stat(const char *filename_utf8, struct stat *buffer);
 #include <time.h>
+#ifndef _MSC_VER
 MLT_EXPORT char *strptime(const char *buf, const char *fmt, struct tm *tm);
+#endif
 #define mlt_fopen win32_fopen
 #define mlt_stat win32_stat
 #define MLT_DIRLIST_DELIMITER ";"

diff --git a/src/win32/win32.c b/src/win32/win32.c
index 55b91d25..7a37be9c 100644
--- a/src/win32/win32.c
+++ b/src/win32/win32.c
@@ -58,6 +58,8 @@ int nanosleep(const struct timespec * rqtp, struct timespec * rmtp)
 }
 #endif
 
+#ifndef _MSC_VER
+
 int setenv(const char *name, const char *value, int overwrite)
 {
 	int result = 1; 
@@ -68,7 +70,9 @@ int setenv(const char *name, const char *value, int overwrite)
 	 }
 
 	return result; 
-} 
+}
+
+#endif
 
 static int iconv_from_utf8( mlt_properties properties, const char *prop_name, const char *prop_name_out, const char* encoding )
 {
diff --git a/src/win32/strptime.c b/src/win32/strptime.c
index c3ed8ced..d06c0285 100644
--- a/src/win32/strptime.c
+++ b/src/win32/strptime.c
@@ -28,6 +28,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#ifndef _MSC_VER
+
 /*
 #include <sys/cdefs.h>
 #if defined(LIBC_SCCS) && !defined(lint)
@@ -598,3 +600,5 @@ find_string(const u_char *bp, int *tgt, const char * const *n1,
 	/* Nothing matched */
 	return NULL;
 }
+
+#endif

