diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff04b20d..bafcebeb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -185,6 +185,7 @@ endif()
 
 if (MSVC)
   find_package(PThreads4W REQUIRED)
+  find_package(Dirent 1.26 REQUIRED)
 endif ()
 find_package(Threads REQUIRED)
 find_package(PkgConfig REQUIRED)
diff --git a/src/framework/CMakeLists.txt b/src/framework/CMakeLists.txt
index aae6c994..6d16e093 100644
--- a/src/framework/CMakeLists.txt
+++ b/src/framework/CMakeLists.txt
@@ -87,7 +87,7 @@ target_compile_options(mlt PRIVATE ${MLT_COMPILE_OPTIONS})
 target_link_libraries(mlt PRIVATE Threads::Threads ${CMAKE_DL_LIBS})
 
 if(MSVC)
-  target_link_libraries(mlt PRIVATE msvccompat PThreads4W::PThreads4W)
+  target_link_libraries(mlt PRIVATE msvccompat PThreads4W::PThreads4W dirent)
 else()
   target_link_libraries(mlt PRIVATE m)
 endif()
