diff --git a/src/modules/sdl2/consumer_sdl2.c b/src/modules/sdl2/consumer_sdl2.c
index 6befb4c8..1e02e662 100644
--- a/src/modules/sdl2/consumer_sdl2.c
+++ b/src/modules/sdl2/consumer_sdl2.c
@@ -35,7 +35,7 @@
 
 #undef MLT_IMAGE_FORMAT // only yuv422 working currently
 
-extern pthread_mutex_t mlt_sdl_mutex;
+__declspec(dllimport) extern pthread_mutex_t mlt_sdl_mutex;
 
 /** This classes definition.
 */
diff --git a/src/modules/sdl2/consumer_sdl2_audio.c b/src/modules/sdl2/consumer_sdl2_audio.c
index 523daf32..87d7385b 100644
--- a/src/modules/sdl2/consumer_sdl2_audio.c
+++ b/src/modules/sdl2/consumer_sdl2_audio.c
@@ -33,7 +33,7 @@
 #include <string.h>
 #include <sys/time.h>
 
-extern pthread_mutex_t mlt_sdl_mutex;
+__declspec(dllimport) extern pthread_mutex_t mlt_sdl_mutex;
 
 /** This classes definition.
 */
