diff --git a/CMakeLists.txt b/CMakeLists.txt
index 180932e3..5343887d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -176,6 +176,7 @@ endif()
 
 if (MSVC)
   find_package(PThreads4W REQUIRED)
+  find_package(KDEWin REQUIRED)
 endif ()
 find_package(Threads REQUIRED)
 find_package(PkgConfig REQUIRED)
diff --git a/src/framework/CMakeLists.txt b/src/framework/CMakeLists.txt
index 159dbc11..bde40496 100644
--- a/src/framework/CMakeLists.txt
+++ b/src/framework/CMakeLists.txt
@@ -83,7 +83,7 @@ target_compile_options(mlt PRIVATE ${MLT_COMPILE_OPTIONS})
 target_link_libraries(mlt PRIVATE Threads::Threads ${CMAKE_DL_LIBS})
 
 if(MSVC)
-  target_link_libraries(mlt PRIVATE msvccompat PThreads4W::PThreads4W)
+  target_link_libraries(mlt PRIVATE msvccompat PThreads4W::PThreads4W ${KDEWIN_LIBRARIES})
 else()
   target_link_libraries(mlt PRIVATE m)
 endif()
diff --git a/src/melt/CMakeLists.txt b/src/melt/CMakeLists.txt
index 592c3c2e..943f9b04 100644
--- a/src/melt/CMakeLists.txt
+++ b/src/melt/CMakeLists.txt
@@ -16,7 +16,7 @@ if(SDL2_FOUND AND NOT ANDROID)
 endif()
 
 if(MSVC)
-  target_link_libraries(melt PRIVATE msvccompat)
+  target_link_libraries(melt PRIVATE msvccompat ${KDEWIN_LIBRARIES})
 endif()
 
 if(MINGW)

diff --git a/src/modules/avformat/CMakeLists.txt b/src/modules/avformat/CMakeLists.txt
index 709d1d82..ca1e6eb6 100644
--- a/src/modules/avformat/CMakeLists.txt
+++ b/src/modules/avformat/CMakeLists.txt
@@ -38,7 +38,7 @@ target_link_libraries(mltavformat PRIVATE
 )
 
 if(MSVC)
-  target_link_libraries(mltavformat PRIVATE PThreads4W::PThreads4W)
+  target_link_libraries(mltavformat PRIVATE PThreads4W::PThreads4W ${KDEWIN_LIBRARIES})
 else()
    target_link_libraries(mltavformat PRIVATE m)
 endif()

diff --git a/src/modules/core/CMakeLists.txt b/src/modules/core/CMakeLists.txt
index b59df794..38681d59 100644
--- a/src/modules/core/CMakeLists.txt
+++ b/src/modules/core/CMakeLists.txt
@@ -60,7 +60,7 @@ target_link_libraries(mltcore PRIVATE mlt Threads::Threads)
 
 target_include_directories(mltcore PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 if(MSVC)
-  target_link_libraries(mltcore PRIVATE PThreads4W::PThreads4W)
+  target_link_libraries(mltcore PRIVATE PThreads4W::PThreads4W ${KDEWIN_LIBRARIES})
 else()
   target_link_libraries(mltcore PRIVATE m)
 endif()

diff --git a/src/modules/frei0r/CMakeLists.txt b/src/modules/frei0r/CMakeLists.txt
index 38f4ab3f..ca428f25 100644
--- a/src/modules/frei0r/CMakeLists.txt
+++ b/src/modules/frei0r/CMakeLists.txt
@@ -27,7 +27,9 @@ endif()
 
 target_link_libraries(mltfrei0r PRIVATE mlt ${CMAKE_DL_LIBS})
 
-if(NOT MSVC)
+if(MSVC)
+  target_link_libraries(mltfrei0r PRIVATE ${KDEWIN_LIBRARIES})
+else()
   target_link_libraries(mltfrei0r PRIVATE m)
 endif()
 
diff --git a/src/modules/kdenlive/CMakeLists.txt b/src/modules/kdenlive/CMakeLists.txt
index 06454404..3e6d370b 100644
--- a/src/modules/kdenlive/CMakeLists.txt
+++ b/src/modules/kdenlive/CMakeLists.txt
@@ -15,7 +15,9 @@ generate_export_header(mltkdenlive)
 target_compile_options(mltkdenlive PRIVATE ${MLT_COMPILE_OPTIONS})
 target_include_directories(mltkdenlive PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 target_link_libraries(mltkdenlive PRIVATE mlt)
-if(NOT MSVC)
+if(MSVC)
+  target_link_libraries(mltkdenlive PRIVATE ${KDEWIN_LIBRARIES})
+else()
   target_link_libraries(mltkdenlive PRIVATE m)
 endif()
 
diff --git a/src/modules/plus/CMakeLists.txt b/src/modules/plus/CMakeLists.txt
index 4d018027..ea24d3ba 100644
--- a/src/modules/plus/CMakeLists.txt
+++ b/src/modules/plus/CMakeLists.txt
@@ -44,7 +44,7 @@ target_compile_options(mltplus PRIVATE ${MLT_COMPILE_OPTIONS})
 target_include_directories(mltplus PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 target_link_libraries(mltplus PRIVATE mlt Threads::Threads)
 if(MSVC)
-  target_link_libraries(mltplus PRIVATE msvccompat)
+  target_link_libraries(mltplus PRIVATE msvccompat ${KDEWIN_LIBRARIES})
 else()
   target_link_libraries(mltplus PRIVATE m)
 endif()

diff --git a/src/modules/plusgpl/CMakeLists.txt b/src/modules/plusgpl/CMakeLists.txt
index f63529ee..10c58731 100644
--- a/src/modules/plusgpl/CMakeLists.txt
+++ b/src/modules/plusgpl/CMakeLists.txt
@@ -20,7 +20,9 @@ generate_export_header(mltplusgpl)
 target_compile_options(mltplusgpl PRIVATE ${MLT_COMPILE_OPTIONS})
 target_include_directories(mltplusgpl PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 target_link_libraries(mltplusgpl PRIVATE mlt mlt++ Threads::Threads)
-if(NOT MSVC)
+if(MSVC)
+  target_link_libraries(mltplusgpl PRIVATE ${KDEWIN_LIBRARIES})
+else()
   target_link_libraries(mltplusgpl PRIVATE m)
 endif()
 
diff --git a/src/modules/rtaudio/CMakeLists.txt b/src/modules/rtaudio/CMakeLists.txt
index 9a4fc134..1c91840b 100644
--- a/src/modules/rtaudio/CMakeLists.txt
+++ b/src/modules/rtaudio/CMakeLists.txt
@@ -12,7 +12,7 @@ target_compile_options(mltrtaudio PRIVATE ${MLT_COMPILE_OPTIONS})
 target_link_libraries(mltrtaudio PRIVATE mlt Threads::Threads)
 
 if(MSVC)
-  target_link_libraries(mltrtaudio PRIVATE PThreads4W::PThreads4W)
+  target_link_libraries(mltrtaudio PRIVATE PThreads4W::PThreads4W ${KDEWIN_LIBRARIES})
 endif()
 
 if(TARGET PkgConfig::rtaudio)
diff --git a/src/modules/sdl2/CMakeLists.txt b/src/modules/sdl2/CMakeLists.txt
index 56299bb6..8091ee8d 100644
--- a/src/modules/sdl2/CMakeLists.txt
+++ b/src/modules/sdl2/CMakeLists.txt
@@ -15,7 +15,7 @@ target_compile_options(mltsdl2 PRIVATE ${MLT_COMPILE_OPTIONS})
 target_include_directories(mltsdl2 PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 target_link_libraries(mltsdl2 PRIVATE mlt Threads::Threads SDL2::SDL2)
 if(MSVC)
-  target_link_libraries(mltsdl2 PRIVATE PThreads4W::PThreads4W)
+  target_link_libraries(mltsdl2 PRIVATE PThreads4W::PThreads4W ${KDEWIN_LIBRARIES})
 else()
   target_link_libraries(mltsdl2 PRIVATE m)
 endif()

diff --git a/src/modules/xml/CMakeLists.txt b/src/modules/xml/CMakeLists.txt
index f9b9e74c..c52ae557 100644
--- a/src/modules/xml/CMakeLists.txt
+++ b/src/modules/xml/CMakeLists.txt
@@ -17,7 +17,7 @@ target_compile_options(mltxml PRIVATE ${MLT_COMPILE_OPTIONS})
 target_include_directories(mltxml PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 target_link_libraries(mltxml PRIVATE mlt Threads::Threads PkgConfig::xml)
 if(MSVC)
-  target_link_libraries(mltxml PRIVATE PThreads4W::PThreads4W)
+  target_link_libraries(mltxml PRIVATE PThreads4W::PThreads4W ${KDEWIN_LIBRARIES})
 else()
   target_link_libraries(mltxml PRIVATE m)
 endif()

diff --git a/src/modules/jackrack/CMakeLists.txt b/src/modules/jackrack/CMakeLists.txt
index b32826e6..f7b08646 100644
--- a/src/modules/jackrack/CMakeLists.txt
+++ b/src/modules/jackrack/CMakeLists.txt
@@ -29,7 +29,9 @@ if(GPL AND TARGET PkgConfig::xml AND TARGET PkgConfig::glib AND ladspa_h_FOUND)
     filter_ladspa.c
   )
   target_link_libraries(mltjackrack PRIVATE ${CMAKE_DL_LIBS} PkgConfig::xml PkgConfig::glib)
-  if(NOT MSVC)
+  if(MSVC)
+    target_link_libraries(mltjackrack PRIVATE ${KDEWIN_LIBRARIES})
+  else()
     target_link_libraries(mltjackrack PRIVATE m)
   endif()
 
@@ -77,7 +79,9 @@ if(GPL AND TARGET PkgConfig::xml AND TARGET PkgConfig::glib AND ladspa_h_FOUND)
     filter_ladspa.c
   )
   target_link_libraries(mltladspa PRIVATE ${CMAKE_DL_LIBS} PkgConfig::xml PkgConfig::glib)
-  if(NOT MSVC)
+  if(MSVC)
+    target_link_libraries(mltladspa PRIVATE ${KDEWIN_LIBRARIES})
+  else()
     target_link_libraries(mltladspa PRIVATE m)
   endif()

diff --git a/src/modules/qt/CMakeLists.txt b/src/modules/qt/CMakeLists.txt
index 208a8d13..223c1ff8 100644
--- a/src/modules/qt/CMakeLists.txt
+++ b/src/modules/qt/CMakeLists.txt
@@ -59,7 +59,7 @@ function(mlt_add_qt_module ARG_TARGET)
     )
     target_include_directories(${ARG_TARGET} PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
     if(MSVC)
-      target_link_libraries(${ARG_TARGET} PRIVATE PThreads4W::PThreads4W)
+      target_link_libraries(${ARG_TARGET} PRIVATE PThreads4W::PThreads4W ${KDEWIN_LIBRARIES})
     else()
         target_link_libraries(${ARG_TARGET} PRIVATE m)
     endif()

diff --git a/src/modules/xine/CMakeLists.txt b/src/modules/xine/CMakeLists.txt
index 5eedb041..39ce971a 100644
--- a/src/modules/xine/CMakeLists.txt
+++ b/src/modules/xine/CMakeLists.txt
@@ -17,6 +17,10 @@ target_compile_options(mltxine PRIVATE ${MLT_COMPILE_OPTIONS})
 target_include_directories(mltxine PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 target_link_libraries(mltxine PRIVATE mlt ${CMAKE_DL_LIBS})
 
+if(MSVC)
+  target_link_libraries(mltxine PRIVATE ${KDEWIN_LIBRARIES})
+endif()
+
 set_target_properties(mltxine PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
 target_compile_definitions(mltxine PRIVATE PIC)

