diff --git a/CMakeLists.txt b/CMakeLists.txt
index 09cc9b43..49bd0463 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,10 +200,7 @@ endif()
 find_package(SDL2)
 
 if(MOD_QT OR MOD_QT6 OR MOD_PLUS)
-  pkg_check_modules(FFTW IMPORTED_TARGET fftw3)
-  if(NOT FFTW_FOUND)
-    pkg_check_modules(FFTW IMPORTED_TARGET fftw)
-  endif()
+  find_package(FFTW3)
 endif()
 
 if(MOD_QT OR MOD_QT6 OR MOD_GDK)
diff --git a/cmake/FindFFTW3.cmake b/cmake/FindFFTW3.cmake
new file mode 100644
index 00000000..70fe6431
--- /dev/null
+++ b/cmake/FindFFTW3.cmake
@@ -0,0 +1,35 @@
+# FFTW3 supports different build systems and the necessary
+# bits for find_package are only installed if CMake was used
+
+# First try to find FFTW3 using find_package
+find_package(FFTW3 CONFIG QUIET)
+
+if(NOT FFTW3_FOUND)
+    # Fallback to pkg-config
+    find_package(PkgConfig)
+
+    if(PkgConfig_FOUND)
+        pkg_check_modules(FFTW3 IMPORTED_TARGET QUIET fftw3)
+    endif()
+endif()
+
+if(NOT TARGET FFTW3::fftw3)
+    add_library(FFTW3::fftw3 INTERFACE IMPORTED)
+
+    if(FFTW3_INCLUDE_DIRS)
+        set_property(TARGET FFTW3::fftw3 PROPERTY
+                INTERFACE_INCLUDE_DIRECTORIES "${FFTW3_INCLUDE_DIRS}")
+    endif()
+    if(FFTW3_LINK_LIBRARIES)
+        set_property(TARGET FFTW3::fftw3 PROPERTY
+                INTERFACE_LINK_LIBRARIES "${FFTW3_LINK_LIBRARIES}")
+    endif()
+    if(FFTW3_LDFLAGS_OTHER)
+        set_property(TARGET FFTW3::fftw3 PROPERTY
+                INTERFACE_LINK_OPTIONS "${FFTW3_LDFLAGS_OTHER}")
+    endif()
+    if(FFTW3_CFLAGS_OTHER)
+        set_property(TARGET FFTW3::fftw3 PROPERTY
+                INTERFACE_COMPILE_OPTIONS "${FFTW3_CFLAGS_OTHER}")
+    endif()
+endif()
diff --git a/src/modules/plus/CMakeLists.txt b/src/modules/plus/CMakeLists.txt
index c8b6f981..f9980bd4 100644
--- a/src/modules/plus/CMakeLists.txt
+++ b/src/modules/plus/CMakeLists.txt
@@ -48,9 +48,9 @@ else()
   target_link_libraries(mltplus PRIVATE m)
 endif()
 
-if(TARGET PkgConfig::FFTW)
+if(FFTW3_FOUND)
   target_sources(mltplus PRIVATE filter_dance.c filter_fft.c)
-  target_link_libraries(mltplus PRIVATE PkgConfig::FFTW)
+  target_link_libraries(mltplus PRIVATE FFTW3::fftw3)
   target_compile_definitions(mltplus PRIVATE USE_FFTW)
   install(FILES filter_dance.yml filter_fft.yml DESTINATION ${MLT_INSTALL_DATA_DIR}/plus)
 endif()
diff --git a/src/modules/qt/CMakeLists.txt b/src/modules/qt/CMakeLists.txt
index 2c8c86ac..a7decfe7 100644
--- a/src/modules/qt/CMakeLists.txt
+++ b/src/modules/qt/CMakeLists.txt
@@ -84,9 +84,9 @@ function(mlt_add_qt_module ARG_TARGET)
       install(FILES transition_vqm.yml DESTINATION ${MLT_INSTALL_DATA_DIR}/${ARG_DATADIR})
     endif()
 
-    if(TARGET PkgConfig::FFTW)
+    if(FFTW3_FOUND)
       target_sources(${ARG_TARGET} PRIVATE filter_audiospectrum.cpp filter_lightshow.cpp)
-      target_link_libraries(${ARG_TARGET} PRIVATE PkgConfig::FFTW)
+      target_link_libraries(${ARG_TARGET} PRIVATE FFTW3::fftw3)
       target_compile_definitions(${ARG_TARGET} PRIVATE USE_FFTW)
       install(FILES filter_audiospectrum.yml filter_lightshow.yml DESTINATION ${MLT_INSTALL_DATA_DIR}/${ARG_DATADIR})
     endif()
-- 
2.44.0.windows.1

