diff -Nrub -x *~ -x *\.rej -x *\.orig -x *\.o -x *\.pyc -x CMakeLists.txt.user x264-b86ae3c66f51ac9eab5ab7ad09a9d62e67961b8a.orig/configure x264-b86ae3c66f51ac9eab5ab7ad09a9d62e67961b8a/configure
--- x264-b86ae3c66f51ac9eab5ab7ad09a9d62e67961b8a.orig/configure	2021-02-11 17:09:43.000000000 +0300
+++ x264-b86ae3c66f51ac9eab5ab7ad09a9d62e67961b8a/configure	2021-05-19 19:30:07.540707000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 if test x"$1" = x"-h" -o x"$1" = x"--help" ; then
 cat <<EOF
