diff --git a/src/gui/kwindowconfig.cpp b/src/gui/kwindowconfig.cpp
index 542b237e9a0869e38f0b442a49bc0426f0a69cbe..c49e295cf0ca3b563ee875d8ab139dd62e5589d0 100644
--- a/src/gui/kwindowconfig.cpp
+++ b/src/gui/kwindowconfig.cpp
@@ -12,6 +12,15 @@
 #include <QScreen>
 #include <QWindow>
 
+// QScreen::name() returns garbage on Windows; see https://bugreports.qt.io/browse/QTBUG-74317
+// So we use the screens' serial numbers to identify them instead
+// FIXME: remove this once we can depend on Qt 6.4, where this is fixed
+#ifdef Q_OS_WIN
+#define SCREENNAME serialNumber
+#else
+#define SCREENNAME name
+#endif
+
 static const char s_initialSizePropertyName[] = "_kconfig_initial_size";
 static const char s_initialScreenSizePropertyName[] = "_kconfig_initial_screen_size";
 
@@ -22,13 +31,7 @@ static QString allConnectedScreens()
     const auto screens = QGuiApplication::screens();
     names.reserve(screens.length());
     for (auto screen : screens) {
-#ifdef Q_OS_WIN
-        // QScreen::name() returns garbage on Windows; see https://bugreports.qt.io/browse/QTBUG-74317
-        // So we use the screens' serial numbers to identify them instead
-        names << screen->serialNumber();
-#else
-        names << screen->name();
-#endif
+        names << screen->SCREENNAME();
     }
     return names.join(QLatin1Char(' '));
 }
@@ -37,11 +40,11 @@ static QString allConnectedScreens()
 // returns current window screen if not found
 static QScreen *findScreenByName(const QWindow *window, const QString screenName)
 {
-    if (screenName == window->screen()->name()) {
+    if (screenName == window->screen()->SCREENNAME()) {
         return window->screen();
     }
     for (QScreen *s : window->screen()->virtualSiblings()) {
-        if (s->name() == screenName) {
+        if (s->SCREENNAME() == screenName) {
             return s;
         }
     }
@@ -55,7 +58,7 @@ static QString configFileString(const QScreen *screen, const QString &key)
     // We include resolution data to also save data on a per-resolution basis
     const QString returnString =
         QStringLiteral("%1 %2 %3x%4 %5")
-            .arg(allConnectedScreens(), key, QString::number(screen->geometry().width()), QString::number(screen->geometry().height()), screen->name());
+            .arg(allConnectedScreens(), key, QString::number(screen->geometry().width()), QString::number(screen->geometry().height()), screen->SCREENNAME());
     return returnString;
 }
 
@@ -126,7 +129,7 @@ void KWindowConfig::restoreWindowSize(QWindow *window, const KConfigGroup &confi
         return;
     }
 
-    const QString screenName = config.readEntry(windowScreenPositionString(), window->screen()->name());
+    const QString screenName = config.readEntry(windowScreenPositionString(), window->screen()->SCREENNAME());
     const QScreen *screen = findScreenByName(window, screenName);
 
     // Fall back to non-per-screen-arrangement info if it's available but
@@ -172,7 +175,7 @@ void KWindowConfig::saveWindowPosition(const QWindow *window, KConfigGroup &conf
     const QScreen *screen = window->screen();
     config.writeEntry(windowXPositionString(screen), window->x(), options);
     config.writeEntry(windowYPositionString(screen), window->y(), options);
-    config.writeEntry(windowScreenPositionString(), screen->name(), options);
+    config.writeEntry(windowScreenPositionString(), screen->SCREENNAME(), options);
 }
 
 void KWindowConfig::restoreWindowPosition(QWindow *window, const KConfigGroup &config)
@@ -193,8 +196,8 @@ void KWindowConfig::restoreWindowPosition(QWindow *window, const KConfigGroup &c
     }
 
     // Move window to proper screen
-    const QString screenName = config.readEntry(windowScreenPositionString(), screen->name());
-    if (screenName != screen->name()) {
+    const QString screenName = config.readEntry(windowScreenPositionString(), screen->SCREENNAME());
+    if (screenName != screen->SCREENNAME()) {
         QScreen *screenConf = findScreenByName(window, screenName);
         window->setScreen(screenConf);
         restoreWindowScreenPosition(window, screenConf, config);
