From 166d1aab0af09f08981deff77334edc25077ac9e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20K=C3=BCnzel?= <julius.kuenzel@kde.org>
Date: Wed, 16 Oct 2024 11:45:33 +0000
Subject: [PATCH] Add namespace for Android as required by newer gradle

---
 src/android/build.gradle | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/android/build.gradle b/src/android/build.gradle
index 63230b96..12b89465 100644
--- a/src/android/build.gradle
+++ b/src/android/build.gradle
@@ -34,5 +34,6 @@ android {
     defaultConfig {
         minSdkVersion @ANDROID_API_LEVEL@
         targetSdkVersion @ANDROID_SDK_COMPILE_API@
+        namespace "org.kde.knotifications"
     }
 }
-- 
GitLab

