diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9b2c70a..0cded49 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -98,7 +98,7 @@ remove_definitions(-DQT_NO_CAST_FROM_BYTEARRAY)
 if (IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/po")
     ecm_install_po_files_as_qm(po)
 endif()
-add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050d00)
+#add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050d00)
 add_subdirectory(src)
 if (BUILD_TESTING)
     add_subdirectory(tests)
