diff --git a/src/qml/CMakeLists.txt b/src/qml/CMakeLists.txt
index 1d32fbf53b3b8fcdbe9093083507899889a60b76..6539a6aa816dc7f0566db469cb83f72544191717 100644
--- a/src/qml/CMakeLists.txt
+++ b/src/qml/CMakeLists.txt
@@ -10,5 +10,9 @@ target_link_libraries(knotificationqmlplugin PRIVATE
     Qt${QT_MAJOR_VERSION}::Qml
 )
 
+if(MINGW)
+    set_target_properties(knotificationqmlplugin PROPERTIES PREFIX "")
+endif()
+
 install(TARGETS knotificationqmlplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/notification)
 install(FILES qmldir DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/notification)
