diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' libgpg-error-1.31.orig/lang/cl/mkerrcodes.awk libgpg-error-1.31/lang/cl/mkerrcodes.awk
--- libgpg-error-1.31.orig/lang/cl/mkerrcodes.awk	2013-03-15 20:24:25.000000000 +0100
+++ libgpg-error-1.31/lang/cl/mkerrcodes.awk	2019-11-19 22:26:58.393237700 +0100
@@ -122,7 +122,7 @@
 }
 
 !header {
-  sub (/\#.+/, "");
+  sub (/#.+/, "");
   sub (/[ 	]+$/, ""); # Strip trailing space and tab characters.
 
   if (/^$/)
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' libgpg-error-1.31.orig/src/Makefile.am libgpg-error-1.31/src/Makefile.am
--- libgpg-error-1.31.orig/src/Makefile.am	2018-03-22 17:43:22.000000000 +0100
+++ libgpg-error-1.31/src/Makefile.am	2019-11-19 22:28:19.412719100 +0100
@@ -281,7 +281,7 @@
 
 errnos-sym.h: Makefile mkstrtable.awk errnos.in
 	$(AWK) -f $(srcdir)/mkstrtable.awk -v textidx=2 -v nogettext=1 \
-		-v prefix=GPG_ERR_ -v namespace=errnos_ \
+		-v prefix=GPG_ERR_ -v pkg_namespace=errnos_ \
 		$(srcdir)/errnos.in >$@
 
 
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' libgpg-error-1.31.orig/src/mkerrcodes.awk libgpg-error-1.31/src/mkerrcodes.awk
--- libgpg-error-1.31.orig/src/mkerrcodes.awk	2013-03-15 20:24:25.000000000 +0100
+++ libgpg-error-1.31/src/mkerrcodes.awk	2019-11-19 22:26:58.400236600 +0100
@@ -85,7 +85,7 @@
 }
 
 !header {
-  sub (/\#.+/, "");
+  sub (/#.+/, "");
   sub (/[ 	]+$/, ""); # Strip trailing space and tab characters.
 
   if (/^$/)
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' libgpg-error-1.31.orig/src/mkerrcodes1.awk libgpg-error-1.31/src/mkerrcodes1.awk
--- libgpg-error-1.31.orig/src/mkerrcodes1.awk	2013-03-15 20:24:25.000000000 +0100
+++ libgpg-error-1.31/src/mkerrcodes1.awk	2019-11-19 22:26:58.402239100 +0100
@@ -81,7 +81,7 @@
 }
 
 !header {
-  sub (/\#.+/, "");
+  sub (/#.+/, "");
   sub (/[ 	]+$/, ""); # Strip trailing space and tab characters.
 
   if (/^$/)
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' libgpg-error-1.31.orig/src/mkerrcodes2.awk libgpg-error-1.31/src/mkerrcodes2.awk
--- libgpg-error-1.31.orig/src/mkerrcodes2.awk	2013-03-15 20:24:25.000000000 +0100
+++ libgpg-error-1.31/src/mkerrcodes2.awk	2019-11-19 22:26:58.404249900 +0100
@@ -91,7 +91,7 @@
 }
 
 !header {
-  sub (/\#.+/, "");
+  sub (/#.+/, "");
   sub (/[ 	]+$/, ""); # Strip trailing space and tab characters.
 
   if (/^$/)
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' libgpg-error-1.31.orig/src/mkerrnos.awk libgpg-error-1.31/src/mkerrnos.awk
--- libgpg-error-1.31.orig/src/mkerrnos.awk	2013-03-15 20:24:25.000000000 +0100
+++ libgpg-error-1.31/src/mkerrnos.awk	2019-11-19 22:26:58.406237500 +0100
@@ -83,7 +83,7 @@
 }
 
 !header {
-  sub (/\#.+/, "");
+  sub (/#.+/, "");
   sub (/[ 	]+$/, ""); # Strip trailing space and tab characters.
 
   if (/^$/)
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' libgpg-error-1.31.orig/src/mkstrtable.awk libgpg-error-1.31/src/mkstrtable.awk
--- libgpg-error-1.31.orig/src/mkstrtable.awk	2013-03-15 20:24:25.000000000 +0100
+++ libgpg-error-1.31/src/mkstrtable.awk	2019-11-19 22:26:58.408245500 +0100
@@ -77,7 +77,7 @@
 #
 # The variable prefix can be used to prepend a string to each message.
 #
-# The variable namespace can be used to prepend a string to each
+# The variable pkg_namespace can be used to prepend a string to each
 # variable and macro name.
 
 BEGIN {
@@ -102,7 +102,7 @@
       print "/* The purpose of this complex string table is to produce";
       print "   optimal code with a minimum of relocations.  */";
       print "";
-      print "static const char " namespace "msgstr[] = ";
+      print "static const char " pkg_namespace "msgstr[] = ";
       header = 0;
     }
   else
@@ -110,7 +110,7 @@
 }
 
 !header {
-  sub (/\#.+/, "");
+  sub (/#.+/, "");
   sub (/[ 	]+$/, ""); # Strip trailing space and tab characters.
 
   if (/^$/)
@@ -150,7 +150,7 @@
   else
     print "  gettext_noop (\"" last_msgstr "\");";
   print "";
-  print "static const int " namespace "msgidx[] =";
+  print "static const int " pkg_namespace "msgidx[] =";
   print "  {";
   for (i = 0; i < coded_msgs; i++)
     print "    " pos[i] ",";
@@ -158,7 +158,7 @@
   print "  };";
   print "";
   print "static GPG_ERR_INLINE int";
-  print namespace "msgidxof (int code)";
+  print pkg_namespace "msgidxof (int code)";
   print "{";
   print "  return (0 ? 0";
 
