diff --git a/doc/Makefile.am b/doc/Makefile.am
index 76af5be..6f3e5a1 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -49,10 +49,15 @@ CLEANFILES += yat2m-for-build
 yat2m-for-build: yat2m.c
        $(CC_FOR_BUILD) -o $@ $(srcdir)/yat2m.c
 else
+if HAVE_W32_SYSTEM
+YAT2M_CMD = ./yat2m.exe
+YAT2M_DEP = yat2m.exe
+else
 YAT2M_CMD = ./yat2m
 YAT2M_DEP = yat2m
 endif
 endif
+endif
 
 yat2m-stamp: $(myman_sources) $(srcdir)/version.texi
        @rm -f yat2m-stamp.tmp