diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' libgpg-error-1.31.orig/doc/Makefile.am libgpg-error-1.31/doc/Makefile.am
--- libgpg-error-1.31.orig/doc/Makefile.am	2018-04-23 12:52:11.000000000 +0200
+++ libgpg-error-1.31/doc/Makefile.am	2018-05-03 14:02:24.484222500 +0200
@@ -49,10 +49,15 @@
 yat2m-for-build: yat2m.c
 	$(CC_FOR_BUILD) -o $@ $(srcdir)/yat2m.c
 else
+if HAVE_W32_SYSTEM
+YAT2M_CMD = ./yat2m.exe
+YAT2M_DEP = yat2m.exe
+else
 YAT2M_CMD = ./yat2m
 YAT2M_DEP = yat2m
 endif
 endif
+endif

 yat2m-stamp: $(myman_sources) $(srcdir)/version.texi
 	@rm -f yat2m-stamp.tmp
