diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' qwt-6.1.3.orig/qwtbuild.pri qwt-6.1.3/qwtbuild.pri
--- qwt-6.1.3.orig/qwtbuild.pri	2016-06-13 09:14:23.000000000 +0200
+++ qwt-6.1.3/qwtbuild.pri	2018-04-29 14:48:27.997829100 +0200
@@ -26,9 +26,6 @@
     # you need a release version. For your own application development you
     # might need a debug version. 
     # Enable debug_and_release + build_all if you want to build both.
-
-    CONFIG           += debug_and_release
-    CONFIG           += build_all
 }
 else {
 
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' qwt-6.1.3.orig/qwtconfig.pri qwt-6.1.3/qwtconfig.pri
--- qwt-6.1.3.orig/qwtconfig.pri	2016-06-13 09:15:20.000000000 -0400
+++ qwt-6.1.3/qwtconfig.pri	2018-06-08 12:18:14.044001961 -0400
@@ -18,16 +18,6 @@
 
 QWT_INSTALL_PREFIX = $$[QT_INSTALL_PREFIX]
 
-unix {
-    QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION
-    # QWT_INSTALL_PREFIX = /usr/local/qwt-$$QWT_VERSION-qt-$$QT_VERSION
-}
-
-win32 {
-    QWT_INSTALL_PREFIX    = C:/Qwt-$$QWT_VERSION
-    # QWT_INSTALL_PREFIX = C:/Qwt-$$QWT_VERSION-qt-$$QT_VERSION
-}
-
 QWT_INSTALL_DOCS      = $${QWT_INSTALL_PREFIX}/doc
 QWT_INSTALL_HEADERS   = $${QWT_INSTALL_PREFIX}/include
 QWT_INSTALL_LIBS      = $${QWT_INSTALL_PREFIX}/lib
@@ -159,11 +149,6 @@
 # to link frameworks instead of regular libs
 ######################################################################
 
-macx:!static:CONFIG(qt_framework, qt_framework|qt_no_framework) {
-
-    QWT_CONFIG += QwtFramework
-}  
-
 ######################################################################
 # Create and install pc files for pkg-config
 # See http://www.freedesktop.org/wiki/Software/pkg-config/
 
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' qwt-6.1.3.orig/qwtfunctions.pri qwt-6.1.3/qwtfunctions.pri
--- qwt-6.1.3.orig/qwtfunctions.pri	2016-06-13 09:15:20.000000000 -0400
+++ qwt-6.1.3/qwtfunctions.pri	2018-06-08 12:48:55.689530162 -0400
@@ -14,12 +14,6 @@
     unset(LIBRARY_NAME)
     LIBRARY_NAME = $$1
 
-    mac:contains(QWT_CONFIG, QwtFramework) {
-
-        QMAKE_FRAMEWORK_BUNDLE_NAME = $$LIBRARY_NAME
-        export(QMAKE_FRAMEWORK_BUNDLE_NAME)
-    }
-
     contains(TEMPLATE, .*lib):CONFIG(debug, debug|release) {
 
         !debug_and_release|build_pass {
@@ -38,11 +32,6 @@
     LIB_PATH = $$1
     LIB_NAME = $$2
 
-    mac:contains(QWT_CONFIG, QwtFramework) {
-
-        LIBS      *= -F$${LIB_PATH}
-    }
-    else {
 
         unix:lessThan(QT_MAJOR_VERSION, 5) {
 
@@ -57,15 +46,9 @@
         else {
             LIBS *= -L$${LIB_PATH}
         }
-    }
 
     unset(LINKAGE)
 
-    mac:contains(QWT_CONFIG, QwtFramework) {
-
-        LINKAGE = -framework $${LIB_NAME}
-    }
-
     isEmpty(LINKAGE) {
 
         if(!debug_and_release|build_pass):CONFIG(debug, debug|release) {

diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' qwt-6.1.3.orig/qwt.prf qwt-6.1.3/qwt.prf
--- qwt-6.1.3.orig/qwt.prf	2016-06-13 09:15:20.000000000 -0400
+++ qwt-6.1.3/qwt.prf	2018-06-08 12:55:40.693912888 -0400
@@ -24,14 +24,7 @@
 	DEFINES *= QWT_NO_SVG
 }
 
-contains(QWT_CONFIG, QwtFramework) {
-
-    INCLUDEPATH *= $${QWT_INSTALL_LIBS}/qwt.framework/Headers
-}
-else {
-
-    INCLUDEPATH *= $${QWT_INSTALL_HEADERS}
-}   
+INCLUDEPATH *= $${QWT_INSTALL_HEADERS}
 
 # QMAKE_RPATHDIR *= $${QWT_INSTALL_LIBS}
 qwtAddLibrary($${QWT_INSTALL_LIBS}, qwt)
