diff -Nrub -x '*~' -x '*\.orig' -x'*\.o' ghostscript-9.19/psi/iapi.h.orig ghostscript-9.19/psi/iapi.h
--- ghostscript-9.19/psi/iapi.h.orig    2016-03-23 11:22:48.000000000 +0300
+++ ghostscript-9.19/psi/iapi.h 2017-08-31 21:12:46.699912800 +0300
@@ -50,7 +50,7 @@
 extern "C" {
 #endif

-#if defined(_WINDOWS_) || defined(__WINDOWS__)
+#if defined(_WIN32) || defined(__WINDOWS__)
 # ifndef _Windows
 #  define _Windows
 # endif
