diff -Nrub -x '*~' -x '*.rej' -x '*.orig' -x '*.o' -x '*.pyc' -x CMakeLists.txt.user "C:\\Users\\hanna\\AppData\\Local\\Temp\\tmptcdmjf6r\\libass-0.17.3/libass/ass.c" "D:\\_\\3c2d9ed3\\libass-0.17.3/libass/ass.c"
--- "libass-0.17.3/libass/ass.c"	2024-07-02 22:59:48.000000000 +0200
+++ "libass/ass.c"	2025-03-31 23:45:04.213805100 +0200
@@ -1398,7 +1398,7 @@

         while (1) {
             if (ileft)
-                rc = iconv(icdsc, &ip, &ileft, &op, &oleft);
+                rc = iconv(icdsc, (const char**)&ip, &ileft, &op, &oleft);
             else {              // clear the conversion state and leave
                 clear = 1;
                 rc = iconv(icdsc, NULL, NULL, &op, &oleft);
