diff --git a/linuxdeploy-plugin-checkrt.sh b/linuxdeploy-plugin-checkrt.sh
index f1edc06..9ce6505 100644
--- a/linuxdeploy-plugin-checkrt.sh
+++ b/linuxdeploy-plugin-checkrt.sh
@@ -1514,7 +1514,7 @@ if [ -x "$APPDIR/checkrt/checkrt" ]; then
 
     # prepend to LD_LIBRARY_PATH
     if [ -n "$CHECKRT_LIBS" ]; then
-        export LD_LIBRARY_PATH="${CHECKRT_LIBS}${LD_LIBRARY_PATH}"
+        export LD_LIBRARY_PATH="${CHECKRT_LIBS}:${LD_LIBRARY_PATH}"
     fi
 fi
 
