diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user qtkeychain-0.15.0.orig/qtkeychain/CMakeLists.txt qtkeychain-0.15.0/qtkeychain/CMakeLists.txt
--- qtkeychain-0.15.0.orig/qtkeychain/CMakeLists.txt	2025-01-16 12:53:01.000000000 +0100
+++ qtkeychain-0.15.0/qtkeychain/CMakeLists.txt	2025-02-01 10:31:39.873310500 +0100
@@ -28,10 +28,12 @@
     #FIXME: mingw bug; otherwise getting undefined refs to RtlSecureZeroMemory there
     if(MINGW)
         add_definitions( -O2 )
+    else()
+        add_definitions( /utf-8 )
     endif()
 
     set(CMAKE_CXX_STANDARD 17)
-    add_definitions( /utf-8 -DUNICODE )
+    add_definitions( -DUNICODE )
 endif()
 
 if(APPLE)
