diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user qtkeychain-0.12.0.orig/CMakeLists.txt qtkeychain-0.12.0/CMakeLists.txt
--- qtkeychain-0.12.0.orig/CMakeLists.txt	2020-12-16 16:43:48.000000000 +0100
+++ qtkeychain-0.12.0/CMakeLists.txt	2021-01-28 15:37:56.756904323 +0100
@@ -182,6 +182,7 @@
         endif()
         INCLUDE_DIRECTORIES(${LIBSECRET_INCLUDE_DIRS})
         list(APPEND qtkeychain_LIBRARIES_PRIVATE ${LIBSECRET_LIBRARIES})
+        LINK_DIRECTORIES(${LIBSECRET_LIBRARY_DIRS})
     endif()
 
     add_definitions(-DKEYCHAIN_DBUS=1)
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user qtkeychain-0.12.0.orig/translations/qtkeychain_de.ts qtkeychain-0.12.0/translations/qtkeychain_de.ts
--- qtkeychain-0.12.0.orig/translations/qtkeychain_de.ts	2020-12-16 16:43:48.000000000 +0100
+++ qtkeychain-0.12.0/translations/qtkeychain_de.ts	2021-01-28 15:24:10.377122674 +0100
@@ -1,177 +1,337 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE TS>
-<TS version="2.0" language="de_DE">
+<TS version="2.1" language="de_DE">
+<context>
+    <name>QKeychain::DeletePasswordJobPrivate</name>
+    <message>
+        <location filename="../keychain_win.cpp" line="104"/>
+        <source>Password entry not found</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="108"/>
+        <source>Could not decrypt data</source>
+        <translation type="unfinished">Kann Daten nicht entschlüsseln</translation>
+    </message>
+    <message>
+        <location filename="../keychain_unix.cpp" line="548"/>
+        <location filename="../keychain_unix.cpp" line="556"/>
+        <source>Unknown error</source>
+        <translation type="unfinished">Unbekannter Fehler</translation>
+    </message>
+    <message>
+        <location filename="../keychain_unix.cpp" line="574"/>
+        <source>Could not open wallet: %1; %2</source>
+        <translation type="unfinished">Konnte Brieftasche nicht öffnen: %1; %2</translation>
+    </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="177"/>
+        <source>Password not found</source>
+        <translation type="unfinished">Passwort nicht gefunden</translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="165"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="171"/>
+        <source>Could not remove private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+</context>
+<context>
+    <name>QKeychain::JobPrivate</name>
+    <message>
+        <location filename="../keychain_unix.cpp" line="265"/>
+        <source>Unknown error</source>
+        <translation type="unfinished">Unbekannter Fehler</translation>
+    </message>
+    <message>
+        <location filename="../keychain_unix.cpp" line="509"/>
+        <source>Access to keychain denied</source>
+        <translation type="unfinished">Zugriff auf Schlüsselbund verweigert</translation>
+    </message>
+</context>
+<context>
+    <name>QKeychain::PlainTextStore</name>
+    <message>
+        <location filename="../plaintextstore.cpp" line="65"/>
+        <source>Could not store data in settings: access error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="67"/>
+        <source>Could not store data in settings: format error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="85"/>
+        <source>Could not delete data from settings: access error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="87"/>
+        <source>Could not delete data from settings: format error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="104"/>
+        <source>Entry not found</source>
+        <translation type="unfinished">Eintrag nicht gefunden</translation>
+    </message>
+</context>
 <context>
     <name>QKeychain::ReadPasswordJobPrivate</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="119"/>
+        <location filename="../keychain_unix.cpp" line="187"/>
+        <location filename="../keychain_unix.cpp" line="197"/>
         <source>Unknown error</source>
         <translation>Unbekannter Fehler</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="133"/>
+        <location filename="../keychain_unix.cpp" line="178"/>
         <source>D-Bus is not running</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="210"/>
+        <location filename="../keychain_unix.cpp" line="286"/>
         <source>No keychain service available</source>
         <translation>Kein Schlüsselbund-Dienst verfügbar</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="212"/>
+        <location filename="../keychain_unix.cpp" line="288"/>
         <source>Could not open wallet: %1; %2</source>
         <translation>Konnte Brieftasche nicht öffnen: %1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="258"/>
+        <location filename="../keychain_unix.cpp" line="333"/>
         <source>Access to keychain denied</source>
         <translation>Zugriff auf Schlüsselbund verweigert</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="279"/>
+        <location filename="../keychain_unix.cpp" line="354"/>
         <source>Could not determine data type: %1; %2</source>
         <translation>Datentyp kann nicht ermittelt werden: %1: %2</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="297"/>
+        <location filename="../keychain_unix.cpp" line="372"/>
         <source>Unsupported entry type &apos;Map&apos;</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="300"/>
+        <location filename="../keychain_unix.cpp" line="375"/>
         <source>Unknown kwallet entry type &apos;%1&apos;</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="315"/>
         <source>Could not read password: %1; %2</source>
-        <translation>Passwort konnte nicht ausgelesen werden: %1; %2</translation>
+        <translation type="vanished">Passwort konnte nicht ausgelesen werden: %1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_mac.cpp" line="76"/>
+        <location filename="../keychain_mac.cpp" line="75"/>
+        <location filename="../keychain_haiku.cpp" line="96"/>
         <source>Password not found</source>
         <translation>Passwort nicht gefunden</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="288"/>
-        <location filename="../keychain_win.cpp" line="27"/>
+        <location filename="../keychain_unix.cpp" line="363"/>
+        <location filename="../keychain_android.cpp" line="50"/>
         <source>Entry not found</source>
         <translation>Eintrag nicht gefunden</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="44"/>
+        <location filename="../keychain_win.cpp" line="32"/>
+        <source>Password entry not found</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="36"/>
+        <location filename="../keychain_win.cpp" line="139"/>
         <source>Could not decrypt data</source>
         <translation>Kann Daten nicht entschlüsseln</translation>
     </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="58"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="66"/>
+        <source>Could not retrieve private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="73"/>
+        <source>Could not create decryption cipher</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 <context>
     <name>QKeychain::WritePasswordJobPrivate</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="336"/>
-        <location filename="../keychain_unix.cpp" line="344"/>
+        <location filename="../keychain_unix.cpp" line="425"/>
+        <location filename="../keychain_unix.cpp" line="452"/>
         <source>Unknown error</source>
         <translation>Unbekannter Fehler</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="359"/>
+        <location filename="../keychain_unix.cpp" line="415"/>
         <source>D-Bus is not running</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="400"/>
-        <location filename="../keychain_unix.cpp" line="485"/>
+        <location filename="../keychain_unix.cpp" line="468"/>
         <source>Could not open wallet: %1; %2</source>
         <translation>Konnte Brieftasche nicht öffnen: %1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="463"/>
         <source>Access to keychain denied</source>
-        <translation>Zugriff auf Schlüsselbund verweigert</translation>
+        <translation type="vanished">Zugriff auf Schlüsselbund verweigert</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="64"/>
         <source>Could not delete encrypted data from settings: access error</source>
-        <translation>Kann verschlüsselte Daten nicht aus den Einstellungen entfernen: Zugriffsfehler</translation>
+        <translation type="vanished">Kann verschlüsselte Daten nicht aus den Einstellungen entfernen: Zugriffsfehler</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="65"/>
         <source>Could not delete encrypted data from settings: format error</source>
-        <translation>Kann verschlüsselte Daten nicht aus den Einstellungen entfernen: Formatfehler</translation>
+        <translation type="vanished">Kann verschlüsselte Daten nicht aus den Einstellungen entfernen: Formatfehler</translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="78"/>
+        <source>Credential size exceeds maximum size of %1</source>
+        <translation type="unfinished"></translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="85"/>
+        <location filename="../keychain_win.cpp" line="87"/>
+        <source>Credential key exceeds maximum size of %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="92"/>
+        <source>Writing credentials failed: Win32 error code %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="162"/>
         <source>Encryption failed</source>
         <translation>Verschlüsselung fehlgeschlagen</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="100"/>
         <source>Could not store encrypted data in settings: access error</source>
-        <translation>Kann verschlüsselte Daten nicht in den Einstellungen speichern: Zugriffsfehler</translation>
+        <translation type="vanished">Kann verschlüsselte Daten nicht in den Einstellungen speichern: Zugriffsfehler</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="101"/>
         <source>Could not store encrypted data in settings: format error</source>
-        <translation>Kann verschlüsselte Daten nicht in den Einstellungen speichern: Formatfehler</translation>
+        <translation type="vanished">Kann verschlüsselte Daten nicht in den Einstellungen speichern: Formatfehler</translation>
+    </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="144"/>
+        <source>Password not found</source>
+        <translation type="unfinished">Passwort nicht gefunden</translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="93"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="116"/>
+        <source>Could not create private key generator</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="123"/>
+        <source>Could not generate new private key</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="131"/>
+        <source>Could not retrieve private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="139"/>
+        <source>Could not create encryption cipher</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="147"/>
+        <source>Could not encrypt data</source>
+        <translation type="unfinished"></translation>
     </message>
 </context>
 <context>
     <name>QObject</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="155"/>
+        <location filename="../keychain_unix.cpp" line="225"/>
         <source>Access to keychain denied</source>
         <translation>Zugriff auf Schlüsselbund verweigert</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="157"/>
+        <location filename="../keychain_unix.cpp" line="227"/>
         <source>No keyring daemon</source>
         <translation>Kein Schlüsselbund-Dienst </translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="159"/>
+        <location filename="../keychain_unix.cpp" line="229"/>
         <source>Already unlocked</source>
         <translation>Bereits entsperrt</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="161"/>
+        <location filename="../keychain_unix.cpp" line="231"/>
         <source>No such keyring</source>
         <translation>Kein solcher Schlüsselbund</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="163"/>
+        <location filename="../keychain_unix.cpp" line="233"/>
         <source>Bad arguments</source>
         <translation>Ungültige Argumente</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="165"/>
+        <location filename="../keychain_unix.cpp" line="235"/>
         <source>I/O error</source>
         <translation>Ein-/Ausgabe-Fehler</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="167"/>
+        <location filename="../keychain_unix.cpp" line="237"/>
         <source>Cancelled</source>
         <translation>Abgebrochen</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="169"/>
+        <location filename="../keychain_unix.cpp" line="239"/>
         <source>Keyring already exists</source>
         <translation>Schlüsselbund existiert bereits</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="171"/>
+        <location filename="../keychain_unix.cpp" line="241"/>
         <source>No match</source>
         <translation>Kein Treffer</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="176"/>
+        <location filename="../keychain_unix.cpp" line="246"/>
         <source>Unknown error</source>
         <translation>Unbekannter Fehler</translation>
     </message>
     <message>
         <location filename="../keychain_mac.cpp" line="31"/>
-        <location filename="../keychain_mac.cpp" line="33"/>
+        <source>OS X Keychain error (OSStatus %1)</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_mac.cpp" line="32"/>
         <source>%1 (OSStatus %2)</source>
         <translation type="unfinished"></translation>
     </message>
+    <message>
+        <location filename="../libsecret.cpp" line="120"/>
+        <source>Entry not found</source>
+        <translation type="unfinished">Eintrag nicht gefunden</translation>
+    </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="72"/>
+        <source>error 0x%1: %2</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 </TS>
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user qtkeychain-0.12.0.orig/translations/qtkeychain_fr.ts qtkeychain-0.12.0/translations/qtkeychain_fr.ts
--- qtkeychain-0.12.0.orig/translations/qtkeychain_fr.ts	2020-12-16 16:43:48.000000000 +0100
+++ qtkeychain-0.12.0/translations/qtkeychain_fr.ts	2021-01-28 15:24:10.377122674 +0100
@@ -4,41 +4,51 @@
 <context>
     <name>QKeychain::DeletePasswordJobPrivate</name>
     <message>
-        <location filename="../keychain_win.cpp" line="79"/>
+        <location filename="../keychain_win.cpp" line="104"/>
         <source>Password entry not found</source>
         <translation>Mot de passe introuvable</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="83"/>
+        <location filename="../keychain_win.cpp" line="108"/>
         <source>Could not decrypt data</source>
         <translation>Impossible de déchiffrer les données</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="546"/>
-        <location filename="../keychain_unix.cpp" line="554"/>
+        <location filename="../keychain_unix.cpp" line="548"/>
+        <location filename="../keychain_unix.cpp" line="556"/>
         <source>Unknown error</source>
         <translation>Erreur inconnue</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="572"/>
+        <location filename="../keychain_unix.cpp" line="574"/>
         <source>Could not open wallet: %1; %2</source>
         <translation>Impossible d&apos;ouvrir le portefeuille : %1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_haiku.cpp" line="120"/>
+        <location filename="../keychain_haiku.cpp" line="177"/>
         <source>Password not found</source>
         <translation>Mot de passe introuvable</translation>
     </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="165"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="171"/>
+        <source>Could not remove private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 <context>
     <name>QKeychain::JobPrivate</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="263"/>
+        <location filename="../keychain_unix.cpp" line="265"/>
         <source>Unknown error</source>
         <translation>Erreur inconnue</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="507"/>
+        <location filename="../keychain_unix.cpp" line="509"/>
         <source>Access to keychain denied</source>
         <translation>Accès au trousseau refusé</translation>
     </message>
@@ -74,43 +84,43 @@
 <context>
     <name>QKeychain::ReadPasswordJobPrivate</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="185"/>
-        <location filename="../keychain_unix.cpp" line="195"/>
+        <location filename="../keychain_unix.cpp" line="187"/>
+        <location filename="../keychain_unix.cpp" line="197"/>
         <source>Unknown error</source>
         <translation>Erreur inconnue</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="176"/>
+        <location filename="../keychain_unix.cpp" line="178"/>
         <source>D-Bus is not running</source>
         <translation>D-Bus n&apos;est pas en cours d&apos;exécution</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="284"/>
+        <location filename="../keychain_unix.cpp" line="286"/>
         <source>No keychain service available</source>
         <translation>Aucun service de trousseau disponible</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="286"/>
+        <location filename="../keychain_unix.cpp" line="288"/>
         <source>Could not open wallet: %1; %2</source>
         <translation>Impossible d&apos;ouvrir le trousseau : %1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="331"/>
+        <location filename="../keychain_unix.cpp" line="333"/>
         <source>Access to keychain denied</source>
         <translation>Accès au trousseau refusé</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="352"/>
+        <location filename="../keychain_unix.cpp" line="354"/>
         <source>Could not determine data type: %1; %2</source>
         <translation>Impossible de déterminer le type de données : %1: %2</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="370"/>
+        <location filename="../keychain_unix.cpp" line="372"/>
         <source>Unsupported entry type &apos;Map&apos;</source>
         <translation>Type d&apos;entrée non supporté &apos;Map&apos;</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="373"/>
+        <location filename="../keychain_unix.cpp" line="375"/>
         <source>Unknown kwallet entry type &apos;%1&apos;</source>
         <translation>Type de trousseau inconnu &apos;%1&apos;</translation>
     </message>
@@ -120,12 +130,13 @@
     </message>
     <message>
         <location filename="../keychain_mac.cpp" line="75"/>
-        <location filename="../keychain_haiku.cpp" line="41"/>
+        <location filename="../keychain_haiku.cpp" line="96"/>
         <source>Password not found</source>
         <translation>Mot de passe introuvable</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="361"/>
+        <location filename="../keychain_unix.cpp" line="363"/>
+        <location filename="../keychain_android.cpp" line="50"/>
         <source>Entry not found</source>
         <translation>Entrée introuvable</translation>
     </message>
@@ -136,26 +147,41 @@
     </message>
     <message>
         <location filename="../keychain_win.cpp" line="36"/>
-        <location filename="../keychain_win.cpp" line="114"/>
+        <location filename="../keychain_win.cpp" line="139"/>
         <source>Could not decrypt data</source>
         <translation>Impossible de déchiffrer les données</translation>
     </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="58"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="66"/>
+        <source>Could not retrieve private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="73"/>
+        <source>Could not create decryption cipher</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 <context>
     <name>QKeychain::WritePasswordJobPrivate</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="423"/>
-        <location filename="../keychain_unix.cpp" line="450"/>
+        <location filename="../keychain_unix.cpp" line="425"/>
+        <location filename="../keychain_unix.cpp" line="452"/>
         <source>Unknown error</source>
         <translation>Erreur inconnue</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="413"/>
+        <location filename="../keychain_unix.cpp" line="415"/>
         <source>D-Bus is not running</source>
         <translation>D-Bus n&apos;est pas en cours d&apos;exécution</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="466"/>
+        <location filename="../keychain_unix.cpp" line="468"/>
         <source>Could not open wallet: %1; %2</source>
         <translation>Impossible d&apos;ouvrir le trousseau : %1; %2</translation>
     </message>
@@ -172,8 +198,22 @@
         <translation type="vanished">Impossible de supprimer des données chiffrées dans les paramètres : Erreur de format</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="64"/>
-        <location filename="../keychain_win.cpp" line="137"/>
+        <location filename="../keychain_win.cpp" line="78"/>
+        <source>Credential size exceeds maximum size of %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="87"/>
+        <source>Credential key exceeds maximum size of %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="92"/>
+        <source>Writing credentials failed: Win32 error code %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="162"/>
         <source>Encryption failed</source>
         <translation>Le chiffrement a échoué</translation>
     </message>
@@ -186,60 +226,90 @@
         <translation type="vanished">Impossible de stocker des données chiffrées dans les paramètres : Erreur de format</translation>
     </message>
     <message>
-        <location filename="../keychain_haiku.cpp" line="88"/>
+        <location filename="../keychain_haiku.cpp" line="144"/>
         <source>Password not found</source>
         <translation>Mot de passe introuvable</translation>
     </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="93"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="116"/>
+        <source>Could not create private key generator</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="123"/>
+        <source>Could not generate new private key</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="131"/>
+        <source>Could not retrieve private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="139"/>
+        <source>Could not create encryption cipher</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="147"/>
+        <source>Could not encrypt data</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 <context>
     <name>QObject</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="223"/>
+        <location filename="../keychain_unix.cpp" line="225"/>
         <source>Access to keychain denied</source>
         <translation>Accès au trousseau refusé</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="225"/>
+        <location filename="../keychain_unix.cpp" line="227"/>
         <source>No keyring daemon</source>
         <translation>Aucun démon de trousseau</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="227"/>
+        <location filename="../keychain_unix.cpp" line="229"/>
         <source>Already unlocked</source>
         <translation>Déjà déverrouillé</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="229"/>
+        <location filename="../keychain_unix.cpp" line="231"/>
         <source>No such keyring</source>
         <translation>Aucun trousseau</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="231"/>
+        <location filename="../keychain_unix.cpp" line="233"/>
         <source>Bad arguments</source>
         <translation>Mauvais arguments</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="233"/>
+        <location filename="../keychain_unix.cpp" line="235"/>
         <source>I/O error</source>
         <translation>Erreur d&apos;E/S</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="235"/>
+        <location filename="../keychain_unix.cpp" line="237"/>
         <source>Cancelled</source>
         <translation>Annulé</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="237"/>
+        <location filename="../keychain_unix.cpp" line="239"/>
         <source>Keyring already exists</source>
         <translation>Trousseau déjà existant</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="239"/>
+        <location filename="../keychain_unix.cpp" line="241"/>
         <source>No match</source>
         <translation>Aucune correspondance</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="244"/>
+        <location filename="../keychain_unix.cpp" line="246"/>
         <source>Unknown error</source>
         <translation>Erreur inconnue</translation>
     </message>
@@ -259,7 +329,7 @@
         <translation>Entrée introuvable</translation>
     </message>
     <message>
-        <location filename="../keychain_haiku.cpp" line="18"/>
+        <location filename="../keychain_haiku.cpp" line="72"/>
         <source>error 0x%1: %2</source>
         <translation>Erreur 0x%1 : %2</translation>
     </message>
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user qtkeychain-0.12.0.orig/translations/qtkeychain_ro.ts qtkeychain-0.12.0/translations/qtkeychain_ro.ts
--- qtkeychain-0.12.0.orig/translations/qtkeychain_ro.ts	2020-12-16 16:43:48.000000000 +0100
+++ qtkeychain-0.12.0/translations/qtkeychain_ro.ts	2021-01-28 15:24:10.377122674 +0100
@@ -1,178 +1,338 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE TS>
-<TS version="2.0" language="ro_RO">
+<TS version="2.1" language="ro_RO">
+<context>
+    <name>QKeychain::DeletePasswordJobPrivate</name>
+    <message>
+        <location filename="../keychain_win.cpp" line="104"/>
+        <source>Password entry not found</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="108"/>
+        <source>Could not decrypt data</source>
+        <translation type="unfinished">Nu se poate decripta data</translation>
+    </message>
+    <message>
+        <location filename="../keychain_unix.cpp" line="548"/>
+        <location filename="../keychain_unix.cpp" line="556"/>
+        <source>Unknown error</source>
+        <translation type="unfinished">Eroare necunoscută</translation>
+    </message>
+    <message>
+        <location filename="../keychain_unix.cpp" line="574"/>
+        <source>Could not open wallet: %1; %2</source>
+        <translation type="unfinished">Nu se poate deschide portofelul: %1; %2</translation>
+    </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="177"/>
+        <source>Password not found</source>
+        <translation type="unfinished">Parola nu a fost găsită</translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="165"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="171"/>
+        <source>Could not remove private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+</context>
+<context>
+    <name>QKeychain::JobPrivate</name>
+    <message>
+        <location filename="../keychain_unix.cpp" line="265"/>
+        <source>Unknown error</source>
+        <translation type="unfinished">Eroare necunoscută</translation>
+    </message>
+    <message>
+        <location filename="../keychain_unix.cpp" line="509"/>
+        <source>Access to keychain denied</source>
+        <translation type="unfinished">Acces interzis la serviciul de chei</translation>
+    </message>
+</context>
+<context>
+    <name>QKeychain::PlainTextStore</name>
+    <message>
+        <location filename="../plaintextstore.cpp" line="65"/>
+        <source>Could not store data in settings: access error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="67"/>
+        <source>Could not store data in settings: format error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="85"/>
+        <source>Could not delete data from settings: access error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="87"/>
+        <source>Could not delete data from settings: format error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="104"/>
+        <source>Entry not found</source>
+        <translation type="unfinished">Înregistrarea nu a fost găsită</translation>
+    </message>
+</context>
 <context>
     <name>QKeychain::ReadPasswordJobPrivate</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="119"/>
+        <location filename="../keychain_unix.cpp" line="187"/>
+        <location filename="../keychain_unix.cpp" line="197"/>
         <source>Unknown error</source>
         <translation>Eroare necunoscută</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="133"/>
+        <location filename="../keychain_unix.cpp" line="178"/>
         <source>D-Bus is not running</source>
         <translation>D-Bus nu rulează</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="210"/>
+        <location filename="../keychain_unix.cpp" line="286"/>
         <source>No keychain service available</source>
         <translatorcomment>Nu există niciun serviciu de chei disponibil</translatorcomment>
         <translation>Kein Schlüsselbund-Dienst verfügbar</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="212"/>
+        <location filename="../keychain_unix.cpp" line="288"/>
         <source>Could not open wallet: %1; %2</source>
         <translation>Nu se poate deschide portofelul: %1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="258"/>
+        <location filename="../keychain_unix.cpp" line="333"/>
         <source>Access to keychain denied</source>
         <translation>Acces interzis la serviciul de chei</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="279"/>
+        <location filename="../keychain_unix.cpp" line="354"/>
         <source>Could not determine data type: %1; %2</source>
         <translation>Nu se poate stabili tipul de date: %1: %2</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="297"/>
+        <location filename="../keychain_unix.cpp" line="372"/>
         <source>Unsupported entry type &apos;Map&apos;</source>
         <translation>Tip de înregistrare nesuportat &apos;Map&apos;</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="300"/>
+        <location filename="../keychain_unix.cpp" line="375"/>
         <source>Unknown kwallet entry type &apos;%1&apos;</source>
         <translation>Tip de înregistrare kwallet necunoscut &apos;%1&apos;</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="315"/>
         <source>Could not read password: %1; %2</source>
-        <translation>Nu se poate citi parola: %1; %2</translation>
+        <translation type="vanished">Nu se poate citi parola: %1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_mac.cpp" line="76"/>
+        <location filename="../keychain_mac.cpp" line="75"/>
+        <location filename="../keychain_haiku.cpp" line="96"/>
         <source>Password not found</source>
         <translation>Parola nu a fost găsită</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="288"/>
-        <location filename="../keychain_win.cpp" line="27"/>
+        <location filename="../keychain_unix.cpp" line="363"/>
+        <location filename="../keychain_android.cpp" line="50"/>
         <source>Entry not found</source>
         <translation>Înregistrarea nu a fost găsită</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="44"/>
+        <location filename="../keychain_win.cpp" line="32"/>
+        <source>Password entry not found</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="36"/>
+        <location filename="../keychain_win.cpp" line="139"/>
         <source>Could not decrypt data</source>
         <translation>Nu se poate decripta data</translation>
     </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="58"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="66"/>
+        <source>Could not retrieve private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="73"/>
+        <source>Could not create decryption cipher</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 <context>
     <name>QKeychain::WritePasswordJobPrivate</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="336"/>
-        <location filename="../keychain_unix.cpp" line="344"/>
+        <location filename="../keychain_unix.cpp" line="425"/>
+        <location filename="../keychain_unix.cpp" line="452"/>
         <source>Unknown error</source>
         <translation>Eroare necunoscută</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="359"/>
+        <location filename="../keychain_unix.cpp" line="415"/>
         <source>D-Bus is not running</source>
         <translation>D-Bus nu rulează</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="400"/>
-        <location filename="../keychain_unix.cpp" line="485"/>
+        <location filename="../keychain_unix.cpp" line="468"/>
         <source>Could not open wallet: %1; %2</source>
         <translation>Nu se poate deschide portofelul: %1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="463"/>
         <source>Access to keychain denied</source>
-        <translation>Acces interzis la serviciul de chei</translation>
+        <translation type="vanished">Acces interzis la serviciul de chei</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="64"/>
         <source>Could not delete encrypted data from settings: access error</source>
-        <translation>Nu se pot șterge datele criptate din setări: eroare de acces</translation>
+        <translation type="vanished">Nu se pot șterge datele criptate din setări: eroare de acces</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="65"/>
         <source>Could not delete encrypted data from settings: format error</source>
-        <translation>Nu se pot șterge datele criptate din setări: eroare de format</translation>
+        <translation type="vanished">Nu se pot șterge datele criptate din setări: eroare de format</translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="78"/>
+        <source>Credential size exceeds maximum size of %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="87"/>
+        <source>Credential key exceeds maximum size of %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="92"/>
+        <source>Writing credentials failed: Win32 error code %1</source>
+        <translation type="unfinished"></translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="85"/>
+        <location filename="../keychain_win.cpp" line="162"/>
         <source>Encryption failed</source>
         <translation>Criptarea a eșuat</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="100"/>
         <source>Could not store encrypted data in settings: access error</source>
-        <translation>Nu se pot stoca datele criptate în setări: eroare de acces</translation>
+        <translation type="vanished">Nu se pot stoca datele criptate în setări: eroare de acces</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="101"/>
         <source>Could not store encrypted data in settings: format error</source>
-        <translation>Nu se pot stoca datele criptate în setări: eroare de format</translation>
+        <translation type="vanished">Nu se pot stoca datele criptate în setări: eroare de format</translation>
+    </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="144"/>
+        <source>Password not found</source>
+        <translation type="unfinished">Parola nu a fost găsită</translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="93"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="116"/>
+        <source>Could not create private key generator</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="123"/>
+        <source>Could not generate new private key</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="131"/>
+        <source>Could not retrieve private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="139"/>
+        <source>Could not create encryption cipher</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="147"/>
+        <source>Could not encrypt data</source>
+        <translation type="unfinished"></translation>
     </message>
 </context>
 <context>
     <name>QObject</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="155"/>
+        <location filename="../keychain_unix.cpp" line="225"/>
         <source>Access to keychain denied</source>
         <translation>Acces interzis la serviciul de chei</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="157"/>
+        <location filename="../keychain_unix.cpp" line="227"/>
         <source>No keyring daemon</source>
         <translation>Niciun demon pentru inelul de chei</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="159"/>
+        <location filename="../keychain_unix.cpp" line="229"/>
         <source>Already unlocked</source>
         <translation>Deja deblocat</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="161"/>
+        <location filename="../keychain_unix.cpp" line="231"/>
         <source>No such keyring</source>
         <translation>Nu există astfel de inel de chei</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="163"/>
+        <location filename="../keychain_unix.cpp" line="233"/>
         <source>Bad arguments</source>
         <translation>Argumente greșite</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="165"/>
+        <location filename="../keychain_unix.cpp" line="235"/>
         <source>I/O error</source>
         <translation>Eroare de I/E</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="167"/>
+        <location filename="../keychain_unix.cpp" line="237"/>
         <source>Cancelled</source>
         <translation>Anulat</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="169"/>
+        <location filename="../keychain_unix.cpp" line="239"/>
         <source>Keyring already exists</source>
         <translation>Inelul de chei deja există</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="171"/>
+        <location filename="../keychain_unix.cpp" line="241"/>
         <source>No match</source>
         <translation>Nicio potrivire</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="176"/>
+        <location filename="../keychain_unix.cpp" line="246"/>
         <source>Unknown error</source>
         <translation>Eroare necunoscută</translation>
     </message>
     <message>
         <location filename="../keychain_mac.cpp" line="31"/>
-        <location filename="../keychain_mac.cpp" line="33"/>
+        <source>OS X Keychain error (OSStatus %1)</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_mac.cpp" line="32"/>
         <source>%1 (OSStatus %2)</source>
         <translation>%1 (OSStatus %2)</translation>
     </message>
+    <message>
+        <location filename="../libsecret.cpp" line="120"/>
+        <source>Entry not found</source>
+        <translation type="unfinished">Înregistrarea nu a fost găsită</translation>
+    </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="72"/>
+        <source>error 0x%1: %2</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 </TS>
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user qtkeychain-0.12.0.orig/translations/qtkeychain_ru.ts qtkeychain-0.12.0/translations/qtkeychain_ru.ts
--- qtkeychain-0.12.0.orig/translations/qtkeychain_ru.ts	2020-12-16 16:43:48.000000000 +0100
+++ qtkeychain-0.12.0/translations/qtkeychain_ru.ts	2021-01-28 15:24:10.377122674 +0100
@@ -24,6 +24,21 @@
         <source>Could not decrypt data</source>
         <translation>Не удалось расшифровать данные</translation>
     </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="177"/>
+        <source>Password not found</source>
+        <translation type="unfinished">Пароль не найден</translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="165"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="171"/>
+        <source>Could not remove private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 <context>
     <name>QKeychain::JobPrivate</name>
@@ -70,6 +85,7 @@
     <name>QKeychain::ReadPasswordJobPrivate</name>
     <message>
         <location filename="../keychain_mac.cpp" line="75"/>
+        <location filename="../keychain_haiku.cpp" line="96"/>
         <source>Password not found</source>
         <translation>Пароль не найден</translation>
     </message>
@@ -106,6 +122,7 @@
     </message>
     <message>
         <location filename="../keychain_unix.cpp" line="363"/>
+        <location filename="../keychain_android.cpp" line="50"/>
         <source>Entry not found</source>
         <translation>Запись не найдена</translation>
     </message>
@@ -130,6 +147,21 @@
         <source>Could not decrypt data</source>
         <translation>Не удалось расшифровать данные</translation>
     </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="58"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="66"/>
+        <source>Could not retrieve private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="73"/>
+        <source>Could not create decryption cipher</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 <context>
     <name>QKeychain::WritePasswordJobPrivate</name>
@@ -169,6 +201,41 @@
         <source>Encryption failed</source>
         <translation>Шифрование не удалось</translation>
     </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="144"/>
+        <source>Password not found</source>
+        <translation type="unfinished">Пароль не найден</translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="93"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="116"/>
+        <source>Could not create private key generator</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="123"/>
+        <source>Could not generate new private key</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="131"/>
+        <source>Could not retrieve private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="139"/>
+        <source>Could not create encryption cipher</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="147"/>
+        <source>Could not encrypt data</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 <context>
     <name>QObject</name>
@@ -237,5 +304,10 @@
         <source>Entry not found</source>
         <translation>Запись не найдена</translation>
     </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="72"/>
+        <source>error 0x%1: %2</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 </TS>
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user qtkeychain-0.12.0.orig/translations/qtkeychain_zh.ts qtkeychain-0.12.0/translations/qtkeychain_zh.ts
--- qtkeychain-0.12.0.orig/translations/qtkeychain_zh.ts	2020-12-16 16:43:48.000000000 +0100
+++ qtkeychain-0.12.0/translations/qtkeychain_zh.ts	2021-01-28 15:24:10.377122674 +0100
@@ -1,177 +1,337 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE TS>
-<TS version="2.0" language="zh_TW">
+<TS version="2.1" language="zh_TW">
+<context>
+    <name>QKeychain::DeletePasswordJobPrivate</name>
+    <message>
+        <location filename="../keychain_win.cpp" line="104"/>
+        <source>Password entry not found</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="108"/>
+        <source>Could not decrypt data</source>
+        <translation type="unfinished">無法解密資料</translation>
+    </message>
+    <message>
+        <location filename="../keychain_unix.cpp" line="548"/>
+        <location filename="../keychain_unix.cpp" line="556"/>
+        <source>Unknown error</source>
+        <translation type="unfinished">未知的錯誤</translation>
+    </message>
+    <message>
+        <location filename="../keychain_unix.cpp" line="574"/>
+        <source>Could not open wallet: %1; %2</source>
+        <translation type="unfinished">無法開啟錢包：%1; %2</translation>
+    </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="177"/>
+        <source>Password not found</source>
+        <translation type="unfinished">找不到密碼</translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="165"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="171"/>
+        <source>Could not remove private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+</context>
+<context>
+    <name>QKeychain::JobPrivate</name>
+    <message>
+        <location filename="../keychain_unix.cpp" line="265"/>
+        <source>Unknown error</source>
+        <translation type="unfinished">未知的錯誤</translation>
+    </message>
+    <message>
+        <location filename="../keychain_unix.cpp" line="509"/>
+        <source>Access to keychain denied</source>
+        <translation type="unfinished">鑰匙圈存取被拒絕</translation>
+    </message>
+</context>
+<context>
+    <name>QKeychain::PlainTextStore</name>
+    <message>
+        <location filename="../plaintextstore.cpp" line="65"/>
+        <source>Could not store data in settings: access error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="67"/>
+        <source>Could not store data in settings: format error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="85"/>
+        <source>Could not delete data from settings: access error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="87"/>
+        <source>Could not delete data from settings: format error</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../plaintextstore.cpp" line="104"/>
+        <source>Entry not found</source>
+        <translation type="unfinished">找不到項目</translation>
+    </message>
+</context>
 <context>
     <name>QKeychain::ReadPasswordJobPrivate</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="119"/>
+        <location filename="../keychain_unix.cpp" line="187"/>
+        <location filename="../keychain_unix.cpp" line="197"/>
         <source>Unknown error</source>
         <translation>未知的錯誤</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="133"/>
+        <location filename="../keychain_unix.cpp" line="178"/>
         <source>D-Bus is not running</source>
         <translation>D-Bus 不在執行中</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="210"/>
+        <location filename="../keychain_unix.cpp" line="286"/>
         <source>No keychain service available</source>
         <translation>沒有可用的鑰匙圈服務</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="212"/>
+        <location filename="../keychain_unix.cpp" line="288"/>
         <source>Could not open wallet: %1; %2</source>
         <translation>無法開啟錢包：%1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="258"/>
+        <location filename="../keychain_unix.cpp" line="333"/>
         <source>Access to keychain denied</source>
         <translation>鑰匙圈存取被拒絕</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="279"/>
+        <location filename="../keychain_unix.cpp" line="354"/>
         <source>Could not determine data type: %1; %2</source>
         <translation>無法判斷資料型別：%1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="297"/>
+        <location filename="../keychain_unix.cpp" line="372"/>
         <source>Unsupported entry type &apos;Map&apos;</source>
         <translation>不支援的項目類型 &apos;Map&apos;</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="300"/>
+        <location filename="../keychain_unix.cpp" line="375"/>
         <source>Unknown kwallet entry type &apos;%1&apos;</source>
         <translation>未知的 kwallet 項目類型 &apos;%1&apos;</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="315"/>
         <source>Could not read password: %1; %2</source>
-        <translation>無法讀取密碼：%1; %2</translation>
+        <translation type="vanished">無法讀取密碼：%1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_mac.cpp" line="76"/>
+        <location filename="../keychain_mac.cpp" line="75"/>
+        <location filename="../keychain_haiku.cpp" line="96"/>
         <source>Password not found</source>
         <translation>找不到密碼</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="288"/>
-        <location filename="../keychain_win.cpp" line="27"/>
+        <location filename="../keychain_unix.cpp" line="363"/>
+        <location filename="../keychain_android.cpp" line="50"/>
         <source>Entry not found</source>
         <translation>找不到項目</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="44"/>
+        <location filename="../keychain_win.cpp" line="32"/>
+        <source>Password entry not found</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="36"/>
+        <location filename="../keychain_win.cpp" line="139"/>
         <source>Could not decrypt data</source>
         <translation>無法解密資料</translation>
     </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="58"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="66"/>
+        <source>Could not retrieve private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="73"/>
+        <source>Could not create decryption cipher</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 <context>
     <name>QKeychain::WritePasswordJobPrivate</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="336"/>
-        <location filename="../keychain_unix.cpp" line="344"/>
+        <location filename="../keychain_unix.cpp" line="425"/>
+        <location filename="../keychain_unix.cpp" line="452"/>
         <source>Unknown error</source>
         <translation>未知的錯誤</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="359"/>
+        <location filename="../keychain_unix.cpp" line="415"/>
         <source>D-Bus is not running</source>
         <translation>D-Bus 不在執行中</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="400"/>
-        <location filename="../keychain_unix.cpp" line="485"/>
+        <location filename="../keychain_unix.cpp" line="468"/>
         <source>Could not open wallet: %1; %2</source>
         <translation>無法開啟錢包：%1; %2</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="463"/>
         <source>Access to keychain denied</source>
-        <translation>鑰匙圈存取被拒絕</translation>
+        <translation type="vanished">鑰匙圈存取被拒絕</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="64"/>
         <source>Could not delete encrypted data from settings: access error</source>
-        <translation>無法從設定刪除加密資料：存取錯誤</translation>
+        <translation type="vanished">無法從設定刪除加密資料：存取錯誤</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="65"/>
         <source>Could not delete encrypted data from settings: format error</source>
-        <translation>無法從設定刪除加密資料：格式錯誤</translation>
+        <translation type="vanished">無法從設定刪除加密資料：格式錯誤</translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="78"/>
+        <source>Credential size exceeds maximum size of %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="87"/>
+        <source>Credential key exceeds maximum size of %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_win.cpp" line="92"/>
+        <source>Writing credentials failed: Win32 error code %1</source>
+        <translation type="unfinished"></translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="85"/>
+        <location filename="../keychain_win.cpp" line="162"/>
         <source>Encryption failed</source>
         <translation>加密失敗</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="100"/>
         <source>Could not store encrypted data in settings: access error</source>
-        <translation>無法將加密資料儲存至設定：存取錯誤</translation>
+        <translation type="vanished">無法將加密資料儲存至設定：存取錯誤</translation>
     </message>
     <message>
-        <location filename="../keychain_win.cpp" line="101"/>
         <source>Could not store encrypted data in settings: format error</source>
-        <translation>無法將加密資料儲存至設定：格式錯誤</translation>
+        <translation type="vanished">無法將加密資料儲存至設定：格式錯誤</translation>
+    </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="144"/>
+        <source>Password not found</source>
+        <translation type="unfinished">找不到密碼</translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="93"/>
+        <source>Could not open keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="116"/>
+        <source>Could not create private key generator</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="123"/>
+        <source>Could not generate new private key</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="131"/>
+        <source>Could not retrieve private key from keystore</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="139"/>
+        <source>Could not create encryption cipher</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_android.cpp" line="147"/>
+        <source>Could not encrypt data</source>
+        <translation type="unfinished"></translation>
     </message>
 </context>
 <context>
     <name>QObject</name>
     <message>
-        <location filename="../keychain_unix.cpp" line="155"/>
+        <location filename="../keychain_unix.cpp" line="225"/>
         <source>Access to keychain denied</source>
         <translation>鑰匙圈存取被拒絕</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="157"/>
+        <location filename="../keychain_unix.cpp" line="227"/>
         <source>No keyring daemon</source>
         <translation>沒有可用的鑰匙圈背景程式</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="159"/>
+        <location filename="../keychain_unix.cpp" line="229"/>
         <source>Already unlocked</source>
         <translation>已解鎖</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="161"/>
+        <location filename="../keychain_unix.cpp" line="231"/>
         <source>No such keyring</source>
         <translation>鑰匙圈不存在</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="163"/>
+        <location filename="../keychain_unix.cpp" line="233"/>
         <source>Bad arguments</source>
         <translation>引數錯誤</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="165"/>
+        <location filename="../keychain_unix.cpp" line="235"/>
         <source>I/O error</source>
         <translation>I/O 錯誤</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="167"/>
+        <location filename="../keychain_unix.cpp" line="237"/>
         <source>Cancelled</source>
         <translation>已取消</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="169"/>
+        <location filename="../keychain_unix.cpp" line="239"/>
         <source>Keyring already exists</source>
         <translation>鑰匙圈已存在</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="171"/>
+        <location filename="../keychain_unix.cpp" line="241"/>
         <source>No match</source>
         <translation>無相符項目</translation>
     </message>
     <message>
-        <location filename="../keychain_unix.cpp" line="176"/>
+        <location filename="../keychain_unix.cpp" line="246"/>
         <source>Unknown error</source>
         <translation>未知的錯誤</translation>
     </message>
     <message>
         <location filename="../keychain_mac.cpp" line="31"/>
-        <location filename="../keychain_mac.cpp" line="33"/>
+        <source>OS X Keychain error (OSStatus %1)</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../keychain_mac.cpp" line="32"/>
         <source>%1 (OSStatus %2)</source>
         <translation>%1 (OSStatus %2)</translation>
     </message>
+    <message>
+        <location filename="../libsecret.cpp" line="120"/>
+        <source>Entry not found</source>
+        <translation type="unfinished">找不到項目</translation>
+    </message>
+    <message>
+        <location filename="../keychain_haiku.cpp" line="72"/>
+        <source>error 0x%1: %2</source>
+        <translation type="unfinished"></translation>
+    </message>
 </context>
 </TS>
