diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' lame-3.100.orig/Makefile.MSVC lame-3.100/Makefile.MSVC
--- lame-3.100.orig/Makefile.MSVC	2016-01-29 21:02:24.000000000 +0100
+++ lame-3.100/Makefile.MSVC	2019-01-12 14:16:55.591647400 +0100
@@ -99,7 +99,7 @@
 !	ENDIF

 OFF = win32
-MACHINE = /machine:I386
+MACHINE = /machine:$(CRAFT_ARCH)
 LIB_OPTS = /nologo $(MACHINE)

 !	MESSAGE ----------------------------------------------------------------------
@@ -544,7 +544,7 @@
 .SUFFIXES : .nas
 .nas.obj:
 	@echo $<
-	@nasmw -f $(OFF) -i libmp3lame/i386/ -DWIN32 $< -o $@
+	@nasm -f $(OFF) -i libmp3lame/i386/ -DWIN32 $< -o $@

 CC_SWITCHES = $(CC_SWITCHES) /DHAVE_NASM
 ASM_OBJ = $(ASM_OBJ) \
