diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' sqlitebrowser-3.10.1.orig/CMakeLists.txt sqlitebrowser-3.10.1/CMakeLists.txt
--- sqlitebrowser-3.10.1.orig/CMakeLists.txt	2017-09-20 15:59:51.000000000 +0200
+++ sqlitebrowser-3.10.1/CMakeLists.txt	2017-12-15 16:32:37.011978600 +0100
@@ -11,39 +11,8 @@
     set(CMAKE_BUILD_TYPE "Release")
 endif()

-add_definitions(-std=c++11)
-
-if(WIN32 AND MSVC)
-	project("DB Browser for SQLite")
-	if(CMAKE_CL_64)
-		# Paths for 64-bit windows builds
-		set(OPENSSL_PATH "C:/dev/OpenSSL-Win64")
-		set(QT5_PATH "C:/dev/Qt/5.7/msvc2013_64")
-		set(VSREDIST "vcredist_x64.exe")
-
-                # Choose between SQLCipher or SQLite, depending whether
-                # -Dsqlcipher=1 is passed on the command line
-                if(sqlcipher)
-                        set(SQLITE3_PATH "C:/git_repos/SQLCipher-Win64")
-                else()
-                        set(SQLITE3_PATH "C:/dev/SQLite-Win64")
-                endif()
-	else()
-		# Paths for 32-bit windows builds
-		set(OPENSSL_PATH "C:/dev/OpenSSL-Win32")
-		set(QT5_PATH "C:/dev/Qt/5.7/msvc2013")
-		set(VSREDIST "vcredist_x86.exe")
-
-                # Choose between SQLCipher or SQLite, depending whether
-                # -Dsqlcipher=1 is passed on the command line
-                if(sqlcipher)
-                        set(SQLITE3_PATH "C:/git_repos/SQLCipher-Win32")
-                else()
-                        set(SQLITE3_PATH "C:/dev/SQLite-Win32")
-                endif()
-	endif()
-	set(CMAKE_PREFIX_PATH "${QT5_PATH};${SQLITE3_PATH}")
-	set(VSREDIST_DIR "C:/dev/dependencies")
+if(NOT MSVC)
+    add_definitions(-std=c++11)
 endif()

 if(NOT FORCE_INTERNAL_ANTLR)
@@ -345,11 +314,10 @@
     endif()
 endif()

-if(NOT WIN32)
-	install(TARGETS ${PROJECT_NAME}
+install(TARGETS ${PROJECT_NAME}
 		RUNTIME DESTINATION bin
-		LIBRARY DESTINATION lib)
-endif()
+    LIBRARY DESTINATION bin
+    ARCHIVE DESTINATION lib)

 if(ENABLE_TESTING)
 	add_subdirectory(src/tests)
@@ -366,46 +334,6 @@
 		DESTINATION share/appdata/)
 endif(UNIX AND NOT APPLE)

-if(WIN32 AND MSVC)
-	install(TARGETS ${PROJECT_NAME}
-		RUNTIME DESTINATION "/"
-		LIBRARY DESTINATION lib)
-
-	set(QT5_BIN_PATH ${QT5_PATH}/bin)
-	# The Qt5 Debug configuration library files have a 'd' postfix
-	install(FILES
-			${QT5_BIN_PATH}/Qt5Cored.dll
-			${QT5_BIN_PATH}/Qt5Guid.dll
-			${QT5_BIN_PATH}/Qt5Networkd.dll
-			${QT5_BIN_PATH}/Qt5PrintSupportd.dll
-			${QT5_BIN_PATH}/Qt5Widgetsd.dll
-		DESTINATION "/"
-		CONFIGURATIONS Debug)
-	# The Qt5 Release configuration files don't have a postfix
-	install(FILES
-			${QT5_BIN_PATH}/Qt5Core.dll
-			${QT5_BIN_PATH}/Qt5Gui.dll
-			${QT5_BIN_PATH}/Qt5Network.dll
-			${QT5_BIN_PATH}/Qt5PrintSupport.dll
-			${QT5_BIN_PATH}/Qt5Widgets.dll
-		DESTINATION "/"
-		CONFIGURATIONS Release)
-	# The files below are common to all configurations
-	install(FILES
-			${SQLITE3_DLL}
-			${OPENSSL_PATH}/libeay32.dll
-			${OPENSSL_PATH}/ssleay32.dll
-		DESTINATION "/")
-	install(FILES
-		${QT5_PATH}/plugins/platforms/qwindows.dll
-		DESTINATION platforms)
-	install(PROGRAMS "${VSREDIST_DIR}/${VSREDIST}" DESTINATION redist)
-	# The batch file launcher
-	install(FILES
-			distri/winlaunch.bat
-		DESTINATION "/")
-endif()
-
 #cpack
 set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "DB Browser for SQLite")
 set(CPACK_PACKAGE_VENDOR "DB Browser for SQLite Team")
