diff -Nrub -x *~ -x *\.rej -x *\.orig -x *\.o -x *\.pyc -x CMakeLists.txt.user libvpx-1.9.0.orig/configure libvpx-1.9.0/configure
--- libvpx-1.9.0.orig/configure	2020-07-30 23:59:15.000000000 +0400
+++ libvpx-1.9.0/configure	2021-05-19 19:15:59.908432000 +0300
@@ -169,10 +169,6 @@
     [ -f "${source_path}/${t}.mk" ] && enable_feature ${t}
 done
 
-if ! diff --version >/dev/null; then
-  die "diff missing: Try installing diffutils via your package manager."
-fi
-
 if ! perl --version >/dev/null; then
     die "Perl is required to build"
 fi
