diff -Nrub -x *~ -x *\.rej -x *\.orig -x *\.o -x *\.pyc -x CMakeLists.txt.user extra-cmake-modules-5.245.0.orig/kde-modules/KDEInstallDirs6.cmake extra-cmake-modules-5.245.0/kde-modules/KDEInstallDirs6.cmake
--- extra-cmake-modules-5.245.0.orig/kde-modules/KDEInstallDirs6.cmake	2023-11-08 15:40:41
+++ extra-cmake-modules-5.245.0/kde-modules/KDEInstallDirs6.cmake	2023-11-11 20:23:49
@@ -295,7 +295,7 @@
 # on the Mac support an extra install directory for application bundles
 if(APPLE)
     set(KF_INSTALL_TARGETS_DEFAULT_ARGS  ${KF_INSTALL_TARGETS_DEFAULT_ARGS}
-                                          BUNDLE DESTINATION "${BUNDLE_INSTALL_DIR}" )
+                                          BUNDLE DESTINATION "${KDE_INSTALL_BUNDLEDIR}" )
 endif()
 
 include(${CMAKE_CURRENT_LIST_DIR}/KDESetupPrefixScript.cmake)
