diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' extra-cmake-modules-5.54.0.orig/kde-modules/KDEFrameworkCompilerSettings.cmake extra-cmake-modules-5.54.0/kde-modules/KDEFrameworkCompilerSettings.cmake
--- extra-cmake-modules-5.54.0.orig/kde-modules/KDEFrameworkCompilerSettings.cmake	2018-12-21 00:31:37.000000000 +0100
+++ extra-cmake-modules-5.54.0/kde-modules/KDEFrameworkCompilerSettings.cmake	2019-01-26 12:26:55.448365000 +0100
@@ -71,7 +71,8 @@
    endif()
 endif()
 
-if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+# don't enable with apple clang
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
    if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5.0.0")
       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wzero-as-null-pointer-constant" )
    endif()
