diff --git a/drivers/telescope/CMakeLists.txt b/drivers/telescope/CMakeLists.txt
index 9c2f9f234..d88a3e7aa 100644
--- a/drivers/telescope/CMakeLists.txt
+++ b/drivers/telescope/CMakeLists.txt
@@ -33,7 +33,7 @@ add_executable(indi_lx200basic
     lx200driver.cpp
     lx200basic.cpp)
 
-target_link_libraries(indi_lx200basic indidriver)
+target_link_libraries(indi_lx200basic indidriver ${CURL})
 
 install(TARGETS indi_lx200basic RUNTIME DESTINATION bin)
 
@@ -45,7 +45,7 @@ add_executable(indi_lx200_TeenAstro
     lx200_TeenAstro.cpp)
 
 target_compile_definitions(indi_lx200_TeenAstro PRIVATE _XOPEN_SOURCE=700)
-target_link_libraries(indi_lx200_TeenAstro indidriver)
+target_link_libraries(indi_lx200_TeenAstro indidriver ${CURL})
 
 install(TARGETS indi_lx200_TeenAstro RUNTIME DESTINATION bin)
 
@@ -115,7 +115,7 @@ add_executable(indi_celestron_gps
     celestrondriver.cpp
     celestrongps.cpp)
 
-target_link_libraries(indi_celestron_gps indidriver)
+target_link_libraries(indi_celestron_gps indidriver ${CURL})
 
 install(TARGETS indi_celestron_gps RUNTIME DESTINATION bin)
 