diff --git a/CMakeLists.txt b/CMakeLists.txt
index f5cda74c4..70e7d6e4b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -298,7 +298,7 @@ if(INDI_BUILD_DRIVERS)
         find_package(GSL REQUIRED)
         find_package(JPEG REQUIRED)
         find_library(M_LIB m)
-        if(CYGWIN)
+        if(CYGWIN OR UNIX)
             find_package(Iconv REQUIRED)
         endif()
 
diff --git a/cmake_modules/FindIconv.cmake b/cmake_modules/FindIconv.cmake
deleted file mode 100644
index dd2e640c0..000000000
--- a/cmake_modules/FindIconv.cmake
+++ /dev/null
@@ -1,88 +0,0 @@
-#
-#  Copyright (C) 2010  Michael Bell <michael.bell@web.de>
-#                2015-2016 MariaDB Corporation AB
-#
-#  Redistribution and use is allowed according to the terms of the New
-#  BSD license.
-#  For details see the  COPYING-CMAKE-SCRIPTS file.
-#
-#  ICONV_EXTERNAL - Iconv is an external library (not libc)
-#  ICONV_FOUND - system has Iconv
-#  ICONV_INCLUDE_DIR - the Iconv include directory
-#  ICONV_LIBRARIES - Link these to use Iconv
-#  ICONV_SECOND_ARGUMENT_IS_CONST - the second argument for iconv() is const
-#  ICONV_VERSION - Iconv version string
-
-if (ICONV_INCLUDE_DIR AND ICONV_LIBRARIES)
-  # Already in cache, be silent
-  set(ICONV_FIND_QUIETLY TRUE)
-endif (ICONV_INCLUDE_DIR AND ICONV_LIBRARIES)
-
-find_path(ICONV_INCLUDE_DIR iconv.h)
-
-IF(CMAKE_SYSTEM_NAME MATCHES "SunOS")
-  # There is some libiconv.so in  /usr/local that must
-  # be avoided, iconv routines are in libc  
-ELSEIF(APPLE)
-  find_library(ICONV_LIBRARIES NAMES iconv libiconv PATHS
-               /usr/lib/
-               NO_CMAKE_SYSTEM_PATH)
-    SET(ICONV_EXTERNAL TRUE)
-ELSEIF(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
-  # Force paths to GNU version of iconv to avoid clashes with
-  # the libc version.
-  find_library(ICONV_LIBRARIES NAMES iconv libiconv PATHS
-               /usr/local/lib
-               NO_CMAKE_SYSTEM_PATH)
-  SET(ICONV_EXTERNAL TRUE)
-ELSE()
-  find_library(ICONV_LIBRARIES NAMES iconv libiconv libiconv-2)
-  IF(ICONV_LIBRARIES)
-    SET(ICONV_EXTERNAL TRUE)
-  ENDIF()
-ENDIF()
-
-if (ICONV_INCLUDE_DIR AND ICONV_LIBRARIES)
-   set (ICONV_FOUND TRUE)
-endif (ICONV_INCLUDE_DIR AND ICONV_LIBRARIES)
-
-set(CMAKE_REQUIRED_INCLUDES ${ICONV_INCLUDE_DIR})
-IF(ICONV_EXTERNAL)
-  set(CMAKE_REQUIRED_LIBRARIES ${ICONV_LIBRARIES})
-ENDIF()
-
-if (ICONV_FOUND)
-  include(CheckCSourceCompiles)
-  CHECK_C_SOURCE_COMPILES("
-  #include <iconv.h>
-  int main(){
-    iconv_t conv = 0;
-    const char* in = 0;
-    size_t ilen = 0;
-    char* out = 0;
-    size_t olen = 0;
-    iconv(conv, &in, &ilen, &out, &olen);
-    return 0;
-  }
-" ICONV_SECOND_ARGUMENT_IS_CONST )
-endif (ICONV_FOUND)
-
-set (CMAKE_REQUIRED_INCLUDES)
-set (CMAKE_REQUIRED_LIBRARIES)
-
-if (ICONV_FOUND)
-  if (NOT ICONV_FIND_QUIETLY)
-    message (STATUS "Found Iconv: ${ICONV_LIBRARIES}")
-  endif (NOT ICONV_FIND_QUIETLY)
-else (ICONV_FOUND)
-  if (Iconv_FIND_REQUIRED)
-    message (FATAL_ERROR "Could not find Iconv")
-  endif (Iconv_FIND_REQUIRED)
-endif (ICONV_FOUND)
-
-MARK_AS_ADVANCED(
-  ICONV_INCLUDE_DIR
-  ICONV_LIBRARIES
-  ICONV_EXTERNAL
-  ICONV_SECOND_ARGUMENT_IS_CONST
-)
diff --git a/drivers/telescope/CMakeLists.txt b/drivers/telescope/CMakeLists.txt
index d88a3e7aa..49b49c1c4 100644
--- a/drivers/telescope/CMakeLists.txt
+++ b/drivers/telescope/CMakeLists.txt
@@ -33,7 +33,7 @@ add_executable(indi_lx200basic
     lx200driver.cpp
     lx200basic.cpp)
 
-target_link_libraries(indi_lx200basic indidriver ${CURL})
+target_link_libraries(indi_lx200basic indidriver ${CURL} ${Iconv_LIBRARIES})
 
 install(TARGETS indi_lx200basic RUNTIME DESTINATION bin)
 
@@ -45,7 +45,7 @@ add_executable(indi_lx200_TeenAstro
     lx200_TeenAstro.cpp)
 
 target_compile_definitions(indi_lx200_TeenAstro PRIVATE _XOPEN_SOURCE=700)
-target_link_libraries(indi_lx200_TeenAstro indidriver ${CURL})
+target_link_libraries(indi_lx200_TeenAstro indidriver ${CURL} ${Iconv_LIBRARIES})
 
 install(TARGETS indi_lx200_TeenAstro RUNTIME DESTINATION bin)
 
@@ -115,7 +115,7 @@ add_executable(indi_celestron_gps
     celestrondriver.cpp
     celestrongps.cpp)
 
-target_link_libraries(indi_celestron_gps indidriver ${CURL})
+target_link_libraries(indi_celestron_gps indidriver ${CURL} ${Iconv_LIBRARIES})
 
 install(TARGETS indi_celestron_gps RUNTIME DESTINATION bin)
 
diff --git a/libs/hid/CMakeLists.txt b/libs/hid/CMakeLists.txt
index 8bd30d98a..d0e73f8ad 100644
--- a/libs/hid/CMakeLists.txt
+++ b/libs/hid/CMakeLists.txt
@@ -43,8 +43,8 @@ install(FILES
 # ########## HID Test ##############
 if(APPLE)
     set(LIBS "-framework IOKit -framework CoreFoundation")
-elseif(CYGWIN)
-    set(LIBS ${ICONV_LIBRARIES})
+elseif(CYGWIN OR UNIX)
+    set(LIBS ${Iconv_LIBRARIES})
 endif()
 
 add_executable(indi_hid_test hidtest.cpp)
