diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' xapian-core-1.4.5.orig/cmake/xapian-config.cmake.in xapian-core-1.4.5/cmake/xapian-config.cmake.in
--- xapian-core-1.4.5.orig/cmake/xapian-config.cmake.in	2017-10-16 04:32:24.000000000 +0200
+++ xapian-core-1.4.5/cmake/xapian-config.cmake.in	2018-05-15 12:03:21.096048500 +0200
@@ -1,5 +1,5 @@
 SET(prefix "@prefix@")
 SET(exec_prefix "@exec_prefix@")
-SET(XAPIAN_LIBRARIES "@libdir@/libxapian@LIBRARY_VERSION_SUFFIX@@SHLIBEXT@" CACHE FILEPATH "Libraries for Xapian")
+SET(XAPIAN_LIBRARIES "@libdir@/libxapian@LIBRARY_VERSION_SUFFIX@${CMAKE_IMPORT_LIBRARY_SUFFIX}" CACHE FILEPATH "Libraries for Xapian")
 SET(XAPIAN_INCLUDE_DIR "@incdir@" CACHE PATH "Include path for Xapian")
 SET(XAPIAN_FOUND "TRUE")
