diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user pcre2-10.37.orig/CMakeLists.txt pcre2-10.37/CMakeLists.txt
--- pcre2-10.37.orig/CMakeLists.txt	2021-01-14 18:11:03.000000000 +0100
+++ pcre2-10.37/CMakeLists.txt	2021-11-20 13:28:08.000000000 +0100
@@ -99,7 +99,7 @@
 
 # Increased minimum to 2.8.5 to support GNUInstallDirs.
 CMAKE_MINIMUM_REQUIRED(VERSION 2.8.5)
-
+set(CMAKE_MACOSX_RPATH ON)
 # Set policy CMP0026 to avoid warnings for the use of LOCATION in
 # GET_TARGET_PROPERTY. This should no longer be required.
 # CMAKE_POLICY(SET CMP0026 OLD)
