diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user pcre2-10.35.orig/CMakeLists.txt pcre2-10.35/CMakeLists.txt
--- pcre2-10.35.orig/CMakeLists.txt	2020-11-06 21:16:25.585379200 +0100
+++ pcre2-10.35/CMakeLists.txt	2020-11-06 21:16:22.412379300 +0100
@@ -110,6 +110,11 @@
 
 SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -I${PROJECT_SOURCE_DIR}/src")
 set(CMAKE_MACOSX_RPATH ON)
+# needed to make sure to not link debug libs
+# against release libs and vice versa
+IF(WIN32)
+  SET(CMAKE_DEBUG_POSTFIX "d")
+ENDIF(WIN32)
 
 # external packages
 FIND_PACKAGE( BZip2 )
@@ -623,11 +628,6 @@
 ENDIF(MSVC)
 
 SET(CMAKE_INCLUDE_CURRENT_DIR 1)
-# needed to make sure to not link debug libs
-# against release libs and vice versa
-IF(WIN32)
-  SET(CMAKE_DEBUG_POSTFIX "d")
-ENDIF(WIN32)
 
 SET(targets)
 
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user pcre2-10.35.orig/libpcre2-16.pc.in pcre2-10.35/libpcre2-16.pc.in
--- pcre2-10.35.orig/libpcre2-16.pc.in	2015-04-23 19:41:55.000000000 +0200
+++ pcre2-10.35/libpcre2-16.pc.in	2020-11-06 21:15:11.390699400 +0100
@@ -8,6 +8,6 @@
 Name: libpcre2-16
 Description: PCRE2 - Perl compatible regular expressions C library (2nd API) with 16 bit character support
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lpcre2-16
+Libs: -L${libdir} -lpcre2-16@CMAKE_DEBUG_POSTFIX@
 Libs.private: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@
 Cflags: -I${includedir} @PCRE2_STATIC_CFLAG@
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user pcre2-10.35.orig/libpcre2-32.pc.in pcre2-10.35/libpcre2-32.pc.in
--- pcre2-10.35.orig/libpcre2-32.pc.in	2015-04-23 19:42:43.000000000 +0200
+++ pcre2-10.35/libpcre2-32.pc.in	2020-11-06 21:15:11.394701900 +0100
@@ -8,6 +8,6 @@
 Name: libpcre2-32
 Description: PCRE2 - Perl compatible regular expressions C library (2nd API) with 32 bit character support
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lpcre2-32
+Libs: -L${libdir} -lpcre2-32@CMAKE_DEBUG_POSTFIX@
 Libs.private: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@
 Cflags: -I${includedir} @PCRE2_STATIC_CFLAG@
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user pcre2-10.35.orig/libpcre2-8.pc.in pcre2-10.35/libpcre2-8.pc.in
--- pcre2-10.35.orig/libpcre2-8.pc.in	2015-04-23 19:42:33.000000000 +0200
+++ pcre2-10.35/libpcre2-8.pc.in	2020-11-06 21:15:11.396702500 +0100
@@ -8,6 +8,6 @@
 Name: libpcre2-8
 Description: PCRE2 - Perl compatible regular expressions C library (2nd API) with 8 bit character support
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lpcre2-8
+Libs: -L${libdir} -lpcre2-8@CMAKE_DEBUG_POSTFIX@
 Libs.private: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@
 Cflags: -I${includedir} @PCRE2_STATIC_CFLAG@
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user pcre2-10.35.orig/libpcre2-posix.pc.in pcre2-10.35/libpcre2-posix.pc.in
--- pcre2-10.35.orig/libpcre2-posix.pc.in	2015-04-23 19:43:11.000000000 +0200
+++ pcre2-10.35/libpcre2-posix.pc.in	2020-11-06 21:15:11.397700800 +0100
@@ -8,6 +8,6 @@
 Name: libpcre2-posix
 Description: Posix compatible interface to libpcre2-8
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lpcre2-posix
+Libs: -L${libdir} -lpcre2-posix@CMAKE_DEBUG_POSTFIX@
 Cflags: -I${includedir} @PCRE2_STATIC_CFLAG@
 Requires.private: libpcre2-8
