diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user pcre2-10.35.orig/CMakeLists.txt pcre2-10.35/CMakeLists.txt
--- pcre2-10.35.orig/CMakeLists.txt	2020-11-17 11:38:43.953302500 +0100
+++ pcre2-10.35/CMakeLists.txt	2020-11-17 11:37:26.172048200 +0100
@@ -112,7 +112,9 @@
 set(CMAKE_MACOSX_RPATH ON)
 # needed to make sure to not link debug libs
 # against release libs and vice versa
-
+IF(WIN32)
+  SET(CMAKE_DEBUG_POSTFIX "d")
+ENDIF(WIN32)
 if (CMAKE_BUILD_TYPE STREQUAL "Debug")
     set(PCRE2_DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
 endif()
