diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' glib-2.49.4.orig/Makefile.in glib-2.49.4/Makefile.in
--- glib-2.49.4.orig/Makefile.in	2016-07-21 02:54:00.000000000 +0200
+++ glib-2.49.4/Makefile.in	2018-03-15 00:52:46.000000000 +0100
@@ -834,7 +834,7 @@
 
 @ENABLE_INSTALLED_TESTS_TRUE@installed_test_meta_DATA = $(installed_testcases:=.test)
 ACLOCAL_AMFLAGS = -I m4macros ${ACLOCAL_FLAGS}
-SUBDIRS = . m4macros glib gmodule gthread gobject gio po docs tests
+SUBDIRS = . m4macros glib gmodule gthread gobject gio po
 DIST_SUBDIRS = $(SUBDIRS) build
 bin_SCRIPTS = glib-gettextize
 AM_CPPFLAGS = \
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' glib-2.49.4.orig/gio/Makefile.in glib-2.49.4/gio/Makefile.in
--- glib-2.49.4.orig/gio/Makefile.in	2016-07-21 02:54:01.000000000 +0200
+++ glib-2.49.4/gio/Makefile.in	2018-03-15 00:53:12.000000000 +0100
@@ -912,7 +912,7 @@
     *) \
       b='$*';; \
   esac
-DIST_SUBDIRS = gdbus-2.0/codegen xdgmime inotify kqueue win32 . tests \
+DIST_SUBDIRS = gdbus-2.0/codegen xdgmime inotify kqueue win32 . \
 	fam
 am__DIST_COMMON = $(srcdir)/Makefile.in $(srcdir)/gio.rc.in \
 	$(srcdir)/gnetworking.h.in \
@@ -1296,7 +1296,7 @@
 
 @ENABLE_INSTALLED_TESTS_TRUE@installed_test_meta_DATA = $(installed_testcases:=.test)
 SUBDIRS = gdbus-2.0/codegen $(am__append_15) $(am__append_19) \
-	$(am__append_22) $(am__append_25) . tests $(am__append_28)
+    $(am__append_22) $(am__append_25) . $(am__append_28)
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@install_ms_lib_cmd = $(INSTALL) gio-2.0.lib $(DESTDIR)$(libdir)
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@uninstall_ms_lib_cmd = -rm $(DESTDIR)$(libdir)/gio-2.0.lib
 AM_CPPFLAGS = \
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' glib-2.49.4.orig/glib/Makefile.in glib-2.49.4/glib/Makefile.in
--- glib-2.49.4.orig/glib/Makefile.in	2016-07-21 02:54:02.000000000 +0200
+++ glib-2.49.4/glib/Makefile.in	2018-03-15 00:53:40.000000000 +0100
@@ -1008,8 +1008,8 @@
 @HAVE_GOOD_PRINTF_FALSE@PRINTF_SUBDIR = gnulib
 @HAVE_GOOD_PRINTF_FALSE@printf_la = gnulib/libgnulib.la
 @USE_SYSTEM_PCRE_FALSE@MAYBE_PCRE = pcre 
-SUBDIRS = libcharset $(PRINTF_SUBDIR) $(MAYBE_PCRE) update-pcre . tests
-DIST_SUBDIRS = libcharset gnulib pcre update-pcre tests
+SUBDIRS = libcharset $(PRINTF_SUBDIR) $(MAYBE_PCRE) update-pcre .
+DIST_SUBDIRS = libcharset gnulib pcre update-pcre
 AM_CPPFLAGS = \
 	$(glib_INCLUDES) 		\
 	$(pcre_inc) 			\
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' glib-2.49.4.orig/gobject/Makefile.in glib-2.49.4/gobject/Makefile.in
--- glib-2.49.4.orig/gobject/Makefile.in	2016-07-21 02:54:04.000000000 +0200
+++ glib-2.49.4/gobject/Makefile.in	2018-03-15 00:54:09.000000000 +0100
@@ -892,7 +892,7 @@
 # GObject - GLib Type, Object, Parameter and Signal Library
 # Copyright (C) 1997,98,99,2000 Tim Janik and Red Hat, Inc.
 #
-SUBDIRS = . tests
+SUBDIRS = .
 AM_CPPFLAGS = \
 	-DG_LOG_DOMAIN=\"GLib-GObject\"		\
 	$(glib_INCLUDES)			\
