diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' glib-glib-2.49.4.orig/build/win32/vs14/gio.vcxproj glib-glib-2.49.4/build/win32/vs14/gio.vcxproj
--- glib-glib-2.49.4.orig/build/win32/vs14/gio.vcxproj	2016-07-24 22:11:42.000000000 +0200
+++ glib-glib-2.49.4/build/win32/vs14/gio.vcxproj	2018-02-13 08:52:32.838013500 +0100
@@ -86,7 +86,7 @@
       <CompileAs>CompileAsC</CompileAs>
     </ClCompile>
     <Link>
-      <AdditionalDependencies>zlib_a.lib;$(WindowsNetworkingLibs);%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>zlib.lib;$(WindowsNetworkingLibs);%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>$(OutDir)$(GlibDllPrefix)$(ProjectName)$(GlibDllSuffix).dll</OutputFile>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <ProgramDatabaseFile>$(OutDir)$(GlibDllPrefix)$(ProjectName)$(GlibDllSuffix).pdb</ProgramDatabaseFile>
@@ -110,7 +110,7 @@
       <CompileAs>CompileAsC</CompileAs>
     </ClCompile>
     <Link>
-      <AdditionalDependencies>zlib_a.lib;$(WindowsNetworkingLibs);%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>zlib.lib;$(WindowsNetworkingLibs);%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>$(OutDir)$(GlibDllPrefix)$(ProjectName)$(GlibDllSuffix).dll</OutputFile>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <ProgramDatabaseFile>$(OutDir)$(GlibDllPrefix)$(ProjectName)$(GlibDllSuffix).pdb</ProgramDatabaseFile>
@@ -139,7 +139,7 @@
       <CompileAs>CompileAsC</CompileAs>
     </ClCompile>
     <Link>
-      <AdditionalDependencies>zlib_a.lib;$(WindowsNetworkingLibs);%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>zlib.lib;$(WindowsNetworkingLibs);%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>$(OutDir)$(GlibDllPrefix)$(ProjectName)$(GlibDllSuffix).dll</OutputFile>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <ProgramDatabaseFile>$(OutDir)$(GlibDllPrefix)$(ProjectName)$(GlibDllSuffix).pdb</ProgramDatabaseFile>
@@ -163,7 +163,7 @@
       <CompileAs>CompileAsC</CompileAs>
     </ClCompile>
     <Link>
-      <AdditionalDependencies>zlib_a.lib;$(WindowsNetworkingLibs);%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>zlib.lib;$(WindowsNetworkingLibs);%(AdditionalDependencies)</AdditionalDependencies>
       <OutputFile>$(OutDir)$(GlibDllPrefix)$(ProjectName)$(GlibDllSuffix).dll</OutputFile>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <ProgramDatabaseFile>$(OutDir)$(GlibDllPrefix)$(ProjectName)$(GlibDllSuffix).pdb</ProgramDatabaseFile>
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' glib-glib-2.49.4.orig/build/win32/vs14/glib-build-defines.props glib-glib-2.49.4/build/win32/vs14/glib-build-defines.props
--- glib-glib-2.49.4.orig/build/win32/vs14/glib-build-defines.props	2016-07-24 22:11:42.000000000 +0200
+++ glib-glib-2.49.4/build/win32/vs14/glib-build-defines.props	2018-02-13 08:53:22.324829200 +0100
@@ -33,7 +33,7 @@
       <AdditionalOptions>/d2Zi+ %(AdditionalOptions)</AdditionalOptions>
     </ClCompile>
     <Link>
-      <AdditionalDependencies>libintl_a.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>intl.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <AdditionalLibraryDirectories>$(GlibEtcInstallRoot)\lib;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
     </Link>
   </ItemDefinitionGroup>
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gdbus.exe and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gdbus.exe differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gdbus.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gdbus.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gdbus.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gdbus.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gdbus.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gdbus.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio-2.0.exp and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio-2.0.exp differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio-2.0.lib and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio-2.0.lib differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio-2.dll and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio-2.dll differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio-2.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio-2.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio-2.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio-2.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio-2.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio-2.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio-querymodules.exe and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio-querymodules.exe differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio-querymodules.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio-querymodules.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio-querymodules.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio-querymodules.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio-querymodules.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio-querymodules.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio.exe and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio.exe differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gio.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gio.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-2.0.exp and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-2.0.exp differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-2.0.lib and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-2.0.lib differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-2.dll and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-2.dll differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-2.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-2.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-2.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-2.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-2.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-2.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-compile-resources.exe and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-compile-resources.exe differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-compile-resources.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-compile-resources.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-compile-resources.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-compile-resources.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-compile-resources.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-compile-resources.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-compile-schemas.exe and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-compile-schemas.exe differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-compile-schemas.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-compile-schemas.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-compile-schemas.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-compile-schemas.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-compile-schemas.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-compile-schemas.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-genmarshal.exe and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-genmarshal.exe differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-genmarshal.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-genmarshal.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-genmarshal.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-genmarshal.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/glib-genmarshal.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/glib-genmarshal.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gmodule-2.0.exp and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gmodule-2.0.exp differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gmodule-2.0.lib and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gmodule-2.0.lib differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gmodule-2.dll and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gmodule-2.dll differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gmodule-2.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gmodule-2.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gmodule-2.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gmodule-2.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gmodule-2.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gmodule-2.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gobject-2.0.exp and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gobject-2.0.exp differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gobject-2.0.lib and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gobject-2.0.lib differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gobject-2.dll and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gobject-2.dll differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gobject-2.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gobject-2.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gobject-2.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gobject-2.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gobject-2.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gobject-2.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gresource.exe and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gresource.exe differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gresource.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gresource.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gresource.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gresource.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gresource.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gresource.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gsettings.exe and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gsettings.exe differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gsettings.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gsettings.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gsettings.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gsettings.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gsettings.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gsettings.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gspawn-win64-helper-console.exe and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gspawn-win64-helper-console.exe differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gspawn-win64-helper-console.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gspawn-win64-helper-console.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gspawn-win64-helper-console.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gspawn-win64-helper-console.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gspawn-win64-helper-console.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gspawn-win64-helper-console.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gspawn-win64-helper.exe and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gspawn-win64-helper.exe differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gspawn-win64-helper.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gspawn-win64-helper.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gspawn-win64-helper.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gspawn-win64-helper.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gspawn-win64-helper.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gspawn-win64-helper.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gthread-2.0.exp and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gthread-2.0.exp differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gthread-2.0.lib and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gthread-2.0.lib differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gthread-2.dll and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gthread-2.dll differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gthread-2.iobj and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gthread-2.iobj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gthread-2.ipdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gthread-2.ipdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/bin/gthread-2.pdb and glib-glib-2.49.4/build/win32/vs14/Release/x64/bin/gthread-2.pdb differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/obj/gdbus/gdbus-tool.obj and glib-glib-2.49.4/build/win32/vs14/Release/x64/obj/gdbus/gdbus-tool.obj differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/obj/gdbus/gdbus.tlog/CL.command.1.tlog and glib-glib-2.49.4/build/win32/vs14/Release/x64/obj/gdbus/gdbus.tlog/CL.command.1.tlog differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/obj/gdbus/gdbus.tlog/CL.read.1.tlog and glib-glib-2.49.4/build/win32/vs14/Release/x64/obj/gdbus/gdbus.tlog/CL.read.1.tlog differ
Files glib-glib-2.49.4.orig/build/win32/vs14/Release/x64/obj/gdbus/gdbus.tlog/CL.write.1.tlog and glib-glib-2.49.4/build/win32/vs14/Release/x64/obj/gdbus/gdbus.tlog/CL.write.1.tlog differ
