diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user glib-2.66.3.orig/meson.build glib-2.66.3/meson.build
--- glib-2.66.3.orig/meson.build	2020-11-16 13:19:46.086260000 +0100
+++ glib-2.66.3/meson.build	2020-12-04 15:10:59.480588300 +0100
@@ -5,7 +5,7 @@
   default_options : [
     'buildtype=debugoptimized',
     'warning_level=1',
-    'c_std=gnu99'
+    'c_std=c11'
   ]
 )
 
