diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user rkward-0.7.3.orig/rkward/rbackend/CMakeLists.txt rkward-0.7.3/rkward/rbackend/CMakeLists.txt
--- rkward-0.7.3.orig/rkward/rbackend/CMakeLists.txt	2022-04-09 10:14:33.000000000 +0200
+++ rkward-0.7.3/rkward/rbackend/CMakeLists.txt	2022-04-26 15:14:55.386914200 +0200
@@ -46,6 +46,16 @@
 ADD_EXECUTABLE(rkward.rbackend ${rbackend_backend_SRCS} ${rbackend_shared_SRCS})
 ECM_MARK_NONGUI_EXECUTABLE(rkward.rbackend)
 TARGET_LINK_LIBRARIES(rkward.rbackend rkgraphicsdevice.backend ${R_USED_LIBS} ${CMAKE_THREAD_LIBS_INIT} Qt5::Network Qt5::Core KF5::I18n)
+IF(WIN32)
+# 64MB stack size is what R itself is build with on Windows, while by default the stack size would be 1MB
+# Not sure, if 64MB is actually needed, but 1MB does cause trouble e.g. with some shiny apps.
+    IF(MSVC)
+        SET_TARGET_PROPERTIES(rkward.rbackend PROPERTIES LINK_FLAGS /STACK:64000000)
+    ELSE(MSVC)
+        SET_TARGET_PROPERTIES(rkward.rbackend PROPERTIES LINK_FLAGS -Wl,--stack,64000000)
+    ENDIF(MSVC)
+ENDIF(WIN32)
+
 IF(APPLE)
     TARGET_LINK_LIBRARIES(rkward.rbackend "-framework CoreFoundation")
     ADD_CUSTOM_COMMAND(TARGET rkward.rbackend POST_BUILD
