diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user libgcrypt-1.8.7.orig/src/gcrypt.h.in libgcrypt-1.8.7/src/gcrypt.h.in
--- libgcrypt-1.8.7.orig/src/gcrypt.h.in	2018-10-24 11:58:06.000000000 +0200
+++ libgcrypt-1.8.7/src/gcrypt.h.in	2020-11-09 12:57:31.264315000 +0100
@@ -35,8 +35,11 @@
 # include <winsock2.h>
 # include <ws2tcpip.h>
 # include <time.h>
+#ifdef _MSC_VER
+#include <BaseTsd.h> // ssize_t on Windows
+#define ssize_t SSIZE_T
+#endif
 # ifndef __GNUC__
-  typedef long ssize_t;
   typedef int  pid_t;
 # endif /*!__GNUC__*/
 #else
