From 8ae26f6eaa30161d7be8aa52594e9ccde9f94ddc Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Fri, 25 Aug 2023 18:05:24 +0300
Subject: [PATCH] fix Windows build

---
 src/KDSingleApplication/kdsingleapplication.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/KDSingleApplication/kdsingleapplication.h b/src/KDSingleApplication/kdsingleapplication.h
index 2d9ab6c..fc3d7cb 100644
--- a/src/KDSingleApplication/kdsingleapplication.h
+++ b/src/KDSingleApplication/kdsingleapplication.h
@@ -18,7 +18,7 @@

 class KDSingleApplicationPrivate;

-class KDSINGLEAPPLICATION_EXPORT KDSingleApplication : public QObject
+class KDSingleApplication : public QObject
 {
     Q_OBJECT
     Q_PROPERTY(QString name READ name CONSTANT)
--
2.38.1.windows.1

