diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' ctemplate-ctemplate-2.3.orig/src/htmlparser/fsm_config.py ctemplate-ctemplate-2.3/src/htmlparser/fsm_config.py
--- ctemplate-ctemplate-2.3.orig/src/htmlparser/fsm_config.py	2014-03-19 20:34:51.000000000 +0100
+++ ctemplate-ctemplate-2.3/src/htmlparser/fsm_config.py	2018-09-20 16:50:06.419285800 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (c) 2008, Google Inc.
 # All rights reserved.
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' ctemplate-ctemplate-2.3.orig/src/htmlparser/generate_fsm.py ctemplate-ctemplate-2.3/src/htmlparser/generate_fsm.py
--- ctemplate-ctemplate-2.3.orig/src/htmlparser/generate_fsm.py	2014-03-19 20:34:51.000000000 +0100
+++ ctemplate-ctemplate-2.3/src/htmlparser/generate_fsm.py	2018-09-20 16:50:09.066210600 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (c) 2008, Google Inc.
 # All rights reserved.
