diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' qtimageformats-everywhere-src-5.10.1.orig/config.tests/libtiff/libtiff.pro qtimageformats-everywhere-src-5.10.1/config.tests/libtiff/libtiff.pro
--- qtimageformats-everywhere-src-5.10.1.orig/config.tests/libtiff/libtiff.pro	2018-02-09 09:22:52.000000000 +0100
+++ qtimageformats-everywhere-src-5.10.1/config.tests/libtiff/libtiff.pro	2018-04-13 16:19:45.650098600 +0200
@@ -3,4 +3,4 @@
 mac:CONFIG -= app_bundle
 win32:CONFIG += console
 unix|mingw: LIBS += -ltiff
-else:win32: LIBS += libtiff.lib
+else:win32: LIBS += tiff.lib
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' qtimageformats-everywhere-src-5.10.1.orig/src/plugins/imageformats/tiff/tiff.pro qtimageformats-everywhere-src-5.10.1/src/plugins/imageformats/tiff/tiff.pro
--- qtimageformats-everywhere-src-5.10.1.orig/src/plugins/imageformats/tiff/tiff.pro	2018-02-09 09:22:52.000000000 +0100
+++ qtimageformats-everywhere-src-5.10.1/src/plugins/imageformats/tiff/tiff.pro	2018-04-13 16:20:50.880800900 +0200
@@ -6,7 +6,7 @@
 
 config_libtiff {
     unix|mingw: LIBS += -ltiff
-    else:win32: LIBS += libtiff.lib
+    else:win32: LIBS += tiff.lib
 } else {
     include($$PWD/../../../3rdparty/libtiff.pri)
 }
