diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' sqlcipher-3.4.2.orig/src/sqlite.h.in sqlcipher-3.4.2/src/sqlite.h.in
--- sqlcipher-3.4.2.orig/src/sqlite.h.in	2017-12-21 20:31:28.000000000 +0100
+++ sqlcipher-3.4.2/src/sqlite.h.in	2018-07-27 18:48:35.594147200 +0100
@@ -49,7 +49,18 @@
 # define SQLITE_EXTERN extern
 #endif
 #ifndef SQLITE_API
+# ifdef _MSC_VER
+#  if defined(SQLITE3_STATIC)
+    /* No export/import for static libraries */
 # define SQLITE_API
+#  elif defined(libsqlcipher_EXPORTS)
+#   define SQLITE_API __declspec(dllexport)
+#  else
+#   define SQLITE_API __declspec(dllimport)
+#  endif
+# else
+#  define SQLITE_API
+# endif
 #endif
 #ifndef SQLITE_CDECL
 # define SQLITE_CDECL
