diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user zstd-1.4.5.orig/build/cmake/lib/CMakeLists.txt zstd-1.4.5/build/cmake/lib/CMakeLists.txt
--- zstd-1.4.5.orig/build/cmake/lib/CMakeLists.txt	2020-05-22 07:04:00.000000000 +0200
+++ zstd-1.4.5/build/cmake/lib/CMakeLists.txt	2020-10-26 13:51:36.868064400 +0100
@@ -163,6 +163,7 @@
     INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
     )
 
 # uninstall target
