diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user zstd-1.5.0.orig/build/cmake/lib/CMakeLists.txt zstd-1.5.0/build/cmake/lib/CMakeLists.txt
--- zstd-1.5.0.orig/build/cmake/lib/CMakeLists.txt	2021-05-14 16:59:34.000000000 +0200
+++ zstd-1.5.0/build/cmake/lib/CMakeLists.txt	2021-11-04 13:06:17.762791000 +0100
@@ -130,7 +130,7 @@
             OUTPUT_NAME ${STATIC_LIBRARY_BASE_NAME})
 endif ()
 
-if (UNIX OR MINGW)
+if (TRUE)
     # pkg-config
     set(PREFIX "${CMAKE_INSTALL_PREFIX}")
     set(EXEC_PREFIX "\${prefix}")
