From 460b2c419d699e049480125b6431385d2f9539a7 Mon Sep 17 00:00:00 2001
From: Hannah von Reth <vonreth@kde.org>
Date: Fri, 15 Dec 2017 10:38:09 +0100
Subject: [PATCH 1/3] define strncasecmp and strcasecmp on Windows

---
 src/versit/vcc.y     | 5 +++++
 src/versit/vobject.c | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/src/versit/vcc.y b/src/versit/vcc.y
index b9fbb0f89..d8188eec6 100644
--- a/src/versit/vcc.y
+++ b/src/versit/vcc.y
@@ -104,6 +104,11 @@ DFARS 252.227-7013 or 48 CFR 52.227-19, as applicable.
 #include <ctype.h>
 #include "vcc.h"

+#ifdef _WIN32
+#define strncasecmp _strnicmp
+#define strcasecmp _stricmp
+#endif
+
 /* The following is a hack that I hope will get things compiling
  * on SunOS 4.1.x systems
  */
diff --git a/src/versit/vobject.c b/src/versit/vobject.c
index db13f9e27..f95f49a25 100644
--- a/src/versit/vobject.c
+++ b/src/versit/vobject.c
@@ -48,6 +48,11 @@ DFARS 252.227-7013 or 48 CFR 52.227-19, as applicable.
 #include <string.h>
 #include <stdio.h>

+#ifdef _WIN32
+#define strncasecmp _strnicmp
+#define strcasecmp _stricmp
+#endif
+
 #define NAME_OF(o)          o->id
 #define VALUE_TYPE(o)           o->valType
 #define STRINGZ_VALUE_OF(o)     o->val.strs
--
2.11.0

