diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' kcalcore-17.12.0.orig/src/versit/vcc.y kcalcore-17.12.0/src/versit/vcc.y
--- kcalcore-17.12.0.orig/src/versit/vcc.y	2017-12-02 08:21:00.000000000 +0100
+++ kcalcore-17.12.0/src/versit/vcc.y	2017-12-15 09:22:03.721503400 +0100
@@ -99,7 +99,6 @@
 #endif
 
 #include <string.h>
-#include <strings.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <ctype.h>
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' kcalcore-17.12.0.orig/src/versit/vobject.c kcalcore-17.12.0/src/versit/vobject.c
--- kcalcore-17.12.0.orig/src/versit/vobject.c	2017-12-02 08:21:00.000000000 +0100
+++ kcalcore-17.12.0/src/versit/vobject.c	2017-12-15 09:21:31.499946600 +0100
@@ -46,7 +46,6 @@
 
 #include "vobject.h"
 #include <string.h>
-#include <strings.h>
 #include <stdio.h>
 
 #define NAME_OF(o)          o->id
