diff --git b/kate/katetabbar.cpp a/kate/katetabbar.cpp
index 23718e728..8f55d0eaa 100644
--- b/kate/katetabbar.cpp
+++ a/kate/katetabbar.cpp
@@ -127,6 +127,14 @@ void KateTabBar::mousePressEvent(QMouseEvent *event)
         emit activateViewSpaceRequested();
     }
     QTabBar::mousePressEvent(event);
+
+    // handle close for middle mouse button
+    if (event->button() == Qt::MidButton) {
+        int id = tabAt(event->pos());
+        if (id >= 0) {
+            emit tabCloseRequested(id);
+        }
+    }
 }
 
 void KateTabBar::contextMenuEvent(QContextMenuEvent *ev)
diff --git b/kate/kateviewmanager.cpp a/kate/kateviewmanager.cpp
index b8467107d..bcbaedd94 100644
--- b/kate/kateviewmanager.cpp
+++ a/kate/kateviewmanager.cpp
@@ -405,7 +405,12 @@ void KateViewManager::documentsDeleted(const QList<KTextEditor::Document *> &)
             }
         }
 
-        emit viewChanged(newActiveView);
+        /**
+         * reactivate will ensure we really merge up the GUI again
+         * this might be missed as above we had m_blockViewCreationAndActivation set to true
+         * see bug 426605, no view XMLGUI stuff merged after tab close
+         */
+        reactivateActiveView();
     }
 
     /**
