diff --git a/kate/katetabbutton.cpp b/kate/katetabbutton.cpp
index f60f482f721877b479bb2dfc6cea60bd01552111..1273eac3b0bf2b08e2e9de2e0b2cd35efea597e4 100644
--- a/kate/katetabbutton.cpp
+++ b/kate/katetabbutton.cpp
@@ -44,7 +44,7 @@ TabCloseButton::TabCloseButton(QWidget *parent)
 
     // closing a tab closes the document
     setToolTip(i18n("Close Document"));
-    setIcon(QIcon::fromTheme(QStringLiteral("document-close")));
+    setIcon(QIcon::fromTheme(QStringLiteral("tab-close")));
 }
 
 void TabCloseButton::paintEvent(QPaintEvent *)
@@ -61,6 +61,7 @@ void TabCloseButton::paintEvent(QPaintEvent *)
     // removes the QStyleOptionButton::HasMenu ButtonFeature
     opt.features = QStyleOptionButton::Flat;
     opt.state.setFlag(QStyle::State_Enabled, isActive);
+    opt.state.setFlag(QStyle::State_MouseOver, false);
 
     QPainter painter(this);
     style()->drawControl(QStyle::CE_PushButton, &opt, &painter, this);
