diff --git a/kate/main.cpp b/kate/main.cpp
index 1313b4cd5..e9528b3b6 100644
--- a/kate/main.cpp
+++ b/kate/main.cpp
@@ -80,6 +80,13 @@ int main(int argc, char **argv)
     QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps, true);
     QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling, true);
 
+    /**
+     * allow fractional scaling
+     */
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+    QGuiApplication::setHighDpiScaleFactorRoundingPolicy(Qt::HighDpiScaleFactorRoundingPolicy::PassThrough);
+#endif
+
     /**
      * Create application first
      */
diff --git a/kwrite/main.cpp b/kwrite/main.cpp
index 51ea46ac4..8bdcae2a9 100644
--- a/kwrite/main.cpp
+++ b/kwrite/main.cpp
@@ -70,6 +70,13 @@ extern "C" Q_DECL_EXPORT int main(int argc, char **argv)
     QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps, true);
     QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling, true);
 
+    /**
+     * allow fractional scaling
+     */
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+    QGuiApplication::setHighDpiScaleFactorRoundingPolicy(Qt::HighDpiScaleFactorRoundingPolicy::PassThrough);
+#endif
+
     /**
      * Create application first
      * Enforce application name even if the executable is renamed
