diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' kate-19.04.3.orig/kate/katemdi.cpp kate-19.04.3/kate/katemdi.cpp
--- kate-19.04.3.orig/kate/katemdi.cpp	2019-07-06 11:17:49.000000000 +0200
+++ kate-19.04.3/kate/katemdi.cpp	2019-08-13 10:59:00.000000000 +0200
@@ -34,6 +34,7 @@
 #include <kwindowconfig.h>
 #include <KToolBar>
 
+#include <QBoxLayout>
 #include <QChildEvent>
 #include <QEvent>
 #include <QContextMenuEvent>
