diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user kactivities-stats-5.74.0.orig/src/utils/lazy_val.h kactivities-stats-5.74.0/src/utils/lazy_val.h
--- kactivities-stats-5.74.0.orig/src/utils/lazy_val.h	2020-09-06 11:06:26.000000000 +0200
+++ kactivities-stats-5.74.0/src/utils/lazy_val.h	2020-09-14 09:59:43.776334000 +0200
@@ -25,11 +25,7 @@
     mutable bool valueRetrieved;
 
 public:
-#ifdef _MSC_VER
-    operator auto() const -> decltype(_f())
-#else
     operator decltype(_f()) () const
-#endif
     {
         if (!valueRetrieved) {
             valueRetrieved = true;
