diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user akonadi-25.08.0.orig/src/agentbase/agentbase.cpp akonadi-25.08.0/src/agentbase/agentbase.cpp
--- akonadi-25.08.0.orig/src/agentbase/agentbase.cpp	2025-08-07 03:24:18.000000000 +0000
+++ akonadi-25.08.0/src/agentbase/agentbase.cpp	2025-08-21 11:21:39.306436475 +0000
@@ -750,7 +750,7 @@
     // man mallinfo for more info
     QString str;
 #if defined __GLIBC__
-#if defined(__GLIBC_PREREQ) && __GNUC_PREREQ(2, 33)
+#if defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2, 33)
     struct mallinfo2 mi;
     mi = mallinfo2();
 #else
