diff --git a/autotests/libs/testresource/CMakeLists.txt b/autotests/libs/testresource/CMakeLists.txt
--- a/autotests/libs/testresource/CMakeLists.txt
+++ b/autotests/libs/testresource/CMakeLists.txt
@@ -26,12 +26,7 @@
 )
 
 add_executable(akonadi_knut_resource ${knutresource_SRCS})
-
-if (APPLE)
-  set_target_properties(akonadi_knut_resource PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
-  set_target_properties(akonadi_knut_resource PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.Akonadi.Knut")
-  set_target_properties(akonadi_knut_resource PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "KDE Akonadi Knut Resource")
-endif ()
+set_target_properties(akonadi_knut_resource PROPERTIES MACOSX_BUNDLE FALSE)
 
 target_link_libraries(akonadi_knut_resource
   KF5::AkonadiXml
diff --git a/src/agentserver/CMakeLists.txt b/src/agentserver/CMakeLists.txt
--- a/src/agentserver/CMakeLists.txt
+++ b/src/agentserver/CMakeLists.txt
@@ -9,6 +9,7 @@
 ecm_qt_declare_logging_category(akonadi_agent_server_srcs HEADER akonadiagentserver_debug.h IDENTIFIER AKONADIAGENTSERVER_LOG CATEGORY_NAME org.kde.pim.akonadiagentserver)
 
 add_executable(akonadi_agent_server ${akonadi_agent_server_srcs})
+set_target_properties(akonadi_agent_server PROPERTIES MACOSX_BUNDLE FALSE)
 
 target_link_libraries(akonadi_agent_server
     akonadi_shared
@@ -25,29 +26,19 @@
     akonadiagentserver_debug.cpp
 )
 
-add_executable(akonadi_agent_launcher MACOSX_BUNDLE ${akonadi_agent_launcher_SRCS})
+add_executable(akonadi_agent_launcher ${akonadi_agent_launcher_SRCS})
+set_target_properties(akonadi_agent_launcher PROPERTIES MACOSX_BUNDLE FALSE)
 
 target_link_libraries(akonadi_agent_launcher
     akonadi_shared
     KF5AkonadiPrivate
     Qt5::Core
     Qt5::Widgets
 )
 
-if(Q_WS_MAC)
-  set_target_properties(akonadi_agent_launcher PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/../Info.plist.template)
-  set_target_properties(akonadi_agent_launcher PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.Akonadi.agentlauncher")
-  set_target_properties(akonadi_agent_launcher PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "Akonadi Agent Launcher")
-endif()
-
 # Install both helper apps.
-if(Q_WS_MAC)
-  install(TARGETS akonadi_agent_launcher
-          DESTINATION ${AKONADI_BUNDLE_PATH})
-else()
-  install(TARGETS akonadi_agent_launcher
+install(TARGETS akonadi_agent_launcher
           DESTINATION ${BIN_INSTALL_DIR})
-endif()
 
 install(TARGETS akonadi_agent_server
         ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
diff --git a/src/akonadicontrol/CMakeLists.txt b/src/akonadicontrol/CMakeLists.txt
--- a/src/akonadicontrol/CMakeLists.txt
+++ b/src/akonadicontrol/CMakeLists.txt
@@ -31,6 +31,7 @@
 qt5_add_dbus_interface(control_SRCS ${Akonadi_SOURCE_DIR}/src/interfaces/org.freedesktop.Akonadi.PreprocessorManager.xml preprocessor_manager)
 
 add_executable(akonadi_control ${control_SRCS})
+set_target_properties(akonadi_control PROPERTIES MACOSX_BUNDLE FALSE)
 set_target_properties(akonadi_control PROPERTIES OUTPUT_NAME akonadi_control)
 
 if (WIN32)
diff --git a/src/akonadictl/CMakeLists.txt b/src/akonadictl/CMakeLists.txt
--- a/src/akonadictl/CMakeLists.txt
+++ b/src/akonadictl/CMakeLists.txt
@@ -15,6 +15,7 @@
 )
 
 add_executable(akonadictl ${akonadictl_SRCS})
+set_target_properties(akonadictl PROPERTIES MACOSX_BUNDLE FALSE)
 set_target_properties(akonadictl PROPERTIES OUTPUT_NAME akonadictl)
 ecm_mark_nongui_executable(akonadictl)
 target_link_libraries(akonadictl
diff --git a/src/asapcat/CMakeLists.txt b/src/asapcat/CMakeLists.txt
--- a/src/asapcat/CMakeLists.txt
+++ b/src/asapcat/CMakeLists.txt
@@ -4,6 +4,7 @@
 )
 
 add_executable(asapcat ${asapcat_srcs})
+set_target_properties(asapcat PROPERTIES MACOSX_BUNDLE FALSE)
 
 target_link_libraries(asapcat
     akonadi_shared
diff --git a/src/private/protocolgen/CMakeLists.txt b/src/private/protocolgen/CMakeLists.txt
--- a/src/private/protocolgen/CMakeLists.txt
+++ b/src/private/protocolgen/CMakeLists.txt
@@ -10,6 +10,7 @@
 )
 
 add_executable(protocolgen ${protocolgen_SRCS})
+set_target_properties(protocolgen PROPERTIES MACOSX_BUNDLE FALSE)
 target_link_libraries(protocolgen
     Qt5::Core
 )
diff --git a/src/rds/CMakeLists.txt b/src/rds/CMakeLists.txt
--- a/src/rds/CMakeLists.txt
+++ b/src/rds/CMakeLists.txt
@@ -7,6 +7,7 @@
 )
 
 add_executable(akonadi_rds ${akonadi_rds_srcs})
+set_target_properties(akonadi_rds PROPERTIES MACOSX_BUNDLE FALSE)
 
 target_link_libraries(akonadi_rds
     akonadi_shared
diff --git a/src/server/CMakeLists.txt b/src/server/CMakeLists.txt
--- a/src/server/CMakeLists.txt
+++ b/src/server/CMakeLists.txt
@@ -170,6 +170,7 @@
 )
 
 add_executable(akonadiserver ${akonadiserver_SRCS})
+set_target_properties(akonadiserver PROPERTIES MACOSX_BUNDLE FALSE)
 set_target_properties(akonadiserver PROPERTIES OUTPUT_NAME akonadiserver)
 target_link_libraries(akonadiserver
     libakonadiserver
diff --git a/src/xml/CMakeLists.txt b/src/xml/CMakeLists.txt
--- a/src/xml/CMakeLists.txt
+++ b/src/xml/CMakeLists.txt
@@ -32,6 +32,7 @@
 )
 
 add_executable(akonadi2xml akonadi2xml.cpp)
+set_target_properties(akonadi2xml PROPERTIES MACOSX_BUNDLE FALSE)
 
 target_link_libraries(akonadi2xml
     KF5::AkonadiXml
diff --git a/templates/akonadiresource/src/CMakeLists.txt b/templates/akonadiresource/src/CMakeLists.txt
--- a/templates/akonadiresource/src/CMakeLists.txt
+++ b/templates/akonadiresource/src/CMakeLists.txt
@@ -24,6 +24,7 @@
 )
 
 add_executable(akonadi_%{APPNAMELC}_resource ${%{APPNAMELC}resource_SRCS})
+set_target_properties(akonadi_%{APPNAMELC}_resource PROPERTIES MACOSX_BUNDLE FALSE)
 
 target_link_libraries(akonadi_%{APPNAMELC}_resource
     Qt5::DBus

