diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' akonadi-17.12.0.orig/CMakeLists.txt akonadi-17.12.0/CMakeLists.txt
--- akonadi-17.12.0.orig/CMakeLists.txt	2017-12-08 02:01:14.000000000 +0100
+++ akonadi-17.12.0/CMakeLists.txt	2017-12-20 14:28:02.914076000 +0100
@@ -72,7 +72,7 @@
 set_package_properties(KF5DesignerPlugin PROPERTIES DESCRIPTION "KF5 designer plugin" TYPE OPTIONAL)
 
 set(Boost_MINIMUM_VERSION "1.34.0")
-find_package(Boost ${Boost_MINIMUM_VERSION})
+find_package(Boost ${Boost_MINIMUM_VERSION} REQUIRED COMPONENTS graph)
 set_package_properties(Boost PROPERTIES
     DESCRIPTION "Boost C++ Libraries"
     URL "http://www.boost.org"
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' akonadi-17.12.0.orig/src/agentbase/CMakeLists.txt akonadi-17.12.0/src/agentbase/CMakeLists.txt
--- akonadi-17.12.0.orig/src/agentbase/CMakeLists.txt	2017-12-04 17:40:18.000000000 +0100
+++ akonadi-17.12.0/src/agentbase/CMakeLists.txt	2017-12-20 14:28:02.922074800 +0100
@@ -47,7 +47,7 @@
 
 add_library(KF5::AkonadiAgentBase ALIAS KF5AkonadiAgentBase)
 
-target_include_directories(KF5AkonadiAgentBase INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/AkonadiAgentBase;${Boost_INCLUDE_DIR}>")
+target_include_directories(KF5AkonadiAgentBase INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/AkonadiAgentBase>")
 
 target_link_libraries(KF5AkonadiAgentBase
 PUBLIC
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' akonadi-17.12.0.orig/src/core/CMakeLists.txt akonadi-17.12.0/src/core/CMakeLists.txt
--- akonadi-17.12.0.orig/src/core/CMakeLists.txt	2017-12-04 17:40:18.000000000 +0100
+++ akonadi-17.12.0/src/core/CMakeLists.txt	2017-12-20 14:28:02.931072900 +0100
@@ -282,10 +282,9 @@
 generate_export_header(KF5AkonadiCore BASE_NAME akonadicore)
 
 add_library(KF5::AkonadiCore ALIAS KF5AkonadiCore)
-target_include_directories(KF5AkonadiCore INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/AkonadiCore;${Boost_INCLUDE_DIR}>")
+target_include_directories(KF5AkonadiCore INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/AkonadiCore>")
 target_include_directories(KF5AkonadiCore PUBLIC "$<BUILD_INTERFACE:${Akonadi_SOURCE_DIR}/src/core;${Akonadi_BINARY_DIR}/src/core;${Akonadi_BINARY_DIR}/src/private>")
 target_include_directories(KF5AkonadiCore PUBLIC "$<BUILD_INTERFACE:${Akonadi_BINARY_DIR};${Akonadi_SOURCE_DIR}/src/core/jobs;${Akonadi_SOURCE_DIR}/src/core/models>")
-target_include_directories(KF5AkonadiCore PUBLIC "$<BUILD_INTERFACE:${Boost_INCLUDE_DIR}>")
 
 kde_target_enable_exceptions(KF5AkonadiCore PUBLIC)
 
@@ -294,6 +293,7 @@
     KF5::CoreAddons # for KJob
     KF5::ItemModels
     Qt5::Gui # for QColor
+    Boost::boost
 PRIVATE
     Qt5::Network
     Qt5::Widgets
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' akonadi-17.12.0.orig/src/shared/akdebug.cpp akonadi-17.12.0/src/shared/akdebug.cpp
--- akonadi-17.12.0.orig/src/shared/akdebug.cpp	2017-12-04 17:40:18.000000000 +0100
+++ akonadi-17.12.0/src/shared/akdebug.cpp	2017-12-20 14:28:02.834578400 +0100
@@ -200,7 +200,7 @@
         QFile file(info.absoluteFilePath());
         const bool success = file.rename(sInstance()->errorLogFileName() + QLatin1String(".old"));
         if (!success) {
-            qFatal("Cannot rename log file - running on a readonly filesystem maybe?");
+            qWarning("Cannot rename log file - running on a readonly filesystem maybe?");
         }
     }
 
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' akonadi-17.12.0.orig/src/widgets/CMakeLists.txt akonadi-17.12.0/src/widgets/CMakeLists.txt
--- akonadi-17.12.0.orig/src/widgets/CMakeLists.txt	2017-12-04 17:40:18.000000000 +0100
+++ akonadi-17.12.0/src/widgets/CMakeLists.txt	2017-12-20 14:28:02.941054200 +0100
@@ -87,7 +87,7 @@
 
 add_library(KF5::AkonadiWidgets ALIAS KF5AkonadiWidgets)
 
-target_include_directories(KF5AkonadiWidgets INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/AkonadiWidgets;${Boost_INCLUDE_DIR}>")
+target_include_directories(KF5AkonadiWidgets INTERFACE "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF5}/AkonadiWidgets>")
 
 target_link_libraries(KF5AkonadiWidgets
 PUBLIC
