commit 1fe4801638f1f34d9bdb139a949cc330c27074b9
Author: David Faure <faure@kde.org>
Date:   Thu Jun 6 16:26:30 2019 +0200

    Show error string from QLocalServer on failure

diff --git a/src/server/akonadi.cpp b/src/server/akonadi.cpp
index d5b714f38..883aca4a2 100644
--- a/src/server/akonadi.cpp
+++ b/src/server/akonadi.cpp
@@ -124,7 +124,7 @@ bool AkonadiServer::init()
     const QString defaultCmdPipe = QStringLiteral("Akonadi-Cmd-") % suffix;
     const QString cmdPipe = settings.value(QStringLiteral("Connection/NamedPipe"), defaultCmdPipe).toString();
     if (!mCmdServer->listen(cmdPipe)) {
-        qCCritical(AKONADISERVER_LOG) << "Unable to listen on Named Pipe" << cmdPipe;
+        qCCritical(AKONADISERVER_LOG) << "Unable to listen on Named Pipe" << cmdPipe << ":" << mCmdServer->errorString();
         quit();
         return false;
     }
@@ -132,7 +132,7 @@ bool AkonadiServer::init()
     const QString defaultNtfPipe = QStringLiteral("Akonadi-Ntf-") % suffix;
     const QString ntfPipe = settings.value(QStringLiteral("Connection/NtfNamedPipe"), defaultNtfPipe).toString();
     if (!mNtfServer->listen(ntfPipe)) {
-        qCCritical(AKONADISERVER_LOG) << "Unable to listen on Named Pipe" << ntfPipe;
+        qCCritical(AKONADISERVER_LOG) << "Unable to listen on Named Pipe" << ntfPipe << ":" << mNtfServer->errorString();
         quit();
         return false;
     }
@@ -146,7 +146,7 @@ bool AkonadiServer::init()
     const QString cmdSocketFile = socketDir % QStringLiteral("/akonadiserver-cmd.socket");
     QFile::remove(cmdSocketFile);
     if (!mCmdServer->listen(cmdSocketFile)) {
-        qCCritical(AKONADISERVER_LOG) << "Unable to listen on Unix socket" << cmdSocketFile;
+        qCCritical(AKONADISERVER_LOG) << "Unable to listen on Unix socket" << cmdSocketFile << ":" << mCmdServer->errorString();
         quit();
         return false;
     }
@@ -154,7 +154,7 @@ bool AkonadiServer::init()
     const QString ntfSocketFile = socketDir % QStringLiteral("/akonadiserver-ntf.socket");
     QFile::remove(ntfSocketFile);
     if (!mNtfServer->listen(ntfSocketFile)) {
-        qCCritical(AKONADISERVER_LOG) << "Unable to listen on Unix socket" << ntfSocketFile;
+        qCCritical(AKONADISERVER_LOG) << "Unable to listen on Unix socket" << ntfSocketFile << ":" << mNtfServer->errorString();
         quit();
         return false;
     }
