commit c3654167bd9f3f9e657f8a3634bf56957e2fde87
Author: Albert Astals Cid <aacid@kde.org>
Date:   Wed Feb 2 18:58:25 2022 +0100

    Fix getting the home dir in Windows

diff --git a/poppler/SignatureHandler.cc b/poppler/SignatureHandler.cc
index 43fc974e..22a620c6 100644
--- a/poppler/SignatureHandler.cc
+++ b/poppler/SignatureHandler.cc
@@ -669,7 +669,7 @@ std::unique_ptr<X509CertificateInfo> SignatureHandler::getCertificateInfo() cons
 static std::optional<std::string> getDefaultFirefoxCertDB()
 {
 #ifdef _WIN32
-    const std::string firefoxPath = std::string(getenv("HOME")) + "/AppData/Roaming/Mozilla/Firefox/Profiles/";
+    const std::string firefoxPath = std::string(getenv("USERPROFILE")) + "/AppData/Roaming/Mozilla/Firefox/Profiles/";
 #else
     const std::string firefoxPath = std::string(getenv("HOME")) + "/.mozilla/firefox/";
 #endif
