diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' poppler-0.62.0.orig/cmake/modules/FindLIBOPENJPEG2.cmake poppler-0.62.0/cmake/modules/FindLIBOPENJPEG2.cmake
--- poppler-0.62.0.orig/cmake/modules/FindLIBOPENJPEG2.cmake	2017-12-03 20:25:06.000000000 +0100
+++ poppler-0.62.0/cmake/modules/FindLIBOPENJPEG2.cmake	2018-03-14 09:17:15.832253700 +0100
@@ -18,12 +18,11 @@
 
 else ()
 
-  set(LIBOPENJPEG2_FOUND FALSE)
-  set(LIBOPENJPEG2_INCLUDE_DIRS)
-  set(LIBOPENJPEG2_LIBRARIES)
+  find_package(OPENJPEG REQUIRED)
+  set(LIBOPENJPEG2_FOUND ${OPENJPEG_FOUND})
+  set(LIBOPENJPEG2_INCLUDE_DIRS ${OPENJPEG_INCLUDE_DIRS})
+  set(LIBOPENJPEG2_LIBRARIES ${OPENJPEG_LIBRARIES})
 
-  find_package(PkgConfig REQUIRED)
-  pkg_check_modules(LIBOPENJPEG2 libopenjp2)
   if (LIBOPENJPEG2_FOUND)
     add_definitions(-DUSE_OPENJPEG2)
   endif ()
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' poppler-0.62.0.orig/CMakeLists.txt poppler-0.62.0/CMakeLists.txt
--- poppler-0.62.0.orig/CMakeLists.txt	2017-12-03 20:25:06.000000000 +0100
+++ poppler-0.62.0/CMakeLists.txt	2018-03-14 09:17:15.834263400 +0100
@@ -477,7 +477,6 @@
 
 if(MSVC)
 add_definitions(-D_CRT_SECURE_NO_WARNINGS)
-set(CMAKE_CXX_FLAGS "/Zc:wchar_t- ${CMAKE_CXX_FLAGS}")
 add_library(poppler STATIC ${poppler_SRCS})
 else()
 add_library(poppler ${poppler_SRCS})
diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' poppler-0.62.0.orig/goo/glibc.h poppler-0.62.0/goo/glibc.h
--- poppler-0.62.0.orig/goo/glibc.h	2017-12-03 20:25:06.000000000 +0100
+++ poppler-0.62.0/goo/glibc.h	2018-03-14 09:24:45.710896200 +0100
@@ -17,8 +17,6 @@
 
 #include <time.h>
 
-extern "C" {
-
 #ifndef HAVE_GMTIME_R
 struct tm *gmtime_r(const time_t *timep, struct tm *result);
 #endif
@@ -35,7 +33,6 @@
 char * strtok_r (char *s, const char *delim, char **save_ptr);
 #endif
 
-}
 
 #endif // GLIBC_H
 
