From cb5cd1e457953bdd77382bd947a3e16e0c15bab8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20K=C3=BCnzel?= <julius.kuenzel@kde.org>
Date: Sat, 17 Aug 2024 23:34:37 +0000
Subject: [PATCH] Allow recoloring of Android icon theme

---
 KF6KirigamiMacros.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/KF6KirigamiMacros.cmake b/KF6KirigamiMacros.cmake
index 4500e7584..435cc0ee5 100644
--- a/KF6KirigamiMacros.cmake
+++ b/KF6KirigamiMacros.cmake
@@ -126,6 +126,6 @@ function(kirigami_package_breeze_icons)
     endforeach()
 
     #generate an index.theme that qiconloader can understand
-    file(WRITE ${CMAKE_BINARY_DIR}/index.theme "[Icon Theme]\nName=Breeze\nDirectories=icons\n[icons]\nSize=32\nType=Scalable")
+    file(WRITE ${CMAKE_BINARY_DIR}/index.theme "[Icon Theme]\nName=Breeze\nDirectories=icons\nFollowsColorScheme=true\n[icons]\nSize=32\nType=Scalable")
     install(FILES ${CMAKE_BINARY_DIR}/index.theme DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/kirigami/breeze-internal/)
 endfunction()
-- 
GitLab

