diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x '*\.o' -x '*\.pyc' -x CMakeLists.txt.user XML-Parser-2.47.orig/Makefile.PL XML-Parser-2.47/Makefile.PL
--- XML-Parser-2.47.orig/Makefile.PL	2023-12-29 01:20:24.000000000 +0100
+++ XML-Parser-2.47/Makefile.PL	2024-04-08 14:33:09.982993900 +0200
@@ -30,7 +30,7 @@
 
 unless (
     check_lib(    # fill in what you prompted the user for here
-        lib     => [qw(expat)],
+        lib     => [qw(libexpat)],
         header  => ['expat.h'],
         incpath => $expat_incpath,
         ( $expat_libpath ? ( libpath => $expat_libpath ) : () ),
