From 444f9bf66b8721542adf7155693d732540d8e749 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Sat, 11 Dec 2021 17:28:24 +0100
Subject: [PATCH] Use GNUInstallDirs also on Windows

---
 CMakeLists.txt | 18 ++----------------
 1 file changed, 2 insertions(+), 16 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 750ed771..7c815ab5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,25 +51,8 @@ else()
     add_compile_options(-fno-exceptions)
 endif()

-if (WIN32)
-    if (NOT CMAKE_INSTALL_LIBDIR)
-        set(CMAKE_INSTALL_LIBDIR ".")
-        set(CMakeFilesLocation "cmake")
-    else()
-        set(CMakeFilesLocation "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
-    endif()
-
-    if (NOT CMAKE_INSTALL_BINDIR)
-        set(CMAKE_INSTALL_BINDIR ".")
-    endif()
-
-    if (NOT CMAKE_INSTALL_INCLUDEDIR)
-        set(CMAKE_INSTALL_INCLUDEDIR "include")
-    endif()
-else()
-    include(GNUInstallDirs)
-    set(CMakeFilesLocation "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
-endif(WIN32)
+include(GNUInstallDirs)
+set(CMakeFilesLocation "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
 set(${PROJECT_NAME}_INSTALL_INCLUDEDIR
         "${CMAKE_INSTALL_INCLUDEDIR}/" CACHE PATH
         "directory to install ${PROJECT_NAME} include files to")
-- 
2.34.1

